---
title: "RBD_explorations"
output: github_document
---

```{r Set up the analysis script}

## Clear existing environment
rm(list = ls())

## Load basic useful packages
library(tidyverse)
library(ggrepel)
library(ggbeeswarm)
library(reshape)
library(sf)
library(here)
library(ggfortify)
library(googlesheets4)
library(rpart)
library(rpart.plot)
library(ggbreak) 

## Set the seed for immediate reproducibility
set.seed(1234567)

## Set the base theme to what I like
theme_set(theme_bw())
theme_update(legend.title = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 10))

## Setting some universal thresholds for some of the subsequent analyses
quantile_cutoff <- 0.99
minquant_fraction <- 0.2

to_single_notation <- function(arg1){
  if(toupper(arg1) == "ALA"){return("A")}
  if(toupper(arg1) == "CYS"){return("C")}
  if(toupper(arg1) == "ASP"){return("D")}
  if(toupper(arg1) == "GLU"){return("E")}
  if(toupper(arg1) == "GLH"){return("E")}
  if(toupper(arg1) == "PHE"){return("F")}
  if(toupper(arg1) == "GLY"){return("G")}
  if(toupper(arg1) == "HIS"){return("H")}
  if(toupper(arg1) == "ILE"){return("I")}
  if(toupper(arg1) == "LYS"){return("K")}
  if(toupper(arg1) == "LEU"){return("L")}
  if(toupper(arg1) == "MET"){return("M")}
  if(toupper(arg1) == "ASN"){return("N")}
  if(toupper(arg1) == "PRO"){return("P")}
  if(toupper(arg1) == "GLN"){return("Q")}
  if(toupper(arg1) == "ARG"){return("R")}
  if(toupper(arg1) == "SER"){return("S")}
  if(toupper(arg1) == "THR"){return("T")}
  if(toupper(arg1) == "VAL"){return("V")}
  if(toupper(arg1) == "TRP"){return("W")}
  if(toupper(arg1) == "TYR"){return("Y")}
  if(toupper(arg1) == "TER"){return("X")}
}

```


```{r Now bringing in our most recent data}
### Next make a function for analyzing the enrichment of the virus in hygromycin antibiotic (thus selecting for infected cells)
index_key1 <- read.csv(file = "Keys/9ntR1_10ntR2.csv", header = T, stringsAsFactors = F) %>% mutate(primer1 = primer, primer2 = primer)
sample_key1 <- read.csv(file = "Keys/Barcode_receptor_samples.csv", header = T, stringsAsFactors = F)

sample_index_key1 <- merge(sample_key1[,c("gene","ortholog","mutant","protease","kozak","plasmid_template","primer1","primer2","concat")], index_key1[,c("primer2","index")], by = "primer2", all.x = T)
sample_index_key1 <- merge(sample_index_key1, index_key1[,c("primer1","index")], by = "primer1", all.x = T)
sample_index_key1$X <- paste(sample_index_key1$index.x,"N",sample_index_key1$index.y, sep = "")
sample_index_key1$sequence <- sample_index_key1$X

## Current list of NGS data
myfiles = list.files(path="Data/Dbl_barcode_NS3", pattern="*.tsv", full.names=TRUE)
myfiles_df <- data.frame("number" = seq(1,length(myfiles)), "index" = myfiles)

## Now defining the function to analyze the enrichement with the NGS data
list_of_indexes <- c(4,5,8,9) ## Delete whenever. Only for troubleshooting.

makeExperimentFrame1 <- function(list_of_indexes){
  ## Deal with the unselected data first
  rep1 <- read.delim(myfiles[list_of_indexes[1]], sep = "\t") %>% mutate(log10_count = log10(count))
  # First filter is to only look at actual designer barcodes
  rep1_filtered1 <- rep1 %>% filter(X %in% sample_index_key1$X)
  # Second filter will be to ignore barcodes with counts so low they are unlikely real
  rep1_filtered2 <- rep1_filtered1 %>% filter(log10_count > (mean(log10_count) - sd(log10_count) * 2))
  
  ## Repeat the process for the second replicate
  rep2 <- read.delim(myfiles[list_of_indexes[2]], sep = "\t") %>% mutate(log10_count = log10(count))
  rep2_filtered1 <- rep2 %>% filter(X %in% sample_index_key1$X)
  rep2_filtered2 <- rep2_filtered1 %>% filter(log10_count > (mean(log10_count) - sd(log10_count) * 2))
  
  unsel <- merge(rep1_filtered2, rep2_filtered2, by = "X", all = T)
  unsel[is.na(unsel)] <- 0
  unsel$rep1_freq <- unsel$count.x / sum(unsel$count.x)
  unsel$rep2_freq <- unsel$count.y / sum(unsel$count.y)
  unsel$u_freq <- 10^((log10(unsel$rep1_freq) + log10(unsel$rep2_freq))/2)
  
  ## Deal with the HygroR data next
  hygro <- merge(read.delim(myfiles[list_of_indexes[3]], sep = "\t"), read.delim(myfiles[list_of_indexes[4]], sep = "\t"), by = "X", all = T)
  ## Combine the data; this should also take care of the filtering, since the unsel was already filtered
  combined_frame <- merge(unsel[,c("X","u_freq")], hygro[,c("X","count.x","count.y")], by = "X", all.x = T) %>% 
    mutate(sel_freq1 = count.x / sum(count.x), sel_freq2 = count.y / sum(count.y)) %>% 
    mutate(h_freq = 10^((log10(sel_freq1) + log10(sel_freq2))/2))
  return_frame <- combined_frame[,c("X","u_freq","h_freq")]
  
  ## Do some additional analysis before returning the data frame
  return_frame$h_enrichment <- return_frame$h_freq / return_frame$u_freq
  colnames(return_frame) <- c("sequence","u_freq","h_freq","h_enrichment")
  return_frame2_troubleshooting <- merge(return_frame, sample_index_key1[,c("gene","ortholog","mutant","protease","kozak","plasmid_template","concat","sequence")], by = "sequence", all = T)
  return_frame2 <- merge(return_frame, sample_index_key1[,c("gene","ortholog","mutant","protease","kozak","plasmid_template","concat","sequence")], by = "sequence", all.x = T)
  return(return_frame2)
}
```


```{r Identity Distance Matrix for the tested RBDs}
identity <- read.csv(file = "Data/Grantham/RBD_identity_matrix.csv", header= T, stringsAsFactors = F)
rownames(identity) <- identity$X
identity2 <- identity[,2:ncol(identity)]
identity_pca = prcomp(identity2, scale. = TRUE)

identity_scores = as.data.frame(identity_pca$x)
identity_scores$label <- c("D614G", "Alpha", "Beta","Delta","Gamma","BA1")

identity_PCA_plot <- ggplot() + 
  xlab(paste("Principal component 1\n(",round(summary(identity_pca)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Principal component 2\n(",round(summary(identity_pca)$importance[5] * 100,0),"% of variance)",sep="")) + 
  geom_point(data = identity_scores, aes(x = PC1, y = PC2)) +
  geom_text_repel(data = identity_scores, aes(x = PC1, y = PC2, label = label), color = "red", size = 2)
identity_PCA_plot
ggsave(file = "Plots/identity_PCA_plot.pdf", identity_PCA_plot, height = 1.8, width = 1.75)

```


```{r Grantham Distance Matrix for the tested RBDs}
grantham <- read.csv(file = "Data/Grantham/RBD_Grantham_matrix.csv", header= T, stringsAsFactors = F)
rownames(grantham) <- grantham$X
grantham2 <- grantham[,2:ncol(grantham)]
grantham_pca = prcomp(grantham2, scale. = TRUE)

grantham_scores = as.data.frame(grantham_pca$x)
grantham_scores$label <- c("D614G", "Alpha", "Beta","Delta","Gamma","BA1")

Grantham_PCA_plot <- ggplot() + 
  xlab(paste("Principal component 1\n(",round(summary(grantham_pca)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Principal component 2\n(",round(summary(grantham_pca)$importance[5] * 100,0),"% of variance)",sep="")) + 
  geom_point(data = grantham_scores, aes(x = PC1, y = PC2)) +
  geom_text_repel(data = grantham_scores, aes(x = PC1, y = PC2, label = label), color = "red", size = 2)
Grantham_PCA_plot
ggsave(file = "Plots/Grantham_PCA_plot.pdf", Grantham_PCA_plot, height = 1.8, width = 1.75)

```

```{r Looking at correlations of the uninfected cells with library v1.0}
i0192 <- read.delim(file = "Data/Dbl_barcode_NS3/I0192_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0193 <- read.delim(file = "Data/Dbl_barcode_NS3/I0193_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0194 <- read.delim(file = "Data/Dbl_barcode_NS3/I0194_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0195 <- read.delim(file = "Data/Dbl_barcode_NS3/I0195_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0196 <- read.delim(file = "Data/Dbl_barcode_NS3/I0196_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0197 <- read.delim(file = "Data/Dbl_barcode_NS3/I0197_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0198 <- read.delim(file = "Data/Dbl_barcode_NS3/I0198_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0199 <- read.delim(file = "Data/Dbl_barcode_NS3/I0199_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0200 <- read.delim(file = "Data/Dbl_barcode_NS3/I0200_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0201 <- read.delim(file = "Data/Dbl_barcode_NS3/I0201_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0202 <- read.delim(file = "Data/Dbl_barcode_NS3/I0202_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0203 <- read.delim(file = "Data/Dbl_barcode_NS3/I0203_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0204 <- read.delim(file = "Data/Dbl_barcode_NS3/I0204_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0205 <- read.delim(file = "Data/Dbl_barcode_NS3/I0205_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0206 <- read.delim(file = "Data/Dbl_barcode_NS3/I0206_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0207 <- read.delim(file = "Data/Dbl_barcode_NS3/I0207_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0208 <- read.delim(file = "Data/Dbl_barcode_NS3/I0208_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)

i0192_filtered <- i0192 %>% filter(X %in% sample_index_key1$X)
i0193_filtered <- i0193 %>% filter(X %in% sample_index_key1$X)
i0194_filtered <- i0194 %>% filter(X %in% sample_index_key1$X)
i0195_filtered <- i0195 %>% filter(X %in% sample_index_key1$X)
i0196_filtered <- i0196 %>% filter(X %in% sample_index_key1$X)
i0197_filtered <- i0197 %>% filter(X %in% sample_index_key1$X)
i0198_filtered <- i0198 %>% filter(X %in% sample_index_key1$X)
i0199_filtered <- i0199 %>% filter(X %in% sample_index_key1$X)
i0200_filtered <- i0200 %>% filter(X %in% sample_index_key1$X)
i0201_filtered <- i0201 %>% filter(X %in% sample_index_key1$X)
i0202_filtered <- i0202 %>% filter(X %in% sample_index_key1$X)
i0203_filtered <- i0203 %>% filter(X %in% sample_index_key1$X)
i0204_filtered <- i0204 %>% filter(X %in% sample_index_key1$X)
i0205_filtered <- i0205 %>% filter(X %in% sample_index_key1$X)
i0206_filtered <- i0206 %>% filter(X %in% sample_index_key1$X)
i0207_filtered <- i0207 %>% filter(X %in% sample_index_key1$X)
i0208_filtered <- i0208 %>% filter(X %in% sample_index_key1$X)

ns3_negs_filtered <- merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(i0192_filtered,i0193_filtered,by = "X", all = T),i0194_filtered,by = "X", all = T),i0195_filtered,by = "X", all = T),i0196_filtered,by = "X", all = T),i0197_filtered,by = "X", all = T),i0198_filtered,by = "X", all = T),i0199_filtered,by = "X", all = T),i0200_filtered,by = "X", all = T),i0201_filtered,by = "X", all = T),i0202_filtered,by = "X", all = T),i0203_filtered,by = "X", all = T),i0204_filtered,by = "X", all = T),i0205_filtered,by = "X", all = T),i0206_filtered,by = "X", all = T),i0207_filtered,by = "X", all = T),i0208_filtered,by = "X", all = T)

colnames(ns3_negs_filtered) <- c("X","ns3_u1","ns3_u2","ns3_u3","ns3_u4","ns3_u5","ns3_u6","ns3_u7","ns3_u8","ns3_u9","ns3_u10","ns3_u11","ns3_u12","ns3_u13","ns3_u14","ns3_u15","ns3_u16","ns3_u17")

ns3_negs_filtered$ns3_u1 <- ns3_negs_filtered$ns3_u1 / sum(ns3_negs_filtered$ns3_u1, na.rm = T)
ns3_negs_filtered$ns3_u2 <- ns3_negs_filtered$ns3_u2 / sum(ns3_negs_filtered$ns3_u2, na.rm = T)
ns3_negs_filtered$ns3_u3 <- ns3_negs_filtered$ns3_u3 / sum(ns3_negs_filtered$ns3_u3, na.rm = T)
ns3_negs_filtered$ns3_u4 <- ns3_negs_filtered$ns3_u4 / sum(ns3_negs_filtered$ns3_u4, na.rm = T)
ns3_negs_filtered$ns3_u5 <- ns3_negs_filtered$ns3_u5 / sum(ns3_negs_filtered$ns3_u5, na.rm = T)
ns3_negs_filtered$ns3_u6 <- ns3_negs_filtered$ns3_u6 / sum(ns3_negs_filtered$ns3_u6, na.rm = T)
ns3_negs_filtered$ns3_u7 <- ns3_negs_filtered$ns3_u7 / sum(ns3_negs_filtered$ns3_u7, na.rm = T)
ns3_negs_filtered$ns3_u8 <- ns3_negs_filtered$ns3_u8 / sum(ns3_negs_filtered$ns3_u8, na.rm = T)
ns3_negs_filtered$ns3_u9 <- ns3_negs_filtered$ns3_u9 / sum(ns3_negs_filtered$ns3_u9, na.rm = T)
ns3_negs_filtered$ns3_u10 <- ns3_negs_filtered$ns3_u10 / sum(ns3_negs_filtered$ns3_u10, na.rm = T)
ns3_negs_filtered$ns3_u11 <- ns3_negs_filtered$ns3_u11 / sum(ns3_negs_filtered$ns3_u11, na.rm = T)
ns3_negs_filtered$ns3_u12 <- ns3_negs_filtered$ns3_u12 / sum(ns3_negs_filtered$ns3_u12, na.rm = T)
ns3_negs_filtered$ns3_u13 <- ns3_negs_filtered$ns3_u13 / sum(ns3_negs_filtered$ns3_u13, na.rm = T)
ns3_negs_filtered$ns3_u14 <- ns3_negs_filtered$ns3_u14 / sum(ns3_negs_filtered$ns3_u14, na.rm = T)
ns3_negs_filtered$ns3_u15 <- ns3_negs_filtered$ns3_u15 / sum(ns3_negs_filtered$ns3_u15, na.rm = T)
ns3_negs_filtered$ns3_u16 <- ns3_negs_filtered$ns3_u16 / sum(ns3_negs_filtered$ns3_u16, na.rm = T)
ns3_negs_filtered$ns3_u17 <- ns3_negs_filtered$ns3_u17 / sum(ns3_negs_filtered$ns3_u17, na.rm = T)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = ns3_negs_filtered, aes(x = ns3_u1, y = ns3_u2))

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = ns3_negs_filtered, aes(x = ns3_u1, y = ns3_u3))

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = ns3_negs_filtered, aes(x = ns3_u2, y = ns3_u3))

Neg_cntrl_count_correlations <- ggplot() + 
  labs(x = "Variant frequency in\nreplicate 1", y = "Variant frequency in\nreplicate 2") +
  scale_x_log10() + scale_y_log10() + 
  geom_hline(yintercept = 1e-5, linetype = 2) +
  geom_vline(xintercept = 1e-5, linetype = 2) +
  geom_abline(slope = 1, linetype = 2, alpha = 0.3) +
  geom_point(data = ns3_negs_filtered, aes(x = ns3_u1, y = ns3_u2), alpha = 0.5) +
  #geom_point(data = ns3_negs_filtered, aes(x = freq192, y = freq194), alpha = 0.5) +
  #geom_point(data = ns3_negs_filtered, aes(x = freq193, y = freq194), alpha = 0.5)
  NULL
Neg_cntrl_count_correlations
ggsave(file = "Plots/Neg_cntrl_count_correlations.pdf", Neg_cntrl_count_correlations, height = 1.75, width = 2)

# Correlations are reasonable, suggesting we can indeed reproducibly quantitate things
# R^2 values to report
paste("The R^2 for the unselected frequency counts shown:", round(cor(ns3_negs_filtered$ns3_u1, ns3_negs_filtered$ns3_u2, use = "complete")^2,2))



```

```{r Nextseq 3}
ns3_g740d1 <- makeExperimentFrame1(c(4,5,18,18)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 195, 196, 209, 210
ns3_g740d2 <- makeExperimentFrame1(c(4,5,19,19)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 195, 196, 209, 210
ns3_alpha1 <- makeExperimentFrame1(c(6,7,20,20)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns3_alpha2 <- makeExperimentFrame1(c(6,7,21,21)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns3_beta1 <- makeExperimentFrame1(c(8,9,22,22)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns3_beta2 <- makeExperimentFrame1(c(8,9,23,23)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns3_delta1 <- makeExperimentFrame1(c(10,11,24,24)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns3_delta2 <- makeExperimentFrame1(c(10,11,25,25)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns3_gamma1 <- makeExperimentFrame1(c(12,13,26,26)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 203, 204, 217, 218
ns3_gamma2 <- makeExperimentFrame1(c(12,13,27,27)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 203, 204, 217, 218
ns3_ba1omicron1 <- makeExperimentFrame1(c(14,15,28,28)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns3_ba1omicron2 <- makeExperimentFrame1(c(14,15,29,29)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
```

```{r NextSeq4 data}
### Next make a function for analyzing the enrichment of the virus in hygromycin antibiotic (thus selecting for infected cells)
index_key2 <- read.csv(file = "Keys/10ntR1_10ntR2.csv", header = T, stringsAsFactors = F) %>% mutate(primer1 = primer, primer2 = primer)
sample_key2 <- read.csv(file = "Keys/Barcode_receptor_samples.csv", header = T, stringsAsFactors = F)

sample_index_key2 <- merge(sample_key2[,c("gene","ortholog","mutant","protease","kozak","plasmid_template","primer1","primer2","concat")], index_key2[,c("primer2","index")], by = "primer2", all.x = T)
sample_index_key2 <- merge(sample_index_key2, index_key2[,c("primer1","index")], by = "primer1", all.x = T)
sample_index_key2$X <- paste(sample_index_key2$index.x,sample_index_key2$index.y, sep = "")
sample_index_key2$sequence <- sample_index_key2$X

## Current list of NGS data
myfiles = list.files(path="Data/Dbl_barcode_NS4", pattern="*.tsv", full.names=TRUE)
myfiles_df <- data.frame("number" = seq(1,length(myfiles)), "index" = myfiles)

## Now defining the function to analyze the enrichement with the NGS data
list_of_indexes <- c(3,3,4,4) ## Delete whenever. Only for troubleshooting.

makeExperimentFrame2 <- function(list_of_indexes){
  ## Deal with the unselected data first
  rep1 <- read.delim(myfiles[list_of_indexes[1]], sep = "\t") %>% mutate(log10_count = log10(count))
  # First filter is to only look at actual designer barcodes
  rep1_filtered1 <- rep1 %>% filter(X %in% sample_index_key2$X)
  # Second filter will be to ignore barcodes with counts so low they are unlikely real
  rep1_filtered2 <- rep1_filtered1 %>% filter(log10_count > (mean(log10_count) - sd(log10_count) * 2))
  
  ## Repeat the process for the second replicate
  rep2 <- read.delim(myfiles[list_of_indexes[2]], sep = "\t") %>% mutate(log10_count = log10(count))
  rep2_filtered1 <- rep2 %>% filter(X %in% sample_index_key2$X)
  rep2_filtered2 <- rep2_filtered1 %>% filter(log10_count > (mean(log10_count) - sd(log10_count) * 2))
  
  unsel <- merge(rep1_filtered2, rep2_filtered2, by = "X", all = T)
  unsel[is.na(unsel)] <- 0
  unsel$rep1_freq <- unsel$count.x / sum(unsel$count.x)
  unsel$rep2_freq <- unsel$count.y / sum(unsel$count.y)
  unsel$u_freq <- 10^((log10(unsel$rep1_freq) + log10(unsel$rep2_freq))/2)
  
  ## Deal with the HygroR data next
  hygro <- merge(read.delim(myfiles[list_of_indexes[3]], sep = "\t"), read.delim(myfiles[list_of_indexes[4]], sep = "\t"), by = "X", all = T)
  ## Combine the data; this should also take care of the filtering, since the unsel was already filtered
  combined_frame <- merge(unsel[,c("X","u_freq")], hygro[,c("X","count.x","count.y")], by = "X", all.x = T)
  combined_frame$sel_freq1 <- combined_frame$count.x / sum(combined_frame$count.x, na.rm = T)
  combined_frame$sel_freq2 <- combined_frame$count.y / sum(combined_frame$count.y, na.rm = T)
  combined_frame$h_freq <- 10^((log10(combined_frame$sel_freq1) + log10(combined_frame$sel_freq2))/2)
  return_frame <- combined_frame[,c("X","u_freq","h_freq")]
  
  ## Do some additional analysis before returning the data frame
  return_frame$h_enrichment <- return_frame$h_freq / return_frame$u_freq
  colnames(return_frame) <- c("sequence","u_freq","h_freq","h_enrichment")
  return_frame2_troubleshooting <- merge(return_frame, sample_index_key2[,c("gene","ortholog","mutant","protease","kozak","plasmid_template","concat","sequence")], by = "sequence", all = T)
  return_frame2 <- merge(return_frame, sample_index_key2[,c("gene","ortholog","mutant","protease","kozak","plasmid_template","concat","sequence")], by = "sequence", all.x = T)
  return(return_frame2)
}
```

```{r Looking at correlations of the uninfected cells with library v1.0}
i0274 <- read.delim(file = "Data/Dbl_barcode_NS4/I0274_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0276 <- read.delim(file = "Data/Dbl_barcode_NS4/I0276_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0278 <- read.delim(file = "Data/Dbl_barcode_NS4/I0278_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0280 <- read.delim(file = "Data/Dbl_barcode_NS4/I0280_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0282 <- read.delim(file = "Data/Dbl_barcode_NS4/I0282_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0284 <- read.delim(file = "Data/Dbl_barcode_NS4/I0284_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0287 <- read.delim(file = "Data/Dbl_barcode_NS4/I0287_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0289 <- read.delim(file = "Data/Dbl_barcode_NS4/I0289_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0291 <- read.delim(file = "Data/Dbl_barcode_NS4/I0291_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0293 <- read.delim(file = "Data/Dbl_barcode_NS4/I0293_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0295 <- read.delim(file = "Data/Dbl_barcode_NS4/I0295_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0297 <- read.delim(file = "Data/Dbl_barcode_NS4/I0297_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0299 <- read.delim(file = "Data/Dbl_barcode_NS4/I0299_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0305 <- read.delim(file = "Data/Dbl_barcode_NS4/I0305_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0308 <- read.delim(file = "Data/Dbl_barcode_NS4/I0308_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)
i0310 <- read.delim(file = "Data/Dbl_barcode_NS4/I0310_lib.tsv", sep = "\t", header = T, stringsAsFactors = F)

unsel_ns4 <- merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(merge(i0274, i0276, by = "X", all = T), i0278, by = "X", all = T), i0280, by = "X", all = T), i0282, by = "X", all = T), i0284, by = "X", all = T), i0287, by = "X", all = T), i0289, by = "X", all = T), i0291, by = "X", all = T), i0293, by = "X", all = T), i0295, by = "X", all = T), i0297, by = "X", all = T), i0299, by = "X", all = T), i0305, by = "X", all = T), i0308, by = "X", all = T), i0310, by = "X", all = T)

colnames(unsel_ns4) <- c("X","u1","u2","u3","u4","u5","u6","u7","u8","u9","u10","u11","u12","u13","u14","u15","u16")
unsel_ns4_filtered <- unsel_ns4 %>% filter(X %in% sample_index_key2$X)
unsel_ns4_filtered_freq <- unsel_ns4_filtered
unsel_ns4_filtered_freq[is.na(unsel_ns4_filtered)] <- 0

for(x in 2:ncol(unsel_ns4_filtered_freq)){
  for(y in 1:nrow(unsel_ns4_filtered_freq)){
    unsel_ns4_filtered_freq[y,x] <- unsel_ns4_filtered_freq[y,x] / sum(unsel_ns4_filtered_freq[,x])
  }
}

unsel_ns4_filtered_freq_melted <- melt(unsel_ns4_filtered_freq, id = "X")
unsel_mean = data.frame("X" = unsel_ns4_filtered_freq$X, "freq" = rowMeans(unsel_ns4_filtered_freq[,2:17]))
unsel_mean_filtered <- unsel_mean %>% filter(freq > 1e-4)
unsel_mean_filtered$count <- unsel_mean_filtered$freq * 1e12
write.table(file = "Data/Dbl_barcode_NS4/zCombined_unsel.tsv", unsel_mean_filtered[,c("X","count")], sep = "\t", quote = F, row.names = F)

### Combining ALL negative control sequencing
ns3_negs_filtered$seq9 <- ns3_negs_filtered$X
ns4_negs_filtered_seq9 <- unsel_ns4_filtered_freq
ns4_negs_filtered_seq9$seq9 <- paste0(substr(ns4_negs_filtered_seq9$X,1,5),substr(ns4_negs_filtered_seq9$X,7,10),"N",substr(ns4_negs_filtered_seq9$X,11,19))

ns34_negs_filtered <- merge(ns3_negs_filtered, ns4_negs_filtered_seq9, by = "seq9", all = T)
ns34_negs_filtered[is.na(ns34_negs_filtered)] <- 0
ns34_negs_filtered <- ns34_negs_filtered[,colnames(ns34_negs_filtered)[!(colnames(ns34_negs_filtered) %in% c("X.x","X.y"))]]

## Do all pairwise correlations of unselected cells
unsel_correlation_vector <- c()
for(x in 2:ncol(ns34_negs_filtered)){
  for(y in 2:ncol(ns34_negs_filtered)){
    unsel_correlation_vector <- c(unsel_correlation_vector, (cor(ns34_negs_filtered[,x],ns34_negs_filtered[,y])))
  }
}

## Do all pairwise correlations of unselected cells for ns3
ns3_unsel_correlation_vector <- c()
for(x in 2:18){
  for(y in 2:18){
    ns3_unsel_correlation_vector <- c(ns3_unsel_correlation_vector, (cor(ns34_negs_filtered[,x],ns34_negs_filtered[,y])))
  }
}

## Do all pairwise correlations of unselected cells for ns4
ns4_unsel_correlation_vector <- c()
for(x in 19:34){
  for(y in 19:34){
    ns4_unsel_correlation_vector <- c(ns4_unsel_correlation_vector, (cor(ns34_negs_filtered[,x],ns34_negs_filtered[,y])))
  }
}

combined_unsel_correlation_df <- rbind(data.frame("grouping" = "All_pairwise", "value" = unsel_correlation_vector),
                                       data.frame("grouping" = "Set_1_pairwise", "value" = ns3_unsel_correlation_vector),
                                       data.frame("grouping" = "Set_2_pairwise", "value" = ns4_unsel_correlation_vector))

combined_unsel_correlation_df$grouping <- factor(combined_unsel_correlation_df$grouping, levels = rev(c("All_pairwise", "Set_1_pairwise", "Set_2_pairwise")))

Comprehensive_neg_cntrl_correlations <- ggplot() + theme(legend.position = "top") +
  labs(x = "Pearson's r^2", y = "Count") +
  scale_x_continuous(limits = c(0.4,1.02)) +
  geom_histogram(data = combined_unsel_correlation_df, aes(x = value^2, fill = grouping), color = "black", binwidth = 0.02, alpha = 0.8)
Comprehensive_neg_cntrl_correlations
ggsave(file = "Plots/Comprehensive_neg_cntrl_correlations.pdf", Comprehensive_neg_cntrl_correlations, height = 1.75, width = 2)
```

```{r Nextseq 4}
myfiles = list.files(path="Data/Dbl_barcode_NS4", pattern="*.tsv", full.names=TRUE)
myfiles_df <- data.frame("number" = seq(1,length(myfiles)), "index" = myfiles)

ns4_g928a <- makeExperimentFrame2(c(1,1,2,2)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns4_g928b <- makeExperimentFrame2(c(9,9,10,10)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns4_g928c <- makeExperimentFrame2(c(16,16,17,17)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns4_g928d <- makeExperimentFrame2(c(26,26,27,27)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
```


g928 <- merge(merge(merge(merge(g928a[,c("plasmid_template","h_enrichment")],
                          g928b[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
                    g928c[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
              g928d[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
              g928a[,5:10], by = "plasmid_template", all = T)
g928$geomean <- 10^(rowMeans(log10(g928[,2:5]), na.rm = T))

s2g740d

s2d614g_kit_comparison <- merge(s2g740d[,c("plasmid_template","geomean")], g928[,c("plasmid_template","geomean")], by = "plasmid_template")


s2d614g <- merge(merge(merge(merge(merge(merge(g928a[,c("plasmid_template","h_enrichment")],
                          g928b[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
                    g928c[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
              g928d[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
              ns3_g740d1[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
              ns3_g740d2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
              g928a[,5:10], by = "plasmid_template", all = T)
s2d614g$geomean <- 10^(rowMeans(log10(s2d614g[,2:7]), na.rm = T))

ns4_combined <- s2d614g[,c("plasmid_template","geomean")]
colnames(ns4_combined) <- c("plasmid_template","D614G")
ns4_combined2 <- merge(ns4_combined, g928a[,5:10], by = "plasmid_template", all = T)


```{r Nextseq 7}
myfiles = list.files(path="Data/Dbl_barcode_NS7", pattern="*.tsv", full.names=TRUE)
myfiles_df <- data.frame("number" = seq(1,length(myfiles)), "index" = myfiles)

ns7_g928a1 <- makeExperimentFrame2(c(1,1,2,2)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_g928a2 <- makeExperimentFrame2(c(8,8,9,9)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_g928a3 <- makeExperimentFrame2(c(14,14,15,15)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_alpha1 <- makeExperimentFrame2(c(1,1,3,3)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_alpha2 <- makeExperimentFrame2(c(8,8,10,10)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_alpha3 <- makeExperimentFrame2(c(14,14,16,16)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_delta1 <- makeExperimentFrame2(c(1,1,4,4)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_delta2 <- makeExperimentFrame2(c(8,8,11,11)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_delta3 <- makeExperimentFrame2(c(14,14,17,17)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_ba1omicron1 <- makeExperimentFrame2(c(1,1,5,5)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_ba1omicron2 <- makeExperimentFrame2(c(8,8,12,12)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_ba4omicron1 <- makeExperimentFrame2(c(1,1,6,6)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_bq1omicron1 <- makeExperimentFrame2(c(1,1,7,7)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
ns7_bq1omicron2 <- makeExperimentFrame2(c(8,8,13,13)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
```

```{r}

d614g_enrichment_example <- merge(merge(merge(merge(merge(merge(merge(merge(merge(ns7_g928a1[,c("plasmid_template","u_freq")], ns7_g928a2[,c("plasmid_template","u_freq")], by = "plasmid_template"), ns7_g928a3[,c("plasmid_template","u_freq")], by = "plasmid_template"), ns4_g928a[,c("plasmid_template","u_freq")], by = "plasmid_template"), ns4_g928b[,c("plasmid_template","u_freq")], by = "plasmid_template"), ns4_g928b[,c("plasmid_template","h_freq")], by = "plasmid_template"), ns4_g928a[,c("plasmid_template","h_freq")], by = "plasmid_template"), ns7_g928a1[,c("plasmid_template","h_freq")], by = "plasmid_template"), ns7_g928a2[,c("plasmid_template","h_freq")], by = "plasmid_template"), ns7_g928a3[,c("plasmid_template","h_freq","ortholog","protease","mutant","kozak")], by = "plasmid_template")

colnames(d614g_enrichment_example)[2:11] <- c("u1","u2","u3","u4","u5","h1","h2","h3","h4","h5")
d614g_enrichment_example$u_freq <- 10^rowMeans(log10(d614g_enrichment_example[,seq(2,6)]))
d614g_enrichment_example$h_freq <- 10^rowMeans(log10(d614g_enrichment_example[,seq(7,10)]))

d614g_enrichment_example2 <- d614g_enrichment_example %>% filter(ortholog %in% c("H.sapiens","Control") & protease == "none" | ortholog %in% c("R.pearsonii", "R.alcyone"))
d614g_enrichment_example2$identifier <- paste0(d614g_enrichment_example2$ortholog," ",d614g_enrichment_example2$mutant," ",d614g_enrichment_example2$kozak)
d614g_enrichment_example3 <- d614g_enrichment_example2 %>% filter(identifier %in% c("H.sapiens WT high","Control dEcto high","H.sapiens WT low", "R.pearsonii WT high", "R.alcyone WT high"))

example_of_enrichment_scatterplot <- ggplot() + 
  labs(x = "Frequency of plasmid before selection", y = "Frequency of plasmid\nafter selection") +
  scale_x_log10() + scale_y_log10() +
  geom_abline(slope = 1, linetype = 2, alpha = 0.3) +
  geom_segment(data = d614g_enrichment_example3, aes(x = u_freq, xend = u_freq, y = u_freq, yend = h_freq), linetype = 1, alpha = 0.3) + 
  geom_point(data = d614g_enrichment_example2, aes(x = u_freq, y = h_freq), alpha = 0.2) +
  geom_point(data = d614g_enrichment_example3, aes(x = u_freq, y = h_freq), alpha = 0.5) +
  geom_text_repel(data = d614g_enrichment_example3, aes(x = u_freq, y = h_freq, label = identifier), segment.color = "orange", min.segment.length = 0, color = "red", size = 2)
example_of_enrichment_scatterplot
ggsave(file = "Plots/example_of_enrichment_scatterplot.pdf", example_of_enrichment_scatterplot, height = 1.75, width = 3)
```

```{r Merging replicates for D614G}
s2d614g <- merge(merge(merge(merge(merge(merge(merge(merge(ns3_g740d1[,c("plasmid_template","h_enrichment")],
               ns3_g740d2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns4_g928a[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns4_g928b[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns4_g928c[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns4_g928d[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_g928a1[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_g928a2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_g928a3[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T)
for(x in 2:ncol(s2d614g)){s2d614g[is.na(s2d614g[,x]),x] <- min(s2d614g[,x], na.rm = T)}

#s2d614g[is.na(s2d614g)] <- mean(unlist(s2d614g[s2d614g$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR",2:ncol(s2d614g)]))
 #min(s2d614g[,2:ncol(s2d614g)], na.rm = T)
s2d614g[,2:ncol(s2d614g)] <- log10(s2d614g[,2:ncol(s2d614g)])
s2d614g$mean_log10 <- rowMeans(s2d614g[,2:ncol(s2d614g)], na.rm = T)
s2d614g$sd_log10 <- apply(s2d614g[,2:(ncol(s2d614g)-1)], 1, sd, na.rm=TRUE)
s2d614g$geomean <- 10^s2d614g$mean_log10
```

```{r Merging repliactes for Alpha}
s2alpha <- merge(merge(merge(merge(ns3_alpha1[,c("plasmid_template","h_enrichment")],
               ns3_alpha2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_alpha1[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_alpha2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_alpha3[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T)
for(x in 2:ncol(s2alpha)){s2alpha[is.na(s2alpha[,x]),x] <- min(s2alpha[,x], na.rm = T)}
#s2alpha[is.na(s2alpha)] <- mean(unlist(s2alpha[s2alpha$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR",2:ncol(s2alpha)]))  #min(s2alpha[,2:ncol(s2alpha)], na.rm = T)
s2alpha[,2:ncol(s2alpha)] <- log10(s2alpha[,2:ncol(s2alpha)])
s2alpha$mean_log10 <- rowMeans(s2alpha[,2:ncol(s2alpha)], na.rm = T)
s2alpha$sd_log10 <- apply(s2alpha[,2:(ncol(s2alpha)-1)], 1, sd, na.rm=TRUE)
s2alpha$geomean <- 10^s2alpha$mean_log10
```

```{r}
s2beta <- merge(ns3_beta1[,c("plasmid_template","h_enrichment")],
               ns3_beta2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T)
for(x in 2:ncol(s2beta)){s2beta[is.na(s2beta[,x]),x] <- min(s2beta[,x], na.rm = T)}
#s2beta[is.na(s2beta)] <- mean(unlist(s2beta[s2beta$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR",2:ncol(s2beta)]))  #min(s2beta[,2:ncol(s2beta)], na.rm = T)
s2beta[,2:ncol(s2beta)] <- log10(s2beta[,2:ncol(s2beta)])
s2beta$mean_log10 <- rowMeans(s2beta[,2:ncol(s2beta)], na.rm = T)
s2beta$sd_log10 <- apply(s2beta[,2:(ncol(s2beta)-1)], 1, sd, na.rm=TRUE)
s2beta$geomean <- 10^s2beta$mean_log10

s2gamma <- merge(ns3_gamma1[,c("plasmid_template","h_enrichment")],
               ns3_gamma2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T)
for(x in 2:ncol(s2gamma)){s2gamma[is.na(s2gamma[,x]),x] <- min(s2gamma[,x], na.rm = T)}
#s2gamma[is.na(s2gamma)] <- mean(unlist(s2gamma[s2gamma$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR",2:ncol(s2gamma)])) #min(s2gamma[,2:ncol(s2gamma)], na.rm = T)
s2gamma[,2:ncol(s2gamma)] <- log10(s2gamma[,2:ncol(s2gamma)])
s2gamma$mean_log10 <- rowMeans(s2gamma[,2:ncol(s2gamma)], na.rm = T)
s2gamma$sd_log10 <- apply(s2gamma[,2:(ncol(s2gamma)-1)], 1, sd, na.rm=TRUE)
s2gamma$geomean <- 10^s2gamma$mean_log10
```

```{r}
s2delta <- merge(merge(merge(merge(ns3_delta1[,c("plasmid_template","h_enrichment")],
               ns3_delta2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_delta1[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_delta2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_delta3[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T)
for(x in 2:ncol(s2delta)){s2delta[is.na(s2delta[,x]),x] <- min(s2delta[,x], na.rm = T)}
#s2delta[is.na(s2delta)] <- mean(unlist(s2delta[s2delta$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR",2:ncol(s2delta)])) #min(s2delta[,2:ncol(s2delta)], na.rm = T)
s2delta[,2:ncol(s2delta)] <- log10(s2delta[,2:ncol(s2delta)])
s2delta$mean_log10 <- rowMeans(s2delta[,2:ncol(s2delta)], na.rm = T)
s2delta$sd_log10 <- apply(s2delta[,2:(ncol(s2delta)-1)], 1, sd, na.rm=TRUE)
s2delta$geomean <- 10^s2delta$mean_log10
```

```{r}
s2ba1omicron <- merge(merge(merge(ns3_ba1omicron1[,c("plasmid_template","h_enrichment")],
               ns3_ba1omicron2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_ba1omicron1[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T),
               ns7_ba1omicron2[,c("plasmid_template","h_enrichment")], by = "plasmid_template", all = T)
for(x in 2:ncol(s2ba1omicron)){s2ba1omicron[is.na(s2ba1omicron[,x]),x] <- min(s2ba1omicron[,x], na.rm = T)}
#s2ba1omicron[is.na(s2ba1omicron)] <- mean(unlist(s2ba1omicron[s2ba1omicron$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR",2:ncol(s2ba1omicron)])) #min(s2ba1omicron[,2:ncol(s2ba1omicron)], na.rm = T)
s2ba1omicron[,2:ncol(s2ba1omicron)] <- log10(s2ba1omicron[,2:ncol(s2ba1omicron)])
s2ba1omicron$mean_log10 <- rowMeans(s2ba1omicron[,2:ncol(s2ba1omicron)], na.rm = T)
s2ba1omicron$sd_log10 <- apply(s2ba1omicron[,2:(ncol(s2ba1omicron)-1)], 1, sd, na.rm=TRUE)
s2ba1omicron$geomean <- 10^s2ba1omicron$mean_log10
```

```{r Combining all data}

combined <- merge(merge(merge(merge(merge(s2d614g[,c("plasmid_template","geomean","sd_log10")],
      s2alpha[,c("plasmid_template","geomean","sd_log10")], by = "plasmid_template", all = T),
      s2beta[,c("plasmid_template","geomean","sd_log10")], by = "plasmid_template", all = T),
      s2gamma[,c("plasmid_template","geomean","sd_log10")], by = "plasmid_template", all = T),
      s2delta[,c("plasmid_template","geomean","sd_log10")], by = "plasmid_template", all = T),
      s2ba1omicron[,c("plasmid_template","geomean","sd_log10")], by = "plasmid_template", all = T)

colnames(combined) <- c("plasmid_template","D614G","sd_d614g","Alpha","sd_alpha","Beta","sd_beta","Gamma","sd_gamma","Delta","sd_delta","BA1","sd_ba1")

write.csv(file = "Output_tables/Supplementary_table_1.csv", combined, row.names = F)

## Remove the E329K variant from the analysis because it was in the plasmid library at abnormally low levels
combined2 <- merge(combined, ns3_alpha1[,5:10], by = "plasmid_template", all = T) %>% filter(plasmid_template != "G879B_AttB_[koz-mut]ACE2[E329K]-IRES-mCherry-H2A-P2A-PuroR")

combined2$label <- ""
for(x in 1:nrow(combined2)){
  if(combined2$ortholog[x] == "H.sapiens" & combined2$kozak[x] == "high"){combined2$label[x] <- combined2$ortholog[x]}
  if(combined2$ortholog[x] == "H.sapiens" & combined2$kozak[x] == "low"){combined2$label[x] <- combined2$mutant[x]} else{combined2$label[x] <- combined2$ortholog[x]}}

combined3 <- combined2 #%>% filter(!(kozak == "high" & mutant == "D355N"))




```

## Make an overall graph that shows the extent of the infection data we are considering here
```{r Showing how ACE2 expression level affects infection}
kozak_protease <- combined2 %>% filter(ortholog == "H.sapiens" | ortholog == "H.sapiens(rep1)" | ortholog == "H.sapiens(rep2)") %>%  filter(mutant == "WT")
for(x in 1:nrow(kozak_protease)){
  kozak_protease$identifier[x] <- strsplit(kozak_protease$plasmid_template[x],"_")[[1]][1]
}

## Compare high and low Kozak
lowhigh_reps <- kozak_protease %>% filter(protease == "none")
lowhigh_reps_t <- data.frame(t(lowhigh_reps[,c("identifier","D614G","Alpha","Beta","Gamma","Delta","BA1")]))
colnames(lowhigh_reps_t) <- lowhigh_reps_t[1,]
lowhigh_reps_t <- lowhigh_reps_t[-1,]
lowhigh_reps_t$G755A <- as.numeric(lowhigh_reps_t$G755A)
lowhigh_reps_t$G752A <- as.numeric(lowhigh_reps_t$G752A)

lowhigh_reps_t$high_low_infection_ratio <- lowhigh_reps_t$G752A / lowhigh_reps_t$G755A
lowhigh_reps_t$label <- rownames(lowhigh_reps_t)

ace2_high_low_plot <- ggplot() + 
  scale_y_log10(limits = c(0.3, 20)) + 
  labs(y = "Fold increase\nto infection", x = NULL) +
  theme(axis.text.x = element_blank(), panel.grid.major.x = element_blank(), axis.ticks.x = element_blank()) + 
  geom_point(data = lowhigh_reps_t, aes(x = 0, y = high_low_infection_ratio, color = label), alpha = 0.4) +
  geom_point(data = lowhigh_reps_t, aes(x = 0, y = 10^mean(log10(high_low_infection_ratio))), alpha = 0.4, shape = 95, color = "red", size = 8)
ace2_high_low_plot
ggsave(file = "Plots/ace2_high_low_plot.pdf", ace2_high_low_plot, height = 0.9, width = 2)

```


```{r Seeing how adding TMPRSS2 alters infection}
## Now let's look at the effect of TMPRSS2 on the high ACE2 expressor cells
high_protease <- kozak_protease %>% filter(identifier %in% c("G828A","G752A"))
high_protease_t <- data.frame(t(high_protease[,c("identifier","D614G","Alpha","Beta","Gamma","Delta","BA1")]))
colnames(high_protease_t) <- high_protease_t[1,]
high_protease_t <- high_protease_t[-1,]
high_protease_t$G828A <- as.numeric(high_protease_t$G828A)
high_protease_t$G752A <- as.numeric(high_protease_t$G752A)
high_protease_t$tmprss2_infection_ratio <- high_protease_t$G828A / high_protease_t$G752A
high_protease_t$label <- rownames(high_protease_t)

ace2_tmprss2_plot <- ggplot() + 
  scale_y_log10(limits = c(0.3, 20)) + 
  labs(y = "Fold increase\nto infection", x = NULL) +
  theme(axis.text.x = element_blank(), panel.grid.major.x = element_blank(), axis.ticks.x = element_blank()) + 
  geom_point(data = high_protease_t, aes(x = 0, y = tmprss2_infection_ratio, color = label), alpha = 0.4) +
  geom_point(data = high_protease_t, aes(x = 0, y = 10^mean(log10(tmprss2_infection_ratio))), alpha = 0.4, shape = 95, color = "red", size = 8)
ace2_tmprss2_plot
ggsave(file = "Plots/ace2_tmprss2_plot.pdf", ace2_tmprss2_plot, height = 0.9, width = 2)
```

```{r}
tmprss2 <- combined2 %>% filter(protease == "TMPRSS2" & mutant == "WT" | kozak == "high" & mutant == "dEcto" | protease == "none" & kozak == "high")

## Values left unscaled
ortholog1 <- tmprss2[colnames(tmprss2)[(colnames(tmprss2) %in% c("ortholog","D614G","Alpha","Beta","Gamma","Delta","BA1"))]] 
ortholog1_melt <- melt(ortholog1, id = "ortholog")

factor_levels_for_large_heatmap <- c("Control", "H.sapiens", "H.sapiens(rep1)", "H.sapiens(rep2)", "M.musculus", "S.scrofa", "M.javanica", "R.landeri", "R.alcyone", "R.ferrumequinum", "R.shameli", "R.affinis",  "R.sinicus_215", "R.sinicus", "R.sinicus_200", "R.sinicus_472", "R.pearsonii")

ortholog1_melt$ortholog <- factor(ortholog1_melt$ortholog, levels = factor_levels_for_large_heatmap)
ortholog1_melt$variable <- factor(ortholog1_melt$variable, levels = c("D614G","Alpha","Beta","Gamma","Delta","BA1"))

orthologs_unscaled <- ggplot() + labs(x = NULL, y = NULL) + 
  scale_fill_gradient(low = "white", high = "red") + 
  theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  geom_tile(data = ortholog1_melt, aes(x = ortholog, y = variable, fill = value))
orthologs_unscaled
ggsave(file = "Plots/orthologs_unscaled.pdf", orthologs_unscaled, height = 3, width = 5)

## Scaled to dEcto and Human High Kozak with TMPRSS2
tmprss2b <- tmprss2
for(x in colnames(tmprss2)[(colnames(tmprss2) %in% c("D614G","Alpha","Beta","Gamma","Delta","BA1"))]){
  temp_high <-  mean(c(tmprss2b[ tmprss2b$plasmid_template == "G828A_AttB_ACE2-2A-TMPRSS2_IRES_mCherry-H2A-P2A-PuroR" ,x],tmprss2b[ tmprss2b$plasmid_template == "G852C_AttB_ACE2-2A-TMPRSS2_IRES_miRFP670-H2A-P2A-PuroR" ,x]))#max(tmprss2b[,x], na.rm = T)
  temp_low <-  tmprss2b[ tmprss2b$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR" ,x]
  tmprss2b[,x] <- (tmprss2b[,x] - temp_low) / (temp_high - temp_low)
}
ortholog2 <- tmprss2b[colnames(tmprss2b)[(colnames(tmprss2b) %in% c("ortholog","D614G","Alpha","Beta","Gamma","Delta","BA1"))]]
ortholog2_melt <- melt(ortholog2, id = "ortholog")


ortholog2_melt$ortholog <- factor(ortholog2_melt$ortholog, levels = factor_levels_for_large_heatmap)
ortholog2_melt$variable <- factor(ortholog2_melt$variable, levels = c("D614G","Alpha","Beta","Gamma","Delta","BA1"))

orthologs2_s2variants <- ggplot() + 
  labs(x = NULL, y = NULL) + 
  scale_fill_gradient(low = "white", high = "red") + 
  theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  geom_tile(data = ortholog2_melt, aes(x = ortholog, y = variable, fill = value)); orthologs2_s2variants
ggsave(file = "Plots/orthologs2_s2variants.pdf", orthologs2_s2variants, height = 3, width = 5)
```

```{r Variants of human ACE2 with low Kozak}
low_kozak <- combined2 %>% filter(kozak == "low" | mutant == "dEcto")

## Normal unscaled
low_kozak1 <- low_kozak

low_kozak1 <- low_kozak[colnames(low_kozak)[(colnames(low_kozak) %in% c("mutant","D614G","Alpha","Beta","Gamma","Delta","BA1"))]]
low_kozak1_melt <- melt(low_kozak1, id = "mutant")
#low_kozak1_melt$variable <- factor(low_kozak1_melt$variable, levels = c(""))
low_kozak1_melt$mutant <- factor(low_kozak1_melt$mutant, levels = c("WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "E329K", "G352V", "K353D", "D355N"))

low_kozak1_ace2mutants <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  scale_fill_gradient(low = "white", high = "red") + 
  geom_tile(data = low_kozak1_melt, aes(x = mutant, y = variable, fill = value))
low_kozak1_ace2mutants
ggsave(file = "Plots/low_kozak1_ace2mutants.pdf", low_kozak1_ace2mutants, height = 1.25, width = 4)


## Scaled to dEcto and Human Low Kozak
low_kozak2 <- low_kozak
for(x in c(colnames(low_kozak)[(colnames(low_kozak) %in% c("D614G","Alpha","Beta","Gamma","Delta","BA1"))])){
  temp_high <-  low_kozak2[ low_kozak2$plasmid_template == "G755A_AttB_[koz-mut]ACE2-IRES-mCherry-H2A-P2A-PuroR" ,x]
  temp_low <-  low_kozak2[ low_kozak2$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR" ,x]
  low_kozak2[,x] <- (low_kozak2[,x] - temp_low) / (temp_high - temp_low)
}

low_kozak2 <- low_kozak2[,c(colnames(low_kozak)[(colnames(low_kozak) %in% c("mutant","D614G","Alpha","Beta","Gamma","Delta","BA1"))])]
low_kozak2_melt <- melt(low_kozak2, id = "mutant")
low_kozak2_melt$variable <- factor(low_kozak2_melt$variable, levels = c("D614G","Alpha","Beta","Gamma","Delta","BA1"))
low_kozak2_melt$mutant <- factor(low_kozak2_melt$mutant, levels = c("WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "E329K", "G352V", "K353D", "D355N"))

low_kozak2_ace2mutants <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  scale_fill_gradient(low = "white", high = "red", limits = c()) + 
  geom_tile(data = low_kozak2_melt, aes(x = mutant, y = variable, fill = value))
low_kozak2_ace2mutants
ggsave(file = "Plots/low_kozak2_ace2mutants.pdf", low_kozak2_ace2mutants, height = 1.1, width = 3.1)


low_kozak2_melt$mutant <- factor(low_kozak2_melt$mutant, levels = rev(c("WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "E329K", "G352V", "K353D", "D355N")))

low_kozak2_ace2mutants_s2variants_flip <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45), legend.position = "top") + 
  scale_fill_gradient(low = "white", high = "red") + 
  geom_tile(data = low_kozak2_melt, aes(x = variable, y = mutant, fill = value))
low_kozak2_ace2mutants_s2variants_flip
ggsave(file = "Plots/low_kozak2_ace2mutants_s2variants_flip.pdf", low_kozak2_ace2mutants_s2variants_flip, height = 3, width = 2.5)

```

```{r Making a combined heatmap with both orthologs and variants}
not_ortholog_not_lowkozak <- combined2 %>% filter(kozak == "high" & protease == "none")

combined_ortholog_variant <- rbind((ortholog1_melt %>% mutate(sample = ortholog))[,c("sample","variable","value")],
                                   (low_kozak1_melt %>% mutate(sample = mutant))[,c("sample","variable","value")]) %>% filter(variable %in% c("D614G","Alpha","Beta","Gamma","Delta","BA1"))

combined_ortholog_variant$variable <- factor(combined_ortholog_variant$variable, levels = c("D614G","Alpha","Beta","Gamma","Delta","BA1"))

combined_ortholog_variant$sample <- factor(combined_ortholog_variant$sample, levels = rev(c("Control", "H.sapiens", "H.sapiens(rep1)", "H.sapiens(rep2)", "M.musculus", "S.scrofa", "M.javanica", "R.landeri", "R.alcyone", "R.ferrumequinum", "R.shameli", "R.affinis",  "R.sinicus_215", "R.sinicus", "R.sinicus_200", "R.sinicus_472", "R.pearsonii", "WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "E329K", "G352V", "K353D", "D355N")))

combined_ortholog_variant_plot <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45), legend.position = "top") + 
  scale_fill_gradient(low = "white", high = "red", na.value = "white") +  #limits=c(-0.25,0.75), 
  geom_tile(data = combined_ortholog_variant, aes(x = variable, y = sample, fill = value))
combined_ortholog_variant_plot
ggsave(file = "Plots/combined_ortholog_variant_plot.pdf", combined_ortholog_variant_plot, height = 4, width = 2)
```

```{r Comparing to 2021 PlosP data}
human_ace2_variants <- read.delim(file = "Data/Other_papers/2021_Shukla_PlosPathog/journal.ppat.1009715.s008.tsv", sep = "\t")
human_ace2_variants2 <- human_ace2_variants %>% group_by(cell_label) %>% summarize(mean_infection = mean(scaled_infection)) #10^mean(log_scaled_infection))

human_ace2_variants2$mutant <- NA
for(x in 3:nrow(human_ace2_variants2)){
  human_ace2_variants2$mutant[x] <- strsplit(human_ace2_variants2$cell_label[x],"-")[[1]][2]
}
human_ace2_variants2$mutant[1] <- "dEcto"
human_ace2_variants2$mutant[2] <- "WT"

#dbl_combined3_temp <- dbl_combined3 %>% filter(kozak == "low" | mutant == "dEcto")
low_kozak1_s2variants <- low_kozak1_melt %>% filter(variable == "D614G")
#low_kozak1_s2variants[low_kozak1_s2variants$mutant == "dEcto","mutant"] <- "Neg cntrl"

human_ace2_variants3 <- merge(human_ace2_variants2,low_kozak1_s2variants[,c("mutant","value")])
human_ace2_variants3$value <- 10^human_ace2_variants3$value / 10^human_ace2_variants3[human_ace2_variants3$mutant == "WT","value"]

Variant_validation_graph <- ggplot() + labs(x = "Flow cytometry infectivity\nmeasurement from published study", y = "Sequencing-based\ninfectivity measurement\nfrom current study") +
  scale_x_log10(breaks = c(0.1,0.5,1,3)) + #scale_x_log10(limits = c(0,1.6), expand = c(0,0), breaks = seq(0,1.5,0.5)) + 
  scale_y_log10(breaks = c(0.1,0.5,1,3)) + #scale_y_log10(limits = c(0,1.7), expand = c(0,0), breaks = seq(0,1.5,0.5)) +
  #geom_abline(slope = 1, linetype = 2, alpha = 0.3) +
  geom_point(data = human_ace2_variants3, aes(x = mean_infection, y = value), alpha = 0.5) +
  geom_text_repel(data = human_ace2_variants3, aes(x = mean_infection, y = value, label = mutant), color = "red", alpha = 0.8, size = 1.5, segment.color = "orange", segment.alpha = 0.5)
Variant_validation_graph
ggsave(file = "Plots/Variant_validation_graph.pdf", Variant_validation_graph, height = 1.75, width = 2)
```

```{r Comparing to 2022 PlosP Biol}
ortholog_2color_data <- read.delim(file = "Data/Other_papers/2022_Roelle_PlosBiol/Fig5.csv", sep = ",")
ortholog_2color_data_flow <- ortholog_2color_data %>% filter(type == "flow_cytometry" & virus_label == "SARS-CoV-2 RBD") %>% mutate(ortholog = cell_label)
ortholog_2color_data_micro <- ortholog_2color_data %>%  filter(type == "microscopy" & cell_label == "SARS-CoV-2 RBD") %>% mutate(ortholog = virus_label)

ortholog_2color_data_flow$norm_geomean <- ortholog_2color_data_flow$geomean / ortholog_2color_data_flow[ortholog_2color_data_flow$ortholog == "H.sapiens","geomean"]
ortholog_2color_data_micro$norm_geomean <- ortholog_2color_data_micro$geomean / ortholog_2color_data_micro[ortholog_2color_data_micro$ortholog == "H.sapiens","geomean"]

norm_ortholog_2color_data <- merge(ortholog_2color_data_flow[,c("ortholog","norm_geomean")], ortholog_2color_data_micro[,c("ortholog","norm_geomean")], by = "ortholog")
norm_ortholog_2color_data$norm_geomean <- rowMeans(norm_ortholog_2color_data[,c("norm_geomean.x","norm_geomean.y")])

norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "R.sinicus200","ortholog"] <- "R.sinicus_200"
norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "R.sinicus215","ortholog"] <- "R.sinicus_215"
norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "R.sinicus472","ortholog"] <- "R.sinicus_472"
norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "NULL (fs)","ortholog"] <- "Neg cntrl"

ortholog1[ortholog1$ortholog == "Control","ortholog"] <- "dEcto"


ortholog1_melt_s2_plosbiol <- merge(ortholog1[,c("ortholog","D614G")], norm_ortholog_2color_data[,c("ortholog","norm_geomean")], by = "ortholog", all = T)
ortholog1_melt_s2_plosbiol$D614G <- ortholog1_melt_s2_plosbiol$D614G / ortholog1_melt_s2_plosbiol[ortholog1_melt_s2_plosbiol$ortholog == "H.sapiens","D614G"]

Variant_validation_graph2 <- ggplot() + labs(x = "Flow cytometry infectivity\nmeasurement from published study", y = "Sequencing-based\ninfectivity measurement\nfrom current study") +
  scale_x_log10(breaks = c(0.1,0.3,1,3)) + #limits = c(0.1,1.05), expand = c(0,0), breaks = c(0.1,0.3,1)) + 
  scale_y_log10(breaks = c(0.1,0.3,1,3)) + #limits = c(0.05,1.8), expand = c(0,0), breaks = c(0.1,0.3,1)) +
  #geom_abline(slope = 1, linetype = 2, alpha = 0.3) +
  geom_point(data = ortholog1_melt_s2_plosbiol, aes(x = norm_geomean, y = D614G), alpha = 0.5) +
  geom_text_repel(data = ortholog1_melt_s2_plosbiol, aes(x = norm_geomean, y = D614G, label = ortholog), color = "red", alpha = 0.8, size = 1.5, segment.color = "orange", segment.alpha = 0.5)
Variant_validation_graph2
ggsave(file = "Plots/Variant_validation_graph2.pdf", Variant_validation_graph2, height = 1.75, width = 2)
```

```{r Comparing correlations with human ACE2 variants}
ggplot() + 
  geom_point(data = low_kozak, aes(x = D614G, y = Alpha)) +
  geom_text_repel(data = low_kozak, aes(x = D614G, y = Alpha, label = mutant), color = "red")

ggplot() + 
  geom_point(data = low_kozak, aes(x = D614G, y = Beta)) +
  geom_text_repel(data = low_kozak, aes(x = D614G, y = Beta, label = mutant), color = "red")

ggplot() + 
  geom_point(data = low_kozak, aes(x = D614G, y = Gamma)) +
  geom_text_repel(data = low_kozak, aes(x = D614G, y = Gamma, label = mutant), color = "red")

ggplot() + 
  geom_point(data = low_kozak, aes(x = D614G, y = Delta)) +
  geom_text_repel(data = low_kozak, aes(x = D614G, y = Delta, label = mutant), color = "red")

D614G_Omicron_scatterplot <- ggplot() + 
  geom_point(data = low_kozak, aes(x = D614G, y = BA1), alpha = 0.5) +
  geom_text_repel(data = low_kozak, aes(x = D614G, y = BA1, label = mutant), color = "red", size = 2, segment.color = "orange", segment.alpha = 0.5)
D614G_Omicron_scatterplot
ggsave(file = "Plots/D614G_Omicron_scatterplot.pdf", D614G_Omicron_scatterplot, height = 1.5, width = 1.7)

```

## The below section looks at all of the SARS-CoV-2 variant RBD co-structures with human ACE2

```{r D614G RBD-ACE2 contact maps}
pdb_7sxy_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7sxy_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7sxy_dist_frame$position <- as.numeric(rownames(pdb_7sxy_dist_frame))
pdb_7sxy_dist_frame_melt <- melt(pdb_7sxy_dist_frame, id = "position")
pdb_7sxy_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7sxy_dist_frame_melt$variable,2,5)) + 329
pdb_7sxy_dist_frame_melt$ace2_position <- as.numeric(pdb_7sxy_dist_frame_melt$position) + 18
pdb_7sxy_dist_ace2_positionlist <- as.numeric(unique(pdb_7sxy_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7sxy_dist_rbd_positionlist <- as.numeric(unique(pdb_7sxy_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7sxy_dist_frame_melt$ace2_rbd <- paste(pdb_7sxy_dist_frame_melt$ace2_pos, pdb_7sxy_dist_frame_melt$rbd_pos, sep = "_")

pdb_6lzg_dist_frame <- read.csv(file = "Data/PDB_contact_maps/6lzg_ace2row_rbdcol_dist_matrix_min.csv")
pdb_6lzg_dist_frame$position <- as.numeric(rownames(pdb_6lzg_dist_frame))
pdb_6lzg_dist_frame_melt <- melt(pdb_6lzg_dist_frame, id = "position")
pdb_6lzg_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_6lzg_dist_frame_melt$variable,2,5)) + 332
pdb_6lzg_dist_frame_melt$ace2_position <- as.numeric(pdb_6lzg_dist_frame_melt$position) + 18
pdb_6lzg_dist_ace2_positionlist <- as.numeric(unique(pdb_6lzg_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_6lzg_dist_rbd_positionlist <- as.numeric(unique(pdb_6lzg_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_6lzg_dist_frame_melt$ace2_rbd <- paste(pdb_6lzg_dist_frame_melt$ace2_pos, pdb_6lzg_dist_frame_melt$rbd_pos, sep = "_")

pdb_6m0j_dist_frame <- read.csv(file = "Data/PDB_contact_maps/6m0j_ace2row_rbdcol_dist_matrix_min.csv")
pdb_6m0j_dist_frame$position <- as.numeric(rownames(pdb_6m0j_dist_frame))
pdb_6m0j_dist_frame_melt <- melt(pdb_6m0j_dist_frame, id = "position")
pdb_6m0j_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_6m0j_dist_frame_melt$variable,2,5)) + 332
pdb_6m0j_dist_frame_melt$ace2_position <- as.numeric(pdb_6m0j_dist_frame_melt$position) + 18
pdb_6m0j_dist_ace2_positionlist <- as.numeric(unique(pdb_6m0j_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_6m0j_dist_rbd_positionlist <- as.numeric(unique(pdb_6m0j_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_6m0j_dist_frame_melt$ace2_rbd <- paste(pdb_6m0j_dist_frame_melt$ace2_pos, pdb_6m0j_dist_frame_melt$rbd_pos, sep = "_")
pdb_6m0j_dist_frame_melt$dist_6m0j <- pdb_6m0j_dist_frame_melt$value

pdb_6m17_dist_frame <- read.csv(file = "Data/PDB_contact_maps/6m17_ace2row_rbdcol_dist_matrix_min.csv")
pdb_6m17_dist_frame$position <- as.numeric(rownames(pdb_6m17_dist_frame))
pdb_6m17_dist_frame_melt <- melt(pdb_6m17_dist_frame, id = "position")
pdb_6m17_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_6m17_dist_frame_melt$variable,2,5)) + 335
pdb_6m17_dist_frame_melt$ace2_position <- as.numeric(pdb_6m17_dist_frame_melt$position) + 20
pdb_6m17_dist_ace2_positionlist <- as.numeric(unique(pdb_6m17_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_6m17_dist_rbd_positionlist <- as.numeric(unique(pdb_6m17_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_6m17_dist_frame_melt$ace2_rbd <- paste(pdb_6m17_dist_frame_melt$ace2_pos, pdb_6m17_dist_frame_melt$rbd_pos, sep = "_")
pdb_6m17_dist_frame_melt$dist_6m17 <- pdb_6m17_dist_frame_melt$value

## D614G
pdb_d614g_combined <- merge(merge(merge(pdb_7sxy_dist_frame_melt[,c("ace2_rbd","ace2_position","rbd_position","value")],pdb_6lzg_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"),pdb_6m0j_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_6m17_dist_frame_melt[,c("ace2_rbd","dist_6m17")], by = "ace2_rbd") %>% filter(!is.na(rbd_position) & !is.na(ace2_position))
colnames(pdb_d614g_combined) <- c("ace2_rbd","ace2_position","rbd_position","dist_7sxy", "dist_6lzg", "dist_6m0j", "dist_6m17")
pdb_d614g_combined[pdb_d614g_combined$dist_7sxy > 50,"dist_7sxy"] <- NA
pdb_d614g_combined[pdb_d614g_combined$dist_6lzg > 50,"dist_6lzg"] <- NA
pdb_d614g_combined[pdb_d614g_combined$dist_6m0j > 50,"dist_6m0j"] <- NA
pdb_d614g_combined[pdb_d614g_combined$dist_6m17 > 50,"dist_6m17"] <- NA
pdb_d614g_combined$dist_d614g <- rowMeans(pdb_d614g_combined[,c("dist_7sxy", "dist_6lzg", "dist_6m0j", "dist_6m17")], na.rm = T)
for(x in 1:nrow(pdb_d614g_combined)){
  pdb_d614g_combined$cv_d614g[x] <- sd(pdb_d614g_combined[x,c("dist_7sxy", "dist_6lzg", "dist_6m0j", "dist_6m17")], na.rm = T) / pdb_d614g_combined$dist_d614g[x]
}

d614g_ace2_31_combined <- pdb_d614g_combined %>% filter(ace2_position == 31)
ggplot() + geom_hline(yintercept = 10) +
  geom_line(data = d614g_ace2_31_combined, aes(x = rbd_position, y = dist_7sxy), color = "red") +
  geom_line(data = d614g_ace2_31_combined, aes(x = rbd_position, y = dist_6lzg), color = "blue") +
  geom_line(data = d614g_ace2_31_combined, aes(x = rbd_position, y = dist_6m0j), color = "green") +
  geom_line(data = d614g_ace2_31_combined, aes(x = rbd_position, y = dist_6m17), color = "black") +
  NULL
```

```{r Alpha RBD-ACE2 contact maps}
pdb_8dlk_dist_frame <- read.csv(file = "Data/PDB_contact_maps/8dlk_ace2row_rbdcol_dist_matrix_min.csv")
pdb_8dlk_dist_frame$position <- as.numeric(rownames(pdb_8dlk_dist_frame))
pdb_8dlk_dist_frame_melt <- melt(pdb_8dlk_dist_frame, id = "position")
pdb_8dlk_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_8dlk_dist_frame_melt$variable,2,5)) + 329
pdb_8dlk_dist_frame_melt$ace2_position <- as.numeric(pdb_8dlk_dist_frame_melt$position) + 18
pdb_8dlk_dist_ace2_positionlist <- as.numeric(unique(pdb_8dlk_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_8dlk_dist_rbd_positionlist <- as.numeric(unique(pdb_8dlk_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_8dlk_dist_frame_melt$ace2_rbd <- paste(pdb_8dlk_dist_frame_melt$ace2_pos, pdb_8dlk_dist_frame_melt$rbd_pos, sep = "_")

pdb_7r0z_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7r0z_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7r0z_dist_frame$position <- as.numeric(rownames(pdb_7r0z_dist_frame))
pdb_7r0z_dist_frame_melt <- melt(pdb_7r0z_dist_frame, id = "position")
pdb_7r0z_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7r0z_dist_frame_melt$variable,2,5)) + 321
pdb_7r0z_dist_frame_melt$ace2_position <- as.numeric(pdb_7r0z_dist_frame_melt$position) + 18
pdb_7r0z_dist_ace2_positionlist <- as.numeric(unique(pdb_7r0z_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7r0z_dist_rbd_positionlist <- as.numeric(unique(pdb_7r0z_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7r0z_dist_frame_melt$ace2_rbd <- paste(pdb_7r0z_dist_frame_melt$ace2_pos, pdb_7r0z_dist_frame_melt$rbd_pos, sep = "_")

pdb_7mjn_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7mjn_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7mjn_dist_frame$position <- as.numeric(rownames(pdb_7mjn_dist_frame))
pdb_7mjn_dist_frame_melt <- melt(pdb_7mjn_dist_frame, id = "position")
pdb_7mjn_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7mjn_dist_frame_melt$variable,2,5)) + 329
pdb_7mjn_dist_frame_melt$ace2_position <- as.numeric(pdb_7mjn_dist_frame_melt$position) + 18
pdb_7mjn_dist_ace2_positionlist <- as.numeric(unique(pdb_7mjn_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7mjn_dist_rbd_positionlist <- as.numeric(unique(pdb_7mjn_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7mjn_dist_frame_melt$ace2_rbd <- paste(pdb_7mjn_dist_frame_melt$ace2_pos, pdb_7mjn_dist_frame_melt$rbd_pos, sep = "_")

pdb_7ekf_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7ekf_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7ekf_dist_frame$position <- as.numeric(rownames(pdb_7ekf_dist_frame))
pdb_7ekf_dist_frame_melt <- melt(pdb_7ekf_dist_frame, id = "position")
pdb_7ekf_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7ekf_dist_frame_melt$variable,2,5)) + 332
pdb_7ekf_dist_frame_melt$ace2_position <- as.numeric(pdb_7ekf_dist_frame_melt$position) + 18
pdb_7ekf_dist_ace2_positionlist <- as.numeric(unique(pdb_7ekf_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7ekf_dist_rbd_positionlist <- as.numeric(unique(pdb_7ekf_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7ekf_dist_frame_melt$ace2_rbd <- paste(pdb_7ekf_dist_frame_melt$ace2_pos, pdb_7ekf_dist_frame_melt$rbd_pos, sep = "_")
pdb_7ekf_dist_frame_melt$dist <- pdb_7ekf_dist_frame_melt$value

## Alpha
pdb_alpha_combined <- merge(merge(pdb_8dlk_dist_frame_melt[,c("ace2_rbd","ace2_position","rbd_position","value")], pdb_7mjn_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_7ekf_dist_frame_melt[,c("ace2_rbd","dist")], by = "ace2_rbd") %>% filter(!is.na(rbd_position) & !is.na(ace2_position))
colnames(pdb_alpha_combined) <- c("ace2_rbd","ace2_position","rbd_position","dist_8dlk","dist_7mjn","dist_7ekf")
pdb_alpha_combined[pdb_alpha_combined$dist_8dlk > 50,"dist_8dlk"] <- NA
pdb_alpha_combined[pdb_alpha_combined$dist_7mjn > 50,"dist_7mjn"] <- NA
pdb_alpha_combined[pdb_alpha_combined$dist_7ekf > 50,"dist_7ekf"] <- NA
pdb_alpha_combined$dist_alpha <- rowMeans(pdb_alpha_combined[,c("dist_8dlk","dist_7mjn","dist_7ekf")], na.rm = T)
for(x in 1:nrow(pdb_alpha_combined)){
  pdb_alpha_combined$cv_alpha[x] <- sd(pdb_alpha_combined[x,c("dist_8dlk", "dist_7mjn","dist_7ekf")], na.rm = T) / pdb_alpha_combined$dist_alpha[x]
}

alpha_ace2_31_combined <- pdb_alpha_combined %>% filter(ace2_position == 31)
ggplot() + geom_hline(yintercept = 10) +
  geom_line(data = alpha_ace2_31_combined, aes(x = rbd_position, y = dist_8dlk), color = "red") +
  geom_line(data = alpha_ace2_31_combined, aes(x = rbd_position, y = dist_7mjn), color = "red") +
  geom_line(data = alpha_ace2_31_combined, aes(x = rbd_position, y = dist_7ekf), color = "green") +
  NULL
```

```{r Beta RBD-ACE2 contact maps}
pdb_7vx4_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7vx4_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7vx4_dist_frame$position <- as.numeric(rownames(pdb_7vx4_dist_frame))
pdb_7vx4_dist_frame_melt <- melt(pdb_7vx4_dist_frame, id = "position")
pdb_7vx4_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7vx4_dist_frame_melt$variable,2,5)) + 332
pdb_7vx4_dist_frame_melt$ace2_position <- as.numeric(pdb_7vx4_dist_frame_melt$position) + 18
pdb_7vx4_dist_ace2_positionlist <- as.numeric(unique(pdb_7vx4_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7vx4_dist_rbd_positionlist <- as.numeric(unique(pdb_7vx4_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7vx4_dist_frame_melt$ace2_rbd <- paste(pdb_7vx4_dist_frame_melt$ace2_pos, pdb_7vx4_dist_frame_melt$rbd_pos, sep = "_")

pdb_8dln_dist_frame <- read.csv(file = "Data/PDB_contact_maps/8dln_ace2row_rbdcol_dist_matrix_min.csv")
pdb_8dln_dist_frame$position <- as.numeric(rownames(pdb_8dln_dist_frame))
pdb_8dln_dist_frame_melt <- melt(pdb_8dln_dist_frame, id = "position")
pdb_8dln_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_8dln_dist_frame_melt$variable,2,5)) + 329
pdb_8dln_dist_frame_melt$ace2_position <- as.numeric(pdb_8dln_dist_frame_melt$position) + 18
pdb_8dln_dist_ace2_positionlist <- as.numeric(unique(pdb_8dln_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_8dln_dist_rbd_positionlist <- as.numeric(unique(pdb_8dln_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_8dln_dist_frame_melt$ace2_rbd <- paste(pdb_8dln_dist_frame_melt$ace2_pos, pdb_8dln_dist_frame_melt$rbd_pos, sep = "_")

pdb_7v80_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7v80_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7v80_dist_frame$position <- as.numeric(rownames(pdb_7v80_dist_frame))
pdb_7v80_dist_frame_melt <- melt(pdb_7v80_dist_frame, id = "position")
pdb_7v80_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7v80_dist_frame_melt$variable,2,5)) + 329
pdb_7v80_dist_frame_melt$ace2_position <- as.numeric(pdb_7v80_dist_frame_melt$position) + 18
pdb_7v80_dist_ace2_positionlist <- as.numeric(unique(pdb_7v80_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7v80_dist_rbd_positionlist <- as.numeric(unique(pdb_7v80_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7v80_dist_frame_melt$ace2_rbd <- paste(pdb_7v80_dist_frame_melt$ace2_pos, pdb_7v80_dist_frame_melt$rbd_pos, sep = "_")

pdb_7ekg_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7ekg_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7ekg_dist_frame$position <- as.numeric(rownames(pdb_7ekg_dist_frame))
pdb_7ekg_dist_frame_melt <- melt(pdb_7ekg_dist_frame, id = "position")
pdb_7ekg_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7ekg_dist_frame_melt$variable,2,5)) + 332
pdb_7ekg_dist_frame_melt$ace2_position <- as.numeric(pdb_7ekg_dist_frame_melt$position) + 18
pdb_7ekg_dist_ace2_positionlist <- as.numeric(unique(pdb_7ekg_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7ekg_dist_rbd_positionlist <- as.numeric(unique(pdb_7ekg_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7ekg_dist_frame_melt$ace2_rbd <- paste(pdb_7ekg_dist_frame_melt$ace2_pos, pdb_7ekg_dist_frame_melt$rbd_pos, sep = "_")
pdb_7ekg_dist_frame_melt$dist_7ekg <- pdb_7ekg_dist_frame_melt$value

pdb_7sy6_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7sy6_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7sy6_dist_frame$position <- as.numeric(rownames(pdb_7sy6_dist_frame))
pdb_7sy6_dist_frame_melt <- melt(pdb_7sy6_dist_frame, id = "position")
pdb_7sy6_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7sy6_dist_frame_melt$variable,2,5)) + 329
pdb_7sy6_dist_frame_melt$ace2_position <- as.numeric(pdb_7sy6_dist_frame_melt$position) + 18
pdb_7sy6_dist_ace2_positionlist <- as.numeric(unique(pdb_7sy6_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7sy6_dist_rbd_positionlist <- as.numeric(unique(pdb_7sy6_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7sy6_dist_frame_melt$ace2_rbd <- paste(pdb_7sy6_dist_frame_melt$ace2_pos, pdb_7sy6_dist_frame_melt$rbd_pos, sep = "_")
pdb_7sy6_dist_frame_melt$dist_7sy6 <- pdb_7sy6_dist_frame_melt$value


## Beta
pdb_beta_combined <- merge(merge(merge(merge(pdb_7vx4_dist_frame_melt[,c("ace2_rbd","ace2_position","rbd_position","value")], pdb_8dln_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_7v80_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_7ekg_dist_frame_melt[,c("ace2_rbd","dist_7ekg")], by = "ace2_rbd"), pdb_7sy6_dist_frame_melt[,c("ace2_rbd","dist_7sy6")], by = "ace2_rbd") %>% filter(!is.na(rbd_position) & !is.na(ace2_position))
colnames(pdb_beta_combined) <- c("ace2_rbd","ace2_position","rbd_position","dist_7vx4", "dist_8dln", "dist_7v80", "dist_7ekg", "dist_7sy6")
pdb_beta_combined[pdb_beta_combined$dist_7vx4 > 50,"dist_7vx4"] <- NA
pdb_beta_combined[pdb_beta_combined$dist_8dln > 50,"dist_8dln"] <- NA
pdb_beta_combined[pdb_beta_combined$dist_7v80 > 50,"dist_7v80"] <- NA
pdb_beta_combined[pdb_beta_combined$dist_7ekg > 50,"dist_7ekg"] <- NA
pdb_beta_combined[pdb_beta_combined$dist_7sy6 > 50,"dist_7sy6"] <- NA
pdb_beta_combined$dist_beta <- rowMeans(pdb_beta_combined[,c("dist_7vx4","dist_8dln", "dist_7v80", "dist_7ekg", "dist_7sy6")], na.rm = T)
for(x in 1:nrow(pdb_beta_combined)){
  pdb_beta_combined$cv_beta[x] <- sd(pdb_beta_combined[x,c("dist_7vx4", "dist_8dln", "dist_7v80", "dist_7ekg", "dist_7sy6")]) / pdb_beta_combined$dist_beta[x]
}

beta_ace2_31_combined <- pdb_beta_combined %>% filter(ace2_position == 31)
ggplot() + geom_hline(yintercept = 10) +
  geom_line(data = beta_ace2_31_combined, aes(x = rbd_position, y = dist_7vx4), color = "red") +
  geom_line(data = beta_ace2_31_combined, aes(x = rbd_position, y = dist_8dln), color = "blue") +
  geom_line(data = beta_ace2_31_combined, aes(x = rbd_position, y = dist_7v80), color = "green") +
  geom_line(data = beta_ace2_31_combined, aes(x = rbd_position, y = dist_7ekg), color = "black")
```

```{r Gamma RBD-ACE2 contact maps}
pdb_7ekc_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7ekc_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7ekc_dist_frame$position <- as.numeric(rownames(pdb_7ekc_dist_frame))
pdb_7ekc_dist_frame_melt <- melt(pdb_7ekc_dist_frame, id = "position")
pdb_7ekc_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7ekc_dist_frame_melt$variable,2,5)) + 332
pdb_7ekc_dist_frame_melt$ace2_position <- as.numeric(pdb_7ekc_dist_frame_melt$position) + 18
pdb_7ekc_dist_ace2_positionlist <- as.numeric(unique(pdb_7ekc_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7ekc_dist_rbd_positionlist <- as.numeric(unique(pdb_7ekc_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7ekc_dist_frame_melt$ace2_rbd <- paste(pdb_7ekc_dist_frame_melt$ace2_pos, pdb_7ekc_dist_frame_melt$rbd_pos, sep = "_")

pdb_7v84_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7v84_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7v84_dist_frame$position <- as.numeric(rownames(pdb_7v84_dist_frame))
pdb_7v84_dist_frame_melt <- melt(pdb_7v84_dist_frame, id = "position") %>% filter(variable != "X")
pdb_7v84_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7v84_dist_frame_melt$variable,2,5)) + 329
pdb_7v84_dist_frame_melt$ace2_position <- as.numeric(pdb_7v84_dist_frame_melt$position) + 18
pdb_7v84_dist_ace2_positionlist <- as.numeric(unique(pdb_7v84_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7v84_dist_rbd_positionlist <- as.numeric(unique(pdb_7v84_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7v84_dist_frame_melt$ace2_rbd <- paste(pdb_7v84_dist_frame_melt$ace2_pos, pdb_7v84_dist_frame_melt$rbd_pos, sep = "_")

pdb_8dlq_dist_frame <- read.csv(file = "Data/PDB_contact_maps/8dlq_ace2row_rbdcol_dist_matrix_min.csv")
pdb_8dlq_dist_frame$position <- as.numeric(rownames(pdb_8dlq_dist_frame))
pdb_8dlq_dist_frame_melt <- melt(pdb_8dlq_dist_frame, id = "position")
pdb_8dlq_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_8dlq_dist_frame_melt$variable,2,5)) + 329
pdb_8dlq_dist_frame_melt$ace2_position <- as.numeric(pdb_8dlq_dist_frame_melt$position) + 18
pdb_8dlq_dist_ace2_positionlist <- as.numeric(unique(pdb_8dlq_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_8dlq_dist_rbd_positionlist <- as.numeric(unique(pdb_8dlq_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_8dlq_dist_frame_melt$ace2_rbd <- paste(pdb_8dlq_dist_frame_melt$ace2_pos, pdb_8dlq_dist_frame_melt$rbd_pos, sep = "_")

pdb_7sy8_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7sy8_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7sy8_dist_frame$position <- as.numeric(rownames(pdb_7sy8_dist_frame))
pdb_7sy8_dist_frame_melt <- melt(pdb_7sy8_dist_frame, id = "position")
pdb_7sy8_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7sy8_dist_frame_melt$variable,2,5)) + 329
pdb_7sy8_dist_frame_melt$ace2_position <- as.numeric(pdb_7sy8_dist_frame_melt$position) + 18
pdb_7sy8_dist_ace2_positionlist <- as.numeric(unique(pdb_7sy8_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7sy8_dist_rbd_positionlist <- as.numeric(unique(pdb_7sy8_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7sy8_dist_frame_melt$ace2_rbd <- paste(pdb_7sy8_dist_frame_melt$ace2_pos, pdb_7sy8_dist_frame_melt$rbd_pos, sep = "_")
pdb_7sy8_dist_frame_melt$dist_7sy8 <- pdb_7sy8_dist_frame_melt$value


## Gamma
pdb_gamma_combined <- merge(merge(merge(pdb_7ekc_dist_frame_melt[,c("ace2_rbd","ace2_position","rbd_position","value")], pdb_7v84_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"),pdb_8dlq_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"),pdb_7sy8_dist_frame_melt[,c("ace2_rbd","dist_7sy8")], by = "ace2_rbd") %>% filter(!is.na(rbd_position) & !is.na(ace2_position))
colnames(pdb_gamma_combined) <- c("ace2_rbd","ace2_position","rbd_position","dist_7ekc", "dist_7v84", "dist_8dlq","dist_7sy8")
pdb_gamma_combined[pdb_gamma_combined$dist_7ekc > 50,"dist_7ekc"] <- NA
pdb_gamma_combined$dist_7v84 <- as.numeric(pdb_gamma_combined$dist_7v84)
for(x in 1:nrow(pdb_gamma_combined)){
  if(!is.na(pdb_gamma_combined$dist_7v84[x])){
    if(pdb_gamma_combined$dist_7v84[x] > 50){
      pdb_gamma_combined$dist_7v84[x] <- NA}}}
pdb_gamma_combined[pdb_gamma_combined$dist_8dlq > 50,"dist_8dlq"] <- NA
pdb_gamma_combined[pdb_gamma_combined$dist_7sy8 > 50,"dist_7sy8"] <- NA
pdb_gamma_combined$dist_gamma <- rowMeans(pdb_gamma_combined[,c("dist_7ekc","dist_7v84", "dist_8dlq","dist_7sy8")], na.rm = T)
for(x in 1:nrow(pdb_gamma_combined)){
  pdb_gamma_combined$cv_gamma[x] <- sd(pdb_gamma_combined[x,c("dist_7ekc","dist_7v84", "dist_8dlq","dist_7sy8")], na.rm = T) / pdb_gamma_combined$dist_gamma[x]
}

gamma_ace2_31_combined <- pdb_gamma_combined %>% filter(ace2_position == 31)
ggplot() + geom_hline(yintercept = 10) +
  geom_line(data = gamma_ace2_31_combined, aes(x = rbd_position, y = dist_7ekc), color = "red") +
  geom_line(data = gamma_ace2_31_combined, aes(x = rbd_position, y = dist_7v84), color = "blue") +
  geom_line(data = gamma_ace2_31_combined, aes(x = rbd_position, y = dist_8dlq), color = "green") +
  geom_line(data = gamma_ace2_31_combined, aes(x = rbd_position, y = dist_8dlq), color = "orange")
```


```{r Delta RBD-ACE2 contact maps}
pdb_7w9i_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7w9i_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7w9i_dist_frame$position <- as.numeric(rownames(pdb_7w9i_dist_frame))
pdb_7w9i_dist_frame_melt <- melt(pdb_7w9i_dist_frame, id = "position")
pdb_7w9i_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7w9i_dist_frame_melt$variable,2,5)) + 332
pdb_7w9i_dist_frame_melt$ace2_position <- as.numeric(pdb_7w9i_dist_frame_melt$position) + 18
pdb_7w9i_dist_ace2_positionlist <- as.numeric(unique(pdb_7w9i_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7w9i_dist_rbd_positionlist <- as.numeric(unique(pdb_7w9i_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7w9i_dist_frame_melt$ace2_rbd <- paste(pdb_7w9i_dist_frame_melt$ace2_pos, pdb_7w9i_dist_frame_melt$rbd_pos, sep = "_")

pdb_7v8b_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7v8b_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7v8b_dist_frame$position <- as.numeric(rownames(pdb_7v8b_dist_frame))
pdb_7v8b_dist_frame_melt <- melt(pdb_7v8b_dist_frame, id = "position")
pdb_7v8b_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7v8b_dist_frame_melt$variable,2,5)) + 329
pdb_7v8b_dist_frame_melt$ace2_position <- as.numeric(pdb_7v8b_dist_frame_melt$position) + 18
pdb_7v8b_dist_ace2_positionlist <- as.numeric(unique(pdb_7v8b_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7v8b_dist_rbd_positionlist <- as.numeric(unique(pdb_7v8b_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7v8b_dist_frame_melt$ace2_rbd <- paste(pdb_7v8b_dist_frame_melt$ace2_pos, pdb_7v8b_dist_frame_melt$rbd_pos, sep = "_")

pdb_7tew_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7tew_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7tew_dist_frame$position <- as.numeric(rownames(pdb_7tew_dist_frame))
pdb_7tew_dist_frame_melt <- melt(pdb_7tew_dist_frame, id = "position")
pdb_7tew_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7tew_dist_frame_melt$variable,2,5)) + 329
pdb_7tew_dist_frame_melt$ace2_position <- as.numeric(pdb_7tew_dist_frame_melt$position) + 18
pdb_7tew_dist_ace2_positionlist <- as.numeric(unique(pdb_7tew_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7tew_dist_rbd_positionlist <- as.numeric(unique(pdb_7tew_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7tew_dist_frame_melt$ace2_rbd <- paste(pdb_7tew_dist_frame_melt$ace2_pos, pdb_7tew_dist_frame_melt$rbd_pos, sep = "_")

pdb_7wbq_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7wbq_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7wbq_dist_frame$position <- as.numeric(rownames(pdb_7wbq_dist_frame))
pdb_7wbq_dist_frame_melt <- melt(pdb_7wbq_dist_frame, id = "position")
pdb_7wbq_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7wbq_dist_frame_melt$variable,2,5)) + 332
pdb_7wbq_dist_frame_melt$ace2_position <- as.numeric(pdb_7wbq_dist_frame_melt$position) + 18
pdb_7wbq_dist_ace2_positionlist <- as.numeric(unique(pdb_7wbq_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7wbq_dist_rbd_positionlist <- as.numeric(unique(pdb_7wbq_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7wbq_dist_frame_melt$ace2_rbd <- paste(pdb_7wbq_dist_frame_melt$ace2_pos, pdb_7wbq_dist_frame_melt$rbd_pos, sep = "_")
pdb_7wbq_dist_frame_melt$dist_7wbq <- pdb_7wbq_dist_frame_melt$value

## Delta
pdb_delta_combined <- merge(merge(merge(pdb_7w9i_dist_frame_melt[,c("ace2_rbd","ace2_position","rbd_position","value")], pdb_7v8b_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_7tew_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_7wbq_dist_frame_melt[,c("ace2_rbd","dist_7wbq")], by = "ace2_rbd") %>% filter(!is.na(rbd_position) & !is.na(ace2_position))
colnames(pdb_delta_combined) <- c("ace2_rbd","ace2_position","rbd_position","dist_7w9i", "dist_7v8b", "dist_7tew", "dist_7wbq")
pdb_delta_combined[pdb_delta_combined$dist_7w9i > 50,"dist_7w9i"] <- NA
pdb_delta_combined[pdb_delta_combined$dist_7v8b > 50,"dist_7v8b"] <- NA
pdb_delta_combined[pdb_delta_combined$dist_7tew > 50,"dist_7tew"] <- NA
pdb_delta_combined[pdb_delta_combined$dist_7wbq > 50,"dist_7wbq"] <- NA
pdb_delta_combined$dist_delta <- rowMeans(pdb_delta_combined[,c("dist_7w9i","dist_7v8b", "dist_7tew", "dist_7wbq")], na.rm = T)
for(x in 1:nrow(pdb_delta_combined)){
  pdb_delta_combined$cv_delta[x] <- sd(pdb_delta_combined[x,c("dist_7w9i","dist_7v8b", "dist_7tew", "dist_7wbq")], na.rm = T) / pdb_delta_combined$dist_delta[x]
}

delta_ace2_31_combined <- pdb_delta_combined %>% filter(ace2_position == 31)
ggplot() + geom_hline(yintercept = 10) +
  geom_line(data = delta_ace2_31_combined, aes(x = rbd_position, y = dist_7w9i), color = "red") +
  geom_line(data = delta_ace2_31_combined, aes(x = rbd_position, y = dist_7v8b), color = "blue") +
  geom_line(data = delta_ace2_31_combined, aes(x = rbd_position, y = dist_7tew), color = "green") +
  NULL
```

```{r BA1 RBD-ACE2 contact maps}
pdb_7wk6_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7wk6_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7wk6_dist_frame$position <- as.numeric(rownames(pdb_7wk6_dist_frame))
pdb_7wk6_dist_frame_melt <- melt(pdb_7wk6_dist_frame, id = "position")
pdb_7wk6_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7wk6_dist_frame_melt$variable,2,5)) + 332
pdb_7wk6_dist_frame_melt$ace2_position <- as.numeric(pdb_7wk6_dist_frame_melt$position) + 18
pdb_7wk6_dist_ace2_positionlist <- as.numeric(unique(pdb_7wk6_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7wk6_dist_rbd_positionlist <- as.numeric(unique(pdb_7wk6_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7wk6_dist_frame_melt$ace2_rbd <- paste(pdb_7wk6_dist_frame_melt$ace2_pos, pdb_7wk6_dist_frame_melt$rbd_pos, sep = "_")

pdb_7t9l_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7t9l_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7t9l_dist_frame$position <- as.numeric(rownames(pdb_7t9l_dist_frame))
pdb_7t9l_dist_frame_melt <- melt(pdb_7t9l_dist_frame, id = "position")
pdb_7t9l_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7t9l_dist_frame_melt$variable,2,5)) + 329
pdb_7t9l_dist_frame_melt$ace2_position <- as.numeric(pdb_7t9l_dist_frame_melt$position) + 18
pdb_7t9l_dist_ace2_positionlist <- as.numeric(unique(pdb_7t9l_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7t9l_dist_rbd_positionlist <- as.numeric(unique(pdb_7t9l_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7t9l_dist_frame_melt$ace2_rbd <- paste(pdb_7t9l_dist_frame_melt$ace2_pos, pdb_7t9l_dist_frame_melt$rbd_pos, sep = "_")

pdb_7wpb_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7wpb_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7wpb_dist_frame$position <- as.numeric(rownames(pdb_7wpb_dist_frame))
pdb_7wpb_dist_frame_melt <- melt(pdb_7wpb_dist_frame, id = "position")
pdb_7wpb_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7wpb_dist_frame_melt$variable,2,5)) + 330
pdb_7wpb_dist_frame_melt$ace2_position <- as.numeric(pdb_7wpb_dist_frame_melt$position) + 18
pdb_7wpb_dist_ace2_positionlist <- as.numeric(unique(pdb_7wpb_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7wpb_dist_rbd_positionlist <- as.numeric(unique(pdb_7wpb_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7wpb_dist_frame_melt$ace2_rbd <- paste(pdb_7wpb_dist_frame_melt$ace2_pos, pdb_7wpb_dist_frame_melt$rbd_pos, sep = "_")

pdb_7wbp_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7wbp_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7wbp_dist_frame$position <- as.numeric(rownames(pdb_7wbp_dist_frame))
pdb_7wbp_dist_frame_melt <- melt(pdb_7wbp_dist_frame, id = "position") %>% filter(variable != "X")
pdb_7wbp_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7wbp_dist_frame_melt$variable,2,5)) + 332
pdb_7wbp_dist_frame_melt$ace2_position <- as.numeric(pdb_7wbp_dist_frame_melt$position) + 18
pdb_7wbp_dist_ace2_positionlist <- as.numeric(unique(pdb_7wbp_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7wbp_dist_rbd_positionlist <- as.numeric(unique(pdb_7wbp_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7wbp_dist_frame_melt$ace2_rbd <- paste(pdb_7wbp_dist_frame_melt$ace2_pos, pdb_7wbp_dist_frame_melt$rbd_pos, sep = "_")
pdb_7wbp_dist_frame_melt$dist_7wbp <- pdb_7wbp_dist_frame_melt$value

pdb_7whh_dist_frame <- read.csv(file = "Data/PDB_contact_maps/7whh_ace2row_rbdcol_dist_matrix_min.csv")
pdb_7whh_dist_frame$position <- as.numeric(rownames(pdb_7whh_dist_frame))
pdb_7whh_dist_frame_melt <- melt(pdb_7whh_dist_frame, id = "position") %>% filter(variable != "X")
pdb_7whh_dist_frame_melt$rbd_position <-as.numeric(substr(pdb_7whh_dist_frame_melt$variable,2,5)) + 332
pdb_7whh_dist_frame_melt$ace2_position <- as.numeric(pdb_7whh_dist_frame_melt$position) + 18
pdb_7whh_dist_ace2_positionlist <- as.numeric(unique(pdb_7whh_dist_frame_melt$ace2_position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7whh_dist_rbd_positionlist <- as.numeric(unique(pdb_7whh_dist_frame_melt$position)) ## For interpretation, need to add 316 later since actually starts at residue 331
pdb_7whh_dist_frame_melt$ace2_rbd <- paste(pdb_7whh_dist_frame_melt$ace2_pos, pdb_7whh_dist_frame_melt$rbd_pos, sep = "_")
pdb_7whh_dist_frame_melt$dist_7whh <- pdb_7whh_dist_frame_melt$value

## BA1
pdb_ba1_combined <- merge(merge(merge(merge(pdb_7wk6_dist_frame_melt[,c("ace2_rbd","ace2_position","rbd_position","value")], pdb_7t9l_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_7wpb_dist_frame_melt[,c("ace2_rbd","value")], by = "ace2_rbd"), pdb_7wbp_dist_frame_melt[,c("ace2_rbd","dist_7wbp")], by = "ace2_rbd"), pdb_7whh_dist_frame_melt[,c("ace2_rbd","dist_7whh")], by = "ace2_rbd") %>% filter(!is.na(rbd_position) & !is.na(ace2_position)) %>% filter(ace2_position != 401)
colnames(pdb_ba1_combined) <- c("ace2_rbd","ace2_position","rbd_position","dist_7wk6", "dist_7t9l", "dist_7wpb", "dist_7wbp", "dist_7whh")
pdb_ba1_combined[pdb_ba1_combined$dist_7wk6 > 50,"dist_7wk6"] <- NA
pdb_ba1_combined[pdb_ba1_combined$dist_7t9l > 50,"dist_7t9l"] <- NA
pdb_ba1_combined[pdb_ba1_combined$dist_7wpb > 50,"dist_7wpb"] <- NA
pdb_ba1_combined[pdb_ba1_combined$dist_7wbp > 50,"dist_7wbp"] <- NA
pdb_ba1_combined[pdb_ba1_combined$dist_7whh > 50,"dist_7whh"] <- NA
pdb_ba1_combined$dist_ba1 <- rowMeans(pdb_ba1_combined[,c("dist_7wk6","dist_7t9l", "dist_7wpb","dist_7wbp", "dist_7whh")], na.rm = T)
for(x in 1:nrow(pdb_ba1_combined)){
  pdb_ba1_combined$cv_ba1[x] <- sd(pdb_ba1_combined[x,c("dist_7wk6","dist_7t9l", "dist_7wpb","dist_7wbp", "dist_7whh")], na.rm = T) / pdb_ba1_combined$dist_ba1[x]
}

ba1_ace2_31_combined <- pdb_ba1_combined %>% filter(ace2_position == 31)
ggplot() + geom_hline(yintercept = 10) +
  geom_line(data = ba1_ace2_31_combined, aes(x = rbd_position, y = dist_7wk6), color = "black") +
  geom_line(data = ba1_ace2_31_combined, aes(x = rbd_position, y = dist_7t9l), color = "black") +
  geom_line(data = ba1_ace2_31_combined, aes(x = rbd_position, y = dist_7wpb), color = "black") +
  geom_line(data = ba1_ace2_31_combined, aes(x = rbd_position, y = dist_7wbp), color = "green") +
  geom_line(data = ba1_ace2_31_combined, aes(x = rbd_position, y = dist_7whh), color = "red") +
  NULL
```

```{r}
pdb_variants <- merge(merge(merge(merge(merge(pdb_d614g_combined[,c("ace2_rbd","ace2_position","rbd_position","dist_d614g")], pdb_alpha_combined[,c("ace2_rbd","dist_alpha")], by = "ace2_rbd"), pdb_beta_combined[,c("ace2_rbd","dist_beta")], by = "ace2_rbd"), pdb_gamma_combined[,c("ace2_rbd","dist_gamma")], by = "ace2_rbd"), pdb_delta_combined[,c("ace2_rbd","dist_delta")], by = "ace2_rbd"), pdb_ba1_combined[,c("ace2_rbd","dist_ba1")], by = "ace2_rbd") #pdb_muba1_combined[,c("ace2_rbd","dist_muba1")], by = "ace2_rbd"), 

## Check to make sure all RBD numbers are aligned
ace2_31_combined <- pdb_variants %>% filter(ace2_position == 31)
ggplot() + scale_y_continuous(limits = c(0,30)) +
  geom_hline(yintercept = 10) +
  geom_line(data = ace2_31_combined, aes(x = rbd_position, y = dist_d614g), color = "black") +
  geom_line(data = ace2_31_combined, aes(x = rbd_position, y = dist_alpha), color = "black") +
  geom_line(data = ace2_31_combined, aes(x = rbd_position, y = dist_beta), color = "green") +
  geom_line(data = ace2_31_combined, aes(x = rbd_position, y = dist_gamma), color = "orange") +
  geom_line(data = ace2_31_combined, aes(x = rbd_position, y = dist_delta), color = "black") +
  geom_line(data = ace2_31_combined, aes(x = rbd_position, y = dist_ba1), color = "black")
  NULL
  
```


```{r distance cutoff}

# Filter for positions less than 10 angstroms for all structures
cutoff_med <- 10
pdb_variants_med <- pdb_variants %>% filter(dist_d614g < cutoff_med & dist_alpha < cutoff_med & dist_beta < cutoff_med & dist_gamma < cutoff_med & dist_delta < cutoff_med & dist_ba1 < cutoff_med)

ggplot() + geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_alpha))
ggplot() + geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_beta))
ggplot() + geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_gamma))
ggplot() + geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_delta))
ggplot() + geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_ba1))
ggplot() + geom_point(data = pdb_variants_med, aes(x = dist_beta, y = dist_gamma))
ggplot() + geom_point(data = pdb_variants_med, aes(x = dist_beta, y = dist_delta))

Angstrom_scatterplot_D614G_Delta <- ggplot() + labs(x = "Angstroms (D614G)", y = "Angstroms (Delta)") +
  geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_delta), alpha = 0.3)
Angstrom_scatterplot_D614G_Delta
ggsave(file = "Plots/Angstrom_scatterplot_D614G_Delta.pdf", Angstrom_scatterplot_D614G_Delta, height = 1.5, width = 1.5)

Angstrom_scatterplot_Beta_Gamma <- ggplot() + labs(x = "Angstroms (Gamma)", y = "Angstroms (Beta)") +
  geom_point(data = pdb_variants_med, aes(x = dist_gamma, y = dist_beta), alpha = 0.3)
Angstrom_scatterplot_Beta_Gamma
ggsave(file = "Plots/Angstrom_scatterplot_Beta_Gamma.pdf", Angstrom_scatterplot_Beta_Gamma, height = 1.5, width = 1.5)

Angstrom_scatterplot_D614G_Beta <- ggplot() + labs(x = "Angstroms (D614G)", y = "Angstroms (Beta)") +
  geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_beta), alpha = 0.3)
Angstrom_scatterplot_D614G_Beta
ggsave(file = "Plots/Angstrom_scatterplot_D614G_Beta.pdf", Angstrom_scatterplot_D614G_Beta, height = 1.5, width = 1.5)

Angstrom_scatterplot_D614G_BA1 <- ggplot() + labs(x = "Angstroms (D614G)", y = "Angstroms (BA1)") +
  geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_ba1), alpha = 0.3)
Angstrom_scatterplot_D614G_BA1
ggsave(file = "Plots/Angstrom_scatterplot_D614G_BA1.pdf", Angstrom_scatterplot_D614G_BA1, height = 1.5, width = 1.5)

```

```{r Filter for positions less than 10 angstroms for all structures}

critical_value <- 1.96 #1.96

## Now do pairwise differences between distances based on the SARS-CoV-2 v ariant
## D614G
pdb_variants_med$d614g_alpha <- (pdb_variants_med$dist_d614g - pdb_variants_med$dist_alpha) / rowMeans(pdb_variants_med[,c("dist_d614g","dist_alpha")])
pdb_variants_med$d614g_alpha_outlier <- pdb_variants_med$d614g_alpha < (mean(pdb_variants_med$d614g_alpha) - sd(pdb_variants_med$d614g_alpha) * critical_value) | pdb_variants_med$d614g_alpha > (mean(pdb_variants_med$d614g_alpha) + sd(pdb_variants_med$d614g_alpha) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = d614g_alpha), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(d614g_alpha_outlier == TRUE), aes(x = d614g_alpha), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$d614g_beta <- (pdb_variants_med$dist_d614g - pdb_variants_med$dist_beta) / rowMeans(pdb_variants_med[,c("dist_d614g","dist_beta")])
pdb_variants_med$d614g_beta_outlier <- pdb_variants_med$d614g_beta < (mean(pdb_variants_med$d614g_beta) - sd(pdb_variants_med$d614g_beta) * critical_value) | pdb_variants_med$d614g_beta > (mean(pdb_variants_med$d614g_beta) + sd(pdb_variants_med$d614g_beta) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = d614g_beta), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(d614g_beta_outlier == TRUE), aes(x = d614g_beta), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$d614g_gamma <- (pdb_variants_med$dist_d614g - pdb_variants_med$dist_gamma) / rowMeans(pdb_variants_med[,c("dist_d614g","dist_gamma")])
pdb_variants_med$d614g_gamma_outlier <- pdb_variants_med$d614g_gamma < (mean(pdb_variants_med$d614g_gamma) - sd(pdb_variants_med$d614g_gamma) * critical_value) | pdb_variants_med$d614g_gamma > (mean(pdb_variants_med$d614g_gamma) + sd(pdb_variants_med$d614g_gamma) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = d614g_gamma), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(d614g_gamma_outlier == TRUE), aes(x = d614g_gamma), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$d614g_delta <- (pdb_variants_med$dist_d614g - pdb_variants_med$dist_delta) / rowMeans(pdb_variants_med[,c("dist_d614g","dist_delta")])
pdb_variants_med$d614g_delta_outlier <- pdb_variants_med$d614g_delta < (mean(pdb_variants_med$d614g_delta) - sd(pdb_variants_med$d614g_delta) * critical_value) | pdb_variants_med$d614g_delta > (mean(pdb_variants_med$d614g_delta) + sd(pdb_variants_med$d614g_delta) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = d614g_delta), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(d614g_delta_outlier == TRUE), aes(x = d614g_delta), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$d614g_ba1 <- (pdb_variants_med$dist_d614g - pdb_variants_med$dist_ba1) / rowMeans(pdb_variants_med[,c("dist_d614g","dist_ba1")])
pdb_variants_med$d614g_ba1_outlier <- pdb_variants_med$d614g_ba1 < (mean(pdb_variants_med$d614g_ba1) - sd(pdb_variants_med$d614g_ba1) * critical_value) | pdb_variants_med$d614g_ba1 > (mean(pdb_variants_med$d614g_ba1) + sd(pdb_variants_med$d614g_ba1) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = d614g_ba1), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(d614g_ba1_outlier == TRUE), aes(x = d614g_ba1), binwidth = 0.04, fill = "red", color = "black")

## Alpha
pdb_variants_med$alpha_beta <- (pdb_variants_med$dist_alpha - pdb_variants_med$dist_beta) / rowMeans(pdb_variants_med[,c("dist_alpha","dist_beta")])
pdb_variants_med$alpha_beta_outlier <- pdb_variants_med$alpha_beta < (mean(pdb_variants_med$alpha_beta) - sd(pdb_variants_med$alpha_beta) * critical_value) | pdb_variants_med$alpha_beta > (mean(pdb_variants_med$alpha_beta) + sd(pdb_variants_med$alpha_beta) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = alpha_beta), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(alpha_beta_outlier == TRUE), aes(x = alpha_beta), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$alpha_gamma <- (pdb_variants_med$dist_alpha - pdb_variants_med$dist_gamma) / rowMeans(pdb_variants_med[,c("dist_alpha","dist_gamma")])
pdb_variants_med$alpha_gamma_outlier <- pdb_variants_med$alpha_gamma < (mean(pdb_variants_med$alpha_gamma) - sd(pdb_variants_med$alpha_gamma) * critical_value) | pdb_variants_med$alpha_gamma > (mean(pdb_variants_med$alpha_gamma) + sd(pdb_variants_med$alpha_gamma) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = alpha_gamma), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(alpha_gamma_outlier == TRUE), aes(x = alpha_gamma), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$alpha_delta <- (pdb_variants_med$dist_alpha - pdb_variants_med$dist_delta) / rowMeans(pdb_variants_med[,c("dist_alpha","dist_delta")])
pdb_variants_med$alpha_delta_outlier <- pdb_variants_med$alpha_delta < (mean(pdb_variants_med$alpha_delta) - sd(pdb_variants_med$alpha_delta) * critical_value) | pdb_variants_med$alpha_delta > (mean(pdb_variants_med$alpha_delta) + sd(pdb_variants_med$alpha_delta) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = alpha_delta), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(alpha_delta_outlier == TRUE), aes(x = alpha_delta), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$alpha_ba1 <- (pdb_variants_med$dist_alpha - pdb_variants_med$dist_ba1) / rowMeans(pdb_variants_med[,c("dist_alpha","dist_ba1")])
pdb_variants_med$alpha_ba1_outlier <- pdb_variants_med$alpha_ba1 < (mean(pdb_variants_med$alpha_ba1) - sd(pdb_variants_med$alpha_ba1) * critical_value) | pdb_variants_med$alpha_ba1 > (mean(pdb_variants_med$alpha_ba1) + sd(pdb_variants_med$alpha_ba1) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = alpha_ba1), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(alpha_ba1_outlier == TRUE), aes(x = alpha_ba1), binwidth = 0.04, fill = "red", color = "black")

## Beta
pdb_variants_med$beta_gamma <- (pdb_variants_med$dist_beta - pdb_variants_med$dist_gamma) / rowMeans(pdb_variants_med[,c("dist_beta","dist_gamma")])
pdb_variants_med$beta_gamma_outlier <- pdb_variants_med$beta_gamma < (mean(pdb_variants_med$beta_gamma) - sd(pdb_variants_med$beta_gamma) * critical_value) | pdb_variants_med$beta_gamma > (mean(pdb_variants_med$beta_gamma) + sd(pdb_variants_med$beta_gamma) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = beta_gamma), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(beta_gamma_outlier == TRUE), aes(x = beta_gamma), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$beta_delta <- (pdb_variants_med$dist_beta - pdb_variants_med$dist_delta) / rowMeans(pdb_variants_med[,c("dist_beta","dist_delta")])
pdb_variants_med$beta_delta_outlier <- pdb_variants_med$beta_delta < (mean(pdb_variants_med$beta_delta) - sd(pdb_variants_med$beta_delta) * critical_value) | pdb_variants_med$beta_delta > (mean(pdb_variants_med$beta_delta) + sd(pdb_variants_med$beta_delta) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = beta_delta), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(beta_delta_outlier == TRUE), aes(x = beta_delta), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$beta_ba1 <- (pdb_variants_med$dist_beta - pdb_variants_med$dist_ba1) / rowMeans(pdb_variants_med[,c("dist_beta","dist_ba1")])
pdb_variants_med$beta_ba1_outlier <- pdb_variants_med$beta_ba1 < (mean(pdb_variants_med$beta_ba1) - sd(pdb_variants_med$beta_ba1) * critical_value) | pdb_variants_med$beta_ba1 > (mean(pdb_variants_med$beta_ba1) + sd(pdb_variants_med$beta_ba1) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = beta_ba1), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(beta_ba1_outlier == TRUE), aes(x = beta_ba1), binwidth = 0.04, fill = "red", color = "black")

## Gamma
pdb_variants_med$gamma_delta <- (pdb_variants_med$dist_gamma - pdb_variants_med$dist_delta) / rowMeans(pdb_variants_med[,c("dist_gamma","dist_delta")])
pdb_variants_med$gamma_delta_outlier <- pdb_variants_med$gamma_delta < (mean(pdb_variants_med$gamma_delta) - sd(pdb_variants_med$gamma_delta) * critical_value) | pdb_variants_med$gamma_delta > (mean(pdb_variants_med$gamma_delta) + sd(pdb_variants_med$gamma_delta) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = gamma_delta), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(gamma_delta_outlier == TRUE), aes(x = gamma_delta), binwidth = 0.04, fill = "red", color = "black")

pdb_variants_med$gamma_ba1 <- (pdb_variants_med$dist_gamma - pdb_variants_med$dist_ba1) / rowMeans(pdb_variants_med[,c("dist_gamma","dist_ba1")])
pdb_variants_med$gamma_ba1_outlier <- pdb_variants_med$gamma_ba1 < (mean(pdb_variants_med$gamma_ba1) - sd(pdb_variants_med$gamma_ba1) * critical_value) | pdb_variants_med$gamma_ba1 > (mean(pdb_variants_med$gamma_ba1) + sd(pdb_variants_med$gamma_ba1) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = gamma_ba1), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(gamma_ba1_outlier == TRUE), aes(x = gamma_ba1), binwidth = 0.04, fill = "red", color = "black")

## Delta
pdb_variants_med$delta_ba1 <- (pdb_variants_med$dist_delta - pdb_variants_med$dist_ba1) / rowMeans(pdb_variants_med[,c("dist_delta","dist_ba1")])
pdb_variants_med$delta_ba1_outlier <- pdb_variants_med$delta_ba1 < (mean(pdb_variants_med$delta_ba1) - sd(pdb_variants_med$delta_ba1) * critical_value) | pdb_variants_med$delta_ba1 > (mean(pdb_variants_med$delta_ba1) + sd(pdb_variants_med$delta_ba1) * critical_value)
ggplot() + scale_x_continuous(limits = c(-0.5,0.5)) + geom_histogram(data = pdb_variants_med, aes(x = delta_ba1), binwidth = 0.04, color = "black") +
   geom_histogram(data = pdb_variants_med %>% filter(delta_ba1_outlier == TRUE), aes(x = delta_ba1), binwidth = 0.04, fill = "red", color = "black")

## Making the scatterplots again but this time coloring the outliers as red

Angstrom_scatterplot_D614G_Delta2 <- ggplot() + labs(x = "Angstroms (D614G)", y = "Angstroms (Delta)") +
  geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_delta), alpha = 0.3) +
  geom_point(data = pdb_variants_med %>% filter(d614g_delta_outlier == TRUE), aes(x = dist_d614g, y = dist_delta), color = "red", alpha = 0.4) +
  NULL
Angstrom_scatterplot_D614G_Delta2
ggsave(file = "Plots/Angstrom_scatterplot_D614G_Delta2.pdf", Angstrom_scatterplot_D614G_Delta2, height = 1.5, width = 1.5)

Angstrom_scatterplot_Beta_Gamma2 <- ggplot() + labs(x = "Angstroms (Gamma)", y = "Angstroms (Beta)") +
  geom_point(data = pdb_variants_med, aes(x = dist_gamma, y = dist_beta), alpha = 0.3) +
  geom_point(data = pdb_variants_med %>% filter(beta_gamma_outlier == TRUE), aes(x = dist_gamma, y = dist_beta), color = "red", alpha = 0.4) +
  NULL
Angstrom_scatterplot_Beta_Gamma2
ggsave(file = "Plots/Angstrom_scatterplot_Beta_Gamma2.pdf", Angstrom_scatterplot_Beta_Gamma2, height = 1.5, width = 1.5)

Angstrom_scatterplot_D614G_Beta2 <- ggplot() + labs(x = "Angstroms (D614G)", y = "Angstroms (Beta)") +
  geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_beta), alpha = 0.3) +
  geom_point(data = pdb_variants_med %>% filter(d614g_beta_outlier == TRUE), aes(x = dist_d614g, y = dist_beta), color = "red", alpha = 0.4) +
  NULL
Angstrom_scatterplot_D614G_Beta2
ggsave(file = "Plots/Angstrom_scatterplot_D614G_Beta2.pdf", Angstrom_scatterplot_D614G_Beta2, height = 1.5, width = 1.5)

Angstrom_scatterplot_D614G_BA1_2 <- ggplot() + labs(x = "Angstroms (D614G)", y = "Angstroms (BA1)") +
  geom_point(data = pdb_variants_med, aes(x = dist_d614g, y = dist_ba1), alpha = 0.3) +
  geom_point(data = pdb_variants_med %>% filter(d614g_ba1_outlier == TRUE), aes(x = dist_d614g, y = dist_ba1), color = "red", alpha = 0.4) +
  NULL
Angstrom_scatterplot_D614G_BA1_2
ggsave(file = "Plots/Angstrom_scatterplot_D614G_BA1_2.pdf", Angstrom_scatterplot_D614G_BA1_2, height = 1.5, width = 1.5)


## Now add everything together 
pdb_variants_med_combined <- data.frame("distance" = c(pdb_variants_med$d614g_alpha, pdb_variants_med$d614g_beta, pdb_variants_med$d614g_beta, pdb_variants_med$d614g_gamma, pdb_variants_med$d614g_ba1, pdb_variants_med$alpha_beta, pdb_variants_med$alpha_gamma, pdb_variants_med$alpha_delta, pdb_variants_med$alpha_ba1, pdb_variants_med$beta_gamma, pdb_variants_med$beta_delta, pdb_variants_med$beta_ba1, pdb_variants_med$gamma_delta, pdb_variants_med$gamma_ba1, pdb_variants_med$delta_ba1),
                                        "outlier" = c(pdb_variants_med$d614g_alpha_outlier, pdb_variants_med$d614g_beta_outlier, pdb_variants_med$d614g_beta_outlier, pdb_variants_med$d614g_gamma_outlier, pdb_variants_med$d614g_ba1_outlier, pdb_variants_med$alpha_beta_outlier, pdb_variants_med$alpha_gamma_outlier, pdb_variants_med$alpha_delta_outlier, pdb_variants_med$alpha_ba1_outlier, pdb_variants_med$beta_gamma_outlier, pdb_variants_med$beta_delta_outlier, pdb_variants_med$beta_ba1_outlier, pdb_variants_med$gamma_delta_outlier, pdb_variants_med$gamma_ba1_outlier, pdb_variants_med$delta_ba1_outlier))


Combined_distance_difference_histogram <- ggplot() + 
  scale_x_continuous(limits = c(-0.4,0.6)) + scale_y_break(c(100, 200), scales = 0.5) + 
  labs(x = "Z-score for difference\nin atomic distance", y = "Number of datapoints") +
  geom_histogram(data = pdb_variants_med_combined, aes(x = distance), binwidth = 0.04, fill = "grey80", color = "grey40") +
   geom_histogram(data = pdb_variants_med_combined %>% filter(outlier == TRUE), aes(x = distance), binwidth = 0.04, fill = "red", color = "red4")
Combined_distance_difference_histogram
ggsave(file = "Plots/Combined_distance_difference_histogram.pdf", Combined_distance_difference_histogram, height = 2, width = 2)

## Figure out what points are outside of the 95% interval here value here

pdb_variants_med_combined$zscore <-  (pdb_variants_med_combined$distance - mean(pdb_variants_med_combined$distance)) / sd(pdb_variants_med_combined$distance)
pdb_variants_med_combined$zscore_outlier <- (pdb_variants_med_combined$zscore > 2.33 | pdb_variants_med_combined$zscore < -2.33)

Combined_zscore_histogram <- ggplot() + 
  scale_x_continuous() + scale_y_break(c(60, 70), scales = 0.5) + 
  labs(x = "Z-score for difference\nin atomic distance", y = "Number of datapoints") +
  geom_histogram(data = pdb_variants_med_combined, aes(x = zscore), binwidth = 0.2, fill = "grey80", color = "grey40") +
  geom_histogram(data = pdb_variants_med_combined %>% filter(zscore_outlier == TRUE), aes(x = zscore), binwidth = 0.2, fill = "red", color = "red4") +
  NULL
Combined_zscore_histogram
ggsave(file = "Plots/Combined_zscore_histogram.pdf", Combined_zscore_histogram, height = 2, width = 2)


Combined_zscore_histogram_continuous <- ggplot() + 
  scale_x_continuous() + 
  labs(x = "Z-score for difference\nin atomic distance", y = "Number of datapoints") +
  geom_histogram(data = pdb_variants_med_combined, aes(x = zscore), binwidth = 0.2, fill = "grey80", color = "grey40") +
  geom_histogram(data = pdb_variants_med_combined %>% filter(zscore_outlier == TRUE), aes(x = zscore), binwidth = 0.2, fill = "red", color = "red4") +
  NULL
Combined_zscore_histogram_continuous
ggsave(file = "Plots/Combined_zscore_histogram_continuous.pdf", Combined_zscore_histogram_continuous, height = 5, width = 4)
```

```{r Making a network of interacting residues}
#network_pairs_orig <- pdb_variants_med %>% filter( d614g_alpha_outlier == TRUE | d614g_beta_outlier == TRUE | d614g_gamma_outlier == TRUE | d614g_delta_outlier == TRUE | d614g_ba1_outlier == TRUE | alpha_beta_outlier == TRUE | alpha_gamma_outlier == TRUE | alpha_delta_outlier == TRUE | alpha_ba1_outlier == TRUE | beta_gamma_outlier == TRUE | beta_delta_outlier == TRUE | beta_ba1_outlier == TRUE | gamma_delta_outlier == TRUE | gamma_ba1_outlier == TRUE | delta_ba1_outlier == TRUE)

network_pairs_orig <- pdb_variants_med %>% filter()

network_pairs_orig$n_outliers <- rowSums(network_pairs_orig[,c("d614g_alpha_outlier","d614g_beta_outlier","d614g_gamma_outlier","d614g_delta_outlier","d614g_ba1_outlier","alpha_beta_outlier","alpha_gamma_outlier","alpha_delta_outlier","alpha_ba1_outlier","beta_gamma_outlier","beta_delta_outlier","beta_ba1_outlier","gamma_delta_outlier","gamma_ba1_outlier","delta_ba1_outlier")])

## If a cutoff of 5 or more, then its looking at outliers that were observed in at least 33% of cases
network_pairs <- network_pairs_orig %>% filter(n_outliers >= 5) %>% select(colnames(network_pairs_orig)[colnames(network_pairs_orig) != "n_outliers"])

## Now make a network on only residues that come close in one structure
cutoff_low <- 3.4
pdb_variants_low <- pdb_variants_med %>% filter(dist_d614g < cutoff_low | dist_alpha < cutoff_low | dist_beta < cutoff_low | dist_gamma < cutoff_low | dist_delta < cutoff_low | dist_ba1 < cutoff_low)

network_pairs2 <- rbind(network_pairs, pdb_variants_low) %>% distinct()
length(unique(network_pairs2$ace2_position))
unique(network_pairs2$ace2_position)
length(unique(network_pairs2$rbd_position))
unique(network_pairs2$rbd_position)

network_pairs2$n_outliers <- rowSums(network_pairs2[,c("d614g_alpha_outlier","d614g_beta_outlier","d614g_gamma_outlier","d614g_delta_outlier","d614g_ba1_outlier","alpha_beta_outlier","alpha_gamma_outlier","alpha_delta_outlier","alpha_ba1_outlier","beta_gamma_outlier","beta_delta_outlier","beta_ba1_outlier","gamma_delta_outlier","gamma_ba1_outlier","delta_ba1_outlier")])
network_pairs2 <- network_pairs2 %>% arrange(desc(n_outliers))
network_pairs2$order <- seq(1,nrow(network_pairs2))

Distance_outlier_histogram <- ggplot() + labs(x = "Number of times the datapoint\nwas outside 95% CI", y = "Number of datapoints") +
  scale_y_continuous(breaks = seq(0,8,2)) +
  geom_histogram(data = network_pairs2, aes(x = n_outliers), binwidth = 1, fill = "grey80", color = "black")
Distance_outlier_histogram
ggsave(file = "Plots/Distance_outlier_histogram.pdf", Distance_outlier_histogram, height = 1.75, width = 2)
```

```{r Listing which positions are now being considered}
## These commands are to output commands into Pymol!

print("fetch 6m17")
print("create d614g_ace2, chain B")
print("create d614g_rbd, chain E")
paste("select ace2_intxn, d614g_ace2 and resi",gsub(", ", "+", toString(unique(network_pairs2$ace2_position))),"and not name c+n+o+ca")
paste("select rbd_intxn, d614g_rbd and resi",gsub(", ", "+", toString(unique(network_pairs2$rbd_position))),"and not name c+n+o+ca")
print("set cartoon_transparency, 0.5")
print("rotate y, 90")
```

```{r pca based on atomic coordinates of key interacting residues}
# pca on atomic coordinates
pdb_variants_for_pca <- data.frame(t(network_pairs2[seq(4,10)]))[1:6,]

pca_for_pdb = prcomp(pdb_variants_for_pca, scale. = TRUE)

#head(pca_for_pdb$rotation)
#round(summary(pca_for_pdb)$importance[2],0)

# create data frame with scores

scores = as.data.frame(pca_for_pdb$x)
scores$labels <- c("D614G", "Alpha", "Beta","Gamma","Delta","BA1")

s2_variant_average_pca_plot <- ggplot() + 
  xlab(paste("Principal component 1\n(",round(summary(pca_for_pdb)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Principal component 2\n(",round(summary(pca_for_pdb)$importance[5] * 100,0),"% of variance)",sep="")) + 
  geom_text_repel(data = scores, aes(x = PC1, y = PC2, label = labels), segment.color = 'grey90', point.padding = 0.05, size = 2, segment.alpha = 0.5, color = "red") +
  geom_point(data = scores, aes(x = PC1, y = PC2), size = 1) +
  NULL
s2_variant_average_pca_plot
ggsave(file = "Plots/s2_variant_average_pca_plot.pdf", s2_variant_average_pca_plot, height = 1.8, width = 1.75)

### Not getting all of the individual structure datapoints in here

all_pdb <- merge(merge(merge(merge(merge(pdb_d614g_combined[,c("ace2_rbd","dist_7sxy","dist_6lzg","dist_6m0j")],
                 pdb_alpha_combined[,c("ace2_rbd","dist_8dlk","dist_7mjn","dist_7ekf")], by = "ace2_rbd"),
                 pdb_beta_combined[,c("ace2_rbd","dist_7vx4","dist_8dln","dist_7v80","dist_7ekg","dist_7sy6")], by = "ace2_rbd"),
                 pdb_gamma_combined[,c("ace2_rbd","dist_7ekc","dist_7v84","dist_8dlq","dist_7sy8")], by = "ace2_rbd"),
                 pdb_delta_combined[,c("ace2_rbd","dist_7w9i","dist_7v8b","dist_7tew","dist_7wbq")], by = "ace2_rbd"),
                 pdb_ba1_combined[,c("ace2_rbd","dist_7wk6","dist_7t9l","dist_7wpb","dist_7wbp","dist_7whh")], by = "ace2_rbd")#,
                 #pdb_ptmuts_combined[,c("ace2_rbd","dist_7sy4","dist_7sy2","dist_7sy0")], by = "ace2_rbd")
                 
all_pdb_intxn <- merge(network_pairs2[,c("ace2_rbd","ace2_position","rbd_position")], all_pdb, by = "ace2_rbd")

pdb_variants_for_pca2 <- data.frame(t(all_pdb_intxn[seq(4,27)]))
#pdb_variants_for_pca3 <- pdb_variants_for_pca2[complete.cases(pdb_variants_for_pca2),] ## This would be keeping only complete rows, but I want to do columns instead
pdb_variants_for_pca3 <- pdb_variants_for_pca2[ , colSums(is.na(pdb_variants_for_pca2)) == 0]

pca_for_pdb2 = prcomp(pdb_variants_for_pca3, scale = TRUE)
scores2 = as.data.frame(pca_for_pdb2$x)
scores2$pdb <- substr(rownames(scores2),6,12)

pdb_contact_maps_key <- read.csv(file = "Data/pdb_contact_maps/pdb_contact_maps_key.csv", header = T)

scores3 <- merge(scores2, pdb_contact_maps_key, by = "pdb") #%>% filter(type == "cryo")
scores4 <- scores3[,c("pdb","group","PC1","PC2")]

s2_variant_pca_plot <- ggplot() + 
  xlab(paste("Principal component 1 (",round(summary(pca_for_pdb2)$importance[2] * 100,0),"% of variance explained)",sep="")) + 
  ylab(paste("Principal component 2 (",round(summary(pca_for_pdb2)$importance[5] * 100,0),"% of variance explained)",sep="")) + 
  geom_text_repel(data = scores3, aes(x = PC1, y = PC2, label = pdb), segment.color = 'grey90', point.padding = 0.05, size = 3, segment.alpha = 0.5, color = "red") +
  geom_point(data = scores3, aes(x = PC1, y = PC2, color = group, shape = type), size = 1) +
  NULL
s2_variant_pca_plot



### Not getting all of the individual structure datapoints in here

all_cryo <- merge(merge(merge(merge(merge(pdb_d614g_combined[,c("ace2_rbd","dist_7sxy")],
                 pdb_alpha_combined[,c("ace2_rbd","dist_8dlk","dist_7mjn")], by = "ace2_rbd"),
                 pdb_beta_combined[,c("ace2_rbd","dist_7vx4","dist_8dln","dist_7v80","dist_7sy6")], by = "ace2_rbd"),
                 pdb_gamma_combined[,c("ace2_rbd","dist_7v84","dist_8dlq","dist_7sy8")], by = "ace2_rbd"),
                 pdb_delta_combined[,c("ace2_rbd","dist_7w9i","dist_7v8b","dist_7tew")], by = "ace2_rbd"),
                 pdb_ba1_combined[,c("ace2_rbd","dist_7wk6","dist_7t9l","dist_7wpb")], by = "ace2_rbd")#,
                 #pdb_ptmuts_combined[,c("ace2_rbd","dist_7sy4","dist_7sy2","dist_7sy0")], by = "ace2_rbd")
                 
all_cryo_intxn <- merge(network_pairs2[,c("ace2_rbd","ace2_position","rbd_position")], all_cryo, by = "ace2_rbd")

cryo_variants_for_pca2 <- data.frame(t(all_cryo_intxn[seq(4,9)]))
#cryo_variants_for_pca3 <- cryo_variants_for_pca2[complete.cases(cryo_variants_for_pca2),] ## This would be keeping only complete rows, but I want to do columns instead
cryo_variants_for_pca3 <- cryo_variants_for_pca2[ , colSums(is.na(cryo_variants_for_pca2)) == 0]

pca_for_cryo2 = prcomp(cryo_variants_for_pca3, scale. = TRUE)
all_cryo_scores = as.data.frame(pca_for_cryo2$x)
all_cryo_scores$pdb <- substr(rownames(all_cryo_scores),6,12)

cryo_contact_maps_key <- read.csv(file = "Data/PDB_contact_maps/PDB_contact_maps_key.csv", header = T)

all_cryo_scores2 <- merge(all_cryo_scores, pdb_contact_maps_key, by = "pdb") #%>% filter(type == "cryo")
all_cryo_scores3 <- all_cryo_scores2[,c("pdb","group","type","PC1","PC2")]

s2_variant_pca_plot <- ggplot() + 
  xlab(paste("Principal component 1 (",round(summary(pca_for_cryo2)$importance[2] * 100,0),"% of variance explained)",sep="")) + 
  ylab(paste("Principal component 2 (",round(summary(pca_for_cryo2)$importance[5] * 100,0),"% of variance explained)",sep="")) + 
  geom_text_repel(data = all_cryo_scores3, aes(x = PC1, y = PC2, label = pdb), segment.color = 'grey90', point.padding = 0.05, size = 3, segment.alpha = 0.5, color = "red") +
  geom_point(data = all_cryo_scores3, aes(x = PC1, y = PC2, color = group, shape = type), size = 1) +
  NULL
s2_variant_pca_plot
```

```{r Making a network of interacting residues}
library(network)
#install.packages("GGally")
library(GGally)

#https://stackoverflow.com/questions/53179442/how-to-set-fix-node-positions-with-ggnet2
set.seed(1234567)
frac_cutoff <- 0.2

## Reimport D614G atom positions
pdb_6lzg_atoms <- read.table(file = "Data/PDB_contact_maps/parsed_6lzg_atoms.tsv", header = TRUE, sep = "\t")
pdb_6lzg_ace2 <- pdb_6lzg_atoms %>% filter(chain == "A") %>% mutate(position = as.numeric(position))
pdb_6lzg_rbd <- pdb_6lzg_atoms %>% filter(chain == "B") %>% mutate(position = as.numeric(position))
pdb_6lzg_ace2_positionlist <- as.numeric(unique(pdb_6lzg_ace2$position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_6lzg_rbd_positionlist <- as.numeric(unique(pdb_6lzg_rbd$position)) ## For interpretation, need to add 316 later since actually starts at residue 331

## ACE2 SEQUENCE
ace2_residues <- pdb_6lzg_ace2 %>% filter(name == "C") %>% mutate(aa3 = aa, ace2_position = position) %>% select(ace2_position,aa3)
ace2_residues$orig <- lapply(ace2_residues$aa3, to_single_notation)
ace2_residues$ace2_residue <- paste0(ace2_residues$orig,ace2_residues$ace2_position)

## D614G SEQUENCE
d614g_residues <- pdb_6lzg_rbd %>% filter(name == "C") %>% mutate(aa3 = aa, rbd_position = position) %>% select(rbd_position,aa3)
d614g_residues$orig <- lapply(d614g_residues$aa3, to_single_notation)
d614g_residues$d614g_residue <- paste0(d614g_residues$orig,d614g_residues$rbd_position)

```


```{r Making a network of interacting residues for D614G with actual distances}
frac_cutoff <- 0
## D614G only
for(x in 1:nrow(network_pairs2)){
  if(network_pairs2$dist_d614g[x] > frac_cutoff | network_pairs2$dist_d614g[x] < -frac_cutoff){network_pairs2$d614g_network[x] <- network_pairs2$dist_d614g[x]} else{network_pairs2$d614g_network[x] <- NA}
}

network_pairs2_temp <- network_pairs2[,c("order","ace2_position","rbd_position")]
network_pairs2_temp2 <- merge(network_pairs2_temp,ace2_residues[,c("ace2_position","ace2_residue")],by = "ace2_position", all.x = T)
network_pairs2_temp3 <- merge(network_pairs2_temp2,d614g_residues[,c("rbd_position","d614g_residue")],by = "rbd_position", all.x = T)
network_pairs2_temp4 <- merge(network_pairs2_temp3,d614g_residues[,c("rbd_position","d614g_residue")],by = "rbd_position", all.x = T)
for(x in 1:nrow(network_pairs2_temp4)){network_pairs2_temp4$rbd_residue[x] <- network_pairs2_temp4$d614g_residue.x[x]}
network_pairs2_temp4 <- network_pairs2_temp4 %>% arrange(order)

simple_edge_df <- data.frame(
  from = network_pairs2_temp4$ace2_residue,
  to = network_pairs2_temp4$rbd_residue,
  weight = round(network_pairs2$d614g_network,3), #n_outliers
  weight2 = round(network_pairs2$d614g_alpha,3), #n_outliers
  stringsAsFactors = FALSE
)
for(x in 1:nrow(simple_edge_df)){if(abs(simple_edge_df$weight2[x]) > 0.2){simple_edge_df$weight2[x] <- simple_edge_df$weight2[x]} else{simple_edge_df$weight2[x] <- NA}}
simple_edge_network <- as.network(simple_edge_df)
# https://briatte.github.io/ggnet/
temp_network <- as.matrix(plot.network(simple_edge_network, mode = "fruchtermanreingold", layout.par = list(repulse.rad = 1000, cell.jitter = 0)))
rs = 1

x = network.vertex.names(simple_edge_network)
x = ifelse(x %in% unique(network_pairs2_temp4$ace2_residue), "ACE2", "RBD")
simple_edge_network %v% "type" = x

set.edge.attribute(simple_edge_network, "color", ifelse(simple_edge_network %e% "weight" > 0 & simple_edge_network %e% "weight" <= 3, "black",
                                                              ifelse(simple_edge_network %e% "weight" > 3 & simple_edge_network %e% "weight" <= 4, "grey50",
                                                                     ifelse(simple_edge_network %e% "weight" > 4 & simple_edge_network %e% "weight" <= 6, "grey70",
                                                                            ifelse(simple_edge_network %e% "weight" > 6 & simple_edge_network %e% "weight" <= 10, "grey90","white")))))


D614G_networkplot <- ggnet2(simple_edge_network, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.9, edge.color = "color", edge.alpha = 0.75, label.color = "purple", label.alpha = 0.9, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(D614G_networkplot)
ggsave(file = "Plots/D614G_networkplot.pdf", D614G_networkplot, height = 2.25, width = 3)
# select pair_42_498, ace2 and resi 42 | rbd and resi 498
```

```{r Making a network of interacting residues for alpha}

pdb_8dlk_atoms <- read.table(file = "Data/PDB_contact_maps/parsed_8dlk_atoms.tsv", header = TRUE, sep = "\t")
pdb_8dlk_ace2 <- pdb_8dlk_atoms %>% filter(chain == "E") %>% mutate(position = as.numeric(position))
pdb_8dlk_rbd <- pdb_8dlk_atoms %>% filter(chain == "B") %>% mutate(position = as.numeric(position))
pdb_8dlk_ace2_positionlist <- as.numeric(unique(pdb_8dlk_ace2$position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_8dlk_rbd_positionlist <- as.numeric(unique(pdb_8dlk_rbd$position)) ## For interpretation, need to add 316 later since actually starts at residue 331

## ALPHA SEQUENCE
alpha_residues <- pdb_8dlk_rbd %>% filter(name == "C") %>% mutate(aa3 = aa, rbd_position = position) %>% select(rbd_position,aa3)
alpha_residues$orig <- lapply(alpha_residues$aa3, to_single_notation)
alpha_residues$alpha_residue <- paste0(alpha_residues$orig,alpha_residues$rbd_position)

## D614G and alpha
network_pairs2$d614g_alpha_network <- NA
for(x in 1:nrow(network_pairs2)){
  if(network_pairs2$d614g_alpha_outlier[x] == TRUE){network_pairs2$d614g_alpha_network[x] <- network_pairs2$dist_d614g[x] - network_pairs2$dist_alpha[x]} else{network_pairs2$d614g_alpha_network[x] <- 0}
}

network_pairs2_alpha_temp <- network_pairs2[,c("order","ace2_position","rbd_position")]
network_pairs2_alpha_temp2 <- merge(network_pairs2_alpha_temp,ace2_residues[,c("ace2_position","ace2_residue")],by = "ace2_position", all.x = T)
network_pairs2_alpha_temp3 <- merge(network_pairs2_alpha_temp2,d614g_residues[,c("rbd_position","d614g_residue")],by = "rbd_position", all.x = T)
network_pairs2_alpha_temp4 <- merge(network_pairs2_alpha_temp3,alpha_residues[,c("rbd_position","alpha_residue")],by = "rbd_position", all.x = T)
for(x in 1:nrow(network_pairs2_alpha_temp4)){
  if(network_pairs2_alpha_temp4$d614g_residue[x] == network_pairs2_alpha_temp4$alpha_residue[x]){
    network_pairs2_alpha_temp4$rbd_mutant[x] <- FALSE
    network_pairs2_alpha_temp4$rbd_residue[x] <- network_pairs2_alpha_temp4$d614g_residue[x]} else{
      network_pairs2_alpha_temp4$rbd_mutant[x] <- TRUE
      network_pairs2_alpha_temp4$rbd_residue[x] <- paste0(network_pairs2_alpha_temp4$d614g_residue[x],substr(network_pairs2_alpha_temp4$alpha_residue[x],1,1))}
}
network_pairs2_alpha_temp4 <- network_pairs2_alpha_temp4 %>% arrange(order)

simple_edge_df_alpha <- data.frame(
  from = network_pairs2_alpha_temp4$ace2_residue,
  to = network_pairs2_alpha_temp4$rbd_residue,
  weight = round(network_pairs2$dist_alpha,2), #n_outliers
  weight2 = round(network_pairs2$d614g_alpha_network,1), #n_outliers
  weight3 = round(network_pairs2$d614g_alpha_network,1), #n_outliers
  stringsAsFactors = FALSE
)
simple_edge_df_alpha[simple_edge_df_alpha$weight2 == 0,"weight3"] <- NA
simple_edge_network_alpha <- as.network(simple_edge_df_alpha)

x = network.vertex.names(simple_edge_network_alpha)
x = ifelse(x %in% unique(network_pairs2_alpha_temp4$ace2_residue), "ACE2", "RBD")
simple_edge_network_alpha %v% "type" = x
simple_edge_network_alpha %v% "rbd_mut" = network_pairs2_alpha_temp4$rbd_mutant

## Distances for Alpha by itself
set.edge.attribute(simple_edge_network_alpha, "color", ifelse(simple_edge_network_alpha %e% "weight" > 0 & simple_edge_network_alpha %e% "weight" <= 3, "black",
                                                              ifelse(simple_edge_network_alpha %e% "weight" > 3 & simple_edge_network_alpha %e% "weight" <= 4, "grey40",
                                                                     ifelse(simple_edge_network_alpha %e% "weight" > 4 & simple_edge_network_alpha %e% "weight" <= 6, "grey60",
                                                                            ifelse(simple_edge_network_alpha %e% "weight" > 6 & simple_edge_network_alpha %e% "weight" <= 10, "grey80","white")))))


Alpha_networkplot <- ggnet2(simple_edge_network_alpha, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha = 0.3, label.color = "purple", label.alpha = 0.4, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(Alpha_networkplot)
ggsave(file = "Plots/Alpha_networkplot.pdf", Alpha_networkplot, height = 2.25, width = 3)

## Changes in distance as compared to D614G
simple_edge_network_alpha2 <- simple_edge_network_alpha

set.edge.attribute(simple_edge_network_alpha2, "color", 
                   ifelse(simple_edge_network_alpha2 %e% "weight2" > 1 & simple_edge_network_alpha2 %e% "weight2" < 2, "orange",
                          ifelse(simple_edge_network_alpha2 %e% "weight2" > 2, "red",
                                 ifelse(simple_edge_network_alpha2 %e% "weight2" < -2, "blue",
                                 ifelse(simple_edge_network_alpha2 %e% "weight2" < -1, "cyan","grey75")))))

D614G_alpha_networkplot <- ggnet2(simple_edge_network_alpha2, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "weight3", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha = 0.3, label.color = "purple", label.alpha = 0.5, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(D614G_alpha_networkplot)
ggsave(file = "Plots/D614G_alpha_networkplot.pdf", D614G_alpha_networkplot, height = 2.25, width = 3)
# select pair_42_498, ace2 and resi 42 | rbd and resi 498
```

```{r Making a network of interacting residues for beta}

pdb_7vx4_atoms <- read.table(file = "Data/PDB_contact_maps/parsed_7vx4_atoms.tsv", header = TRUE, sep = "\t")
pdb_7vx4_ace2 <- pdb_7vx4_atoms %>% filter(chain == "A") %>% mutate(position = as.numeric(position))
pdb_7vx4_rbd <- pdb_7vx4_atoms %>% filter(chain == "E") %>% mutate(position = as.numeric(position))
pdb_7vx4_ace2_positionlist <- as.numeric(unique(pdb_7vx4_ace2$position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7vx4_rbd_positionlist <- as.numeric(unique(pdb_7vx4_rbd$position)) ## For interpretation, need to add 316 later since actually starts at residue 331

## beta SEQUENCE
beta_residues <- pdb_7vx4_rbd %>% filter(name == "C") %>% mutate(aa3 = aa, rbd_position = position) %>% select(rbd_position,aa3)
beta_residues$orig <- lapply(beta_residues$aa3, to_single_notation)
beta_residues$beta_residue <- paste0(beta_residues$orig,beta_residues$rbd_position)

## D614G and beta
network_pairs2$d614g_beta_network <- NA
for(x in 1:nrow(network_pairs2)){
  if(network_pairs2$d614g_beta_outlier[x] == TRUE){network_pairs2$d614g_beta_network[x] <- network_pairs2$dist_d614g[x] - network_pairs2$dist_beta[x]} else{network_pairs2$d614g_beta_network[x] <- 0}
}

network_pairs2_beta_temp <- network_pairs2[,c("order","ace2_position","rbd_position")]
network_pairs2_beta_temp2 <- merge(network_pairs2_beta_temp,ace2_residues[,c("ace2_position","ace2_residue")],by = "ace2_position", all.x = T)
network_pairs2_beta_temp3 <- merge(network_pairs2_beta_temp2,d614g_residues[,c("rbd_position","d614g_residue")],by = "rbd_position", all.x = T)
network_pairs2_beta_temp4 <- merge(network_pairs2_beta_temp3,beta_residues[,c("rbd_position","beta_residue")],by = "rbd_position", all.x = T)
for(x in 1:nrow(network_pairs2_beta_temp4)){
  if(network_pairs2_beta_temp4$d614g_residue[x] == network_pairs2_beta_temp4$beta_residue[x]){
    network_pairs2_beta_temp4$rbd_mutant[x] <- FALSE
    network_pairs2_beta_temp4$rbd_residue[x] <- network_pairs2_beta_temp4$d614g_residue[x]} else{
      network_pairs2_beta_temp4$rbd_mutant[x] <- TRUE
      network_pairs2_beta_temp4$rbd_residue[x] <- paste0(network_pairs2_beta_temp4$d614g_residue[x],substr(network_pairs2_beta_temp4$beta_residue[x],1,1))}
}
network_pairs2_beta_temp4 <- network_pairs2_beta_temp4 %>% arrange(order)

simple_edge_df_beta <- data.frame(
  from = network_pairs2_beta_temp4$ace2_residue,
  to = network_pairs2_beta_temp4$rbd_residue,
  weight = round(network_pairs2$dist_beta,2), #n_outliers
  weight2 = round(network_pairs2$d614g_beta_network,1), #n_outliers
  weight3 = round(network_pairs2$d614g_beta_network,1), #n_outliers
  stringsAsFactors = FALSE
)
simple_edge_df_beta[simple_edge_df_beta$weight2 == 0,"weight3"] <- NA
simple_edge_network_beta <- as.network(simple_edge_df_beta)

x = network.vertex.names(simple_edge_network_beta)
x = ifelse(x %in% unique(network_pairs2_beta_temp4$ace2_residue), "ACE2", "RBD")
simple_edge_network_beta %v% "type" = x
simple_edge_network_beta %v% "rbd_mut" = network_pairs2_beta_temp4$rbd_mutant

## Distances for beta by itself
set.edge.attribute(simple_edge_network_beta, "color", ifelse(simple_edge_network_beta %e% "weight" > 0 & simple_edge_network_beta %e% "weight" <= 3, "black",
                                                              ifelse(simple_edge_network_beta %e% "weight" > 3 & simple_edge_network_beta %e% "weight" <= 4, "grey30",
                                                                     ifelse(simple_edge_network_beta %e% "weight" > 4 & simple_edge_network_beta %e% "weight" <= 6, "grey50",
                                                                            ifelse(simple_edge_network_beta %e% "weight" > 6 & simple_edge_network_beta %e% "weight" <= 8, "grey70",
                                                                                   ifelse(simple_edge_network_beta %e% "weight" > 8 & simple_edge_network_beta %e% "weight" <= 10, "grey90","white"))))))


beta_networkplot <- ggnet2(simple_edge_network_beta, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha= 0.3, label.color = "purple", label.beta = 0.4, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(beta_networkplot)
ggsave(file = "Plots/beta_networkplot.pdf", beta_networkplot, height = 2.25, width = 3)

## Changes in distance as compared to D614G
simple_edge_network_beta2 <- simple_edge_network_beta

set.edge.attribute(simple_edge_network_beta2, "color", 
                   ifelse(simple_edge_network_beta2 %e% "weight2" > 1 & simple_edge_network_beta2 %e% "weight2" < 2, "orange",
                          ifelse(simple_edge_network_beta2 %e% "weight2" > 2, "red",
                                 ifelse(simple_edge_network_beta2 %e% "weight2" < -2, "blue",
                                 ifelse(simple_edge_network_beta2 %e% "weight2" < -1, "cyan","grey75")))))

D614G_beta_networkplot <- ggnet2(simple_edge_network_beta2, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "weight3", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha = 0.3, label.color = "purple", label.beta = 0.5, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(D614G_beta_networkplot)
ggsave(file = "Plots/D614G_beta_networkplot.pdf", D614G_beta_networkplot, height = 2.25, width = 3)
# select pair_42_498, ace2 and resi 42 | rbd and resi 498
```

```{r Making a network of interacting residues for gamma}

pdb_7ekc_atoms <- read.table(file = "Data/PDB_contact_maps/parsed_7ekc_atoms.tsv", header = TRUE, sep = "\t")
pdb_7ekc_ace2 <- pdb_7ekc_atoms %>% filter(chain == "A") %>% mutate(position = as.numeric(position))
pdb_7ekc_rbd <- pdb_7ekc_atoms %>% filter(chain == "B") %>% mutate(position = as.numeric(position))
pdb_7ekc_ace2_positionlist <- as.numeric(unique(pdb_7ekc_ace2$position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7ekc_rbd_positionlist <- as.numeric(unique(pdb_7ekc_rbd$position)) ## For interpretation, need to add 316 later since actually starts at residue 331

## gamma SEQUENCE
gamma_residues <- pdb_7ekc_rbd %>% filter(name == "C") %>% mutate(aa3 = aa, rbd_position = position) %>% select(rbd_position,aa3)
gamma_residues$orig <- lapply(gamma_residues$aa3, to_single_notation)
gamma_residues$gamma_residue <- paste0(gamma_residues$orig,gamma_residues$rbd_position)

## D614G and gamma
network_pairs2$d614g_gamma_network <- NA
for(x in 1:nrow(network_pairs2)){
  if(network_pairs2$d614g_gamma_outlier[x] == TRUE){network_pairs2$d614g_gamma_network[x] <- network_pairs2$dist_d614g[x] - network_pairs2$dist_gamma[x]} else{network_pairs2$d614g_gamma_network[x] <- 0}
}

network_pairs2_gamma_temp <- network_pairs2[,c("order","ace2_position","rbd_position")]
network_pairs2_gamma_temp2 <- merge(network_pairs2_gamma_temp,ace2_residues[,c("ace2_position","ace2_residue")],by = "ace2_position", all.x = T)
network_pairs2_gamma_temp3 <- merge(network_pairs2_gamma_temp2,d614g_residues[,c("rbd_position","d614g_residue")],by = "rbd_position", all.x = T)
network_pairs2_gamma_temp4 <- merge(network_pairs2_gamma_temp3,gamma_residues[,c("rbd_position","gamma_residue")],by = "rbd_position", all.x = T)
for(x in 1:nrow(network_pairs2_gamma_temp4)){
  if(network_pairs2_gamma_temp4$d614g_residue[x] == network_pairs2_gamma_temp4$gamma_residue[x]){
    network_pairs2_gamma_temp4$rbd_mutant[x] <- FALSE
    network_pairs2_gamma_temp4$rbd_residue[x] <- network_pairs2_gamma_temp4$d614g_residue[x]} else{
      network_pairs2_gamma_temp4$rbd_mutant[x] <- TRUE
      network_pairs2_gamma_temp4$rbd_residue[x] <- paste0(network_pairs2_gamma_temp4$d614g_residue[x],substr(network_pairs2_gamma_temp4$gamma_residue[x],1,1))}
}
network_pairs2_gamma_temp4 <- network_pairs2_gamma_temp4 %>% arrange(order)

simple_edge_df_gamma <- data.frame(
  from = network_pairs2_gamma_temp4$ace2_residue,
  to = network_pairs2_gamma_temp4$rbd_residue,
  weight = round(network_pairs2$dist_gamma,2), #n_outliers
  weight2 = round(network_pairs2$d614g_gamma_network,1), #n_outliers
  weight3 = round(network_pairs2$d614g_gamma_network,1), #n_outliers
  stringsAsFactors = FALSE
)
simple_edge_df_gamma[simple_edge_df_gamma$weight2 == 0,"weight3"] <- NA
simple_edge_network_gamma <- as.network(simple_edge_df_gamma)

x = network.vertex.names(simple_edge_network_gamma)
x = ifelse(x %in% unique(network_pairs2_gamma_temp4$ace2_residue), "ACE2", "RBD")
simple_edge_network_gamma %v% "type" = x
simple_edge_network_gamma %v% "rbd_mut" = network_pairs2_gamma_temp4$rbd_mutant

## Distances for gamma by itself
set.edge.attribute(simple_edge_network_gamma, "color", ifelse(simple_edge_network_gamma %e% "weight" > 0 & simple_edge_network_gamma %e% "weight" <= 3, "black",
                                                              ifelse(simple_edge_network_gamma %e% "weight" > 3 & simple_edge_network_gamma %e% "weight" <= 4, "grey30",
                                                                     ifelse(simple_edge_network_gamma %e% "weight" > 4 & simple_edge_network_gamma %e% "weight" <= 6, "grey50",
                                                                            ifelse(simple_edge_network_gamma %e% "weight" > 6 & simple_edge_network_gamma %e% "weight" <= 8, "grey70",
                                                                                   ifelse(simple_edge_network_gamma %e% "weight" > 8 & simple_edge_network_gamma %e% "weight" <= 10, "grey90","white"))))))


gamma_networkplot <- ggnet2(simple_edge_network_gamma, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha= 0.3, label.color = "purple", label.gamma = 0.4, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(gamma_networkplot)
ggsave(file = "Plots/gamma_networkplot.pdf", gamma_networkplot, height = 2.25, width = 3)

## Changes in distance as compared to D614G
simple_edge_network_gamma2 <- simple_edge_network_gamma

set.edge.attribute(simple_edge_network_gamma2, "color", 
                   ifelse(simple_edge_network_gamma2 %e% "weight2" > 1 & simple_edge_network_gamma2 %e% "weight2" < 2, "orange",
                          ifelse(simple_edge_network_gamma2 %e% "weight2" > 2, "red",
                                 ifelse(simple_edge_network_gamma2 %e% "weight2" < -2, "blue",
                                 ifelse(simple_edge_network_gamma2 %e% "weight2" < -1, "cyan","grey75")))))

D614G_gamma_networkplot <- ggnet2(simple_edge_network_gamma2, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "weight3", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha = 0.3, label.color = "purple", label.gamma = 0.5, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(D614G_gamma_networkplot)
ggsave(file = "Plots/D614G_gamma_networkplot.pdf", D614G_gamma_networkplot, height = 2.25, width = 3)
# select pair_42_498, ace2 and resi 42 | rbd and resi 498
```

```{r Making a network of interacting residues for delta}

pdb_7w9i_atoms <- read.table(file = "Data/PDB_contact_maps/parsed_7w9i_atoms.tsv", header = TRUE, sep = "\t")
pdb_7w9i_ace2 <- pdb_7w9i_atoms %>% filter(chain == "A") %>% mutate(position = as.numeric(position))
pdb_7w9i_rbd <- pdb_7w9i_atoms %>% filter(chain == "E") %>% mutate(position = as.numeric(position))
pdb_7w9i_ace2_positionlist <- as.numeric(unique(pdb_7w9i_ace2$position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7w9i_rbd_positionlist <- as.numeric(unique(pdb_7w9i_rbd$position)) ## For interpretation, need to add 316 later since actually starts at residue 331

## delta SEQUENCE
delta_residues <- pdb_7w9i_rbd %>% filter(name == "C") %>% mutate(aa3 = aa, rbd_position = position) %>% select(rbd_position,aa3)
delta_residues$orig <- lapply(delta_residues$aa3, to_single_notation)
delta_residues$delta_residue <- paste0(delta_residues$orig,delta_residues$rbd_position)

## D614G and delta
network_pairs2$d614g_delta_network <- NA
for(x in 1:nrow(network_pairs2)){
  if(network_pairs2$d614g_delta_outlier[x] == TRUE){network_pairs2$d614g_delta_network[x] <- network_pairs2$dist_d614g[x] - network_pairs2$dist_delta[x]} else{network_pairs2$d614g_delta_network[x] <- 0}
}

network_pairs2_delta_temp <- network_pairs2[,c("order","ace2_position","rbd_position")]
network_pairs2_delta_temp2 <- merge(network_pairs2_delta_temp,ace2_residues[,c("ace2_position","ace2_residue")],by = "ace2_position", all.x = T)
network_pairs2_delta_temp3 <- merge(network_pairs2_delta_temp2,d614g_residues[,c("rbd_position","d614g_residue")],by = "rbd_position", all.x = T)
network_pairs2_delta_temp4 <- merge(network_pairs2_delta_temp3,delta_residues[,c("rbd_position","delta_residue")],by = "rbd_position", all.x = T)
for(x in 1:nrow(network_pairs2_delta_temp4)){
  if(network_pairs2_delta_temp4$d614g_residue[x] == network_pairs2_delta_temp4$delta_residue[x]){
    network_pairs2_delta_temp4$rbd_mutant[x] <- FALSE
    network_pairs2_delta_temp4$rbd_residue[x] <- network_pairs2_delta_temp4$d614g_residue[x]} else{
      network_pairs2_delta_temp4$rbd_mutant[x] <- TRUE
      network_pairs2_delta_temp4$rbd_residue[x] <- paste0(network_pairs2_delta_temp4$d614g_residue[x],substr(network_pairs2_delta_temp4$delta_residue[x],1,1))}
}
network_pairs2_delta_temp4 <- network_pairs2_delta_temp4 %>% arrange(order)

simple_edge_df_delta <- data.frame(
  from = network_pairs2_delta_temp4$ace2_residue,
  to = network_pairs2_delta_temp4$rbd_residue,
  weight = round(network_pairs2$dist_delta,2), #n_outliers
  weight2 = round(network_pairs2$d614g_delta_network,1), #n_outliers
  weight3 = round(network_pairs2$d614g_delta_network,1), #n_outliers
  stringsAsFactors = FALSE
)
simple_edge_df_delta[simple_edge_df_delta$weight2 == 0,"weight3"] <- NA
simple_edge_network_delta <- as.network(simple_edge_df_delta)

x = network.vertex.names(simple_edge_network_delta)
x = ifelse(x %in% unique(network_pairs2_delta_temp4$ace2_residue), "ACE2", "RBD")
simple_edge_network_delta %v% "type" = x
simple_edge_network_delta %v% "rbd_mut" = network_pairs2_delta_temp4$rbd_mutant

## Distances for delta by itself
set.edge.attribute(simple_edge_network_delta, "color", ifelse(simple_edge_network_delta %e% "weight" > 0 & simple_edge_network_delta %e% "weight" <= 3, "black",
                                                              ifelse(simple_edge_network_delta %e% "weight" > 3 & simple_edge_network_delta %e% "weight" <= 4, "grey30",
                                                                     ifelse(simple_edge_network_delta %e% "weight" > 4 & simple_edge_network_delta %e% "weight" <= 6, "grey50",
                                                                            ifelse(simple_edge_network_delta %e% "weight" > 6 & simple_edge_network_delta %e% "weight" <= 8, "grey70",
                                                                                   ifelse(simple_edge_network_delta %e% "weight" > 8 & simple_edge_network_delta %e% "weight" <= 10, "grey90","white"))))))


delta_networkplot <- ggnet2(simple_edge_network_delta, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha= 0.3, label.color = "purple", label.delta = 0.4, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(delta_networkplot)
ggsave(file = "Plots/delta_networkplot.pdf", delta_networkplot, height = 2.25, width = 3)

## Changes in distance as compared to D614G
simple_edge_network_delta2 <- simple_edge_network_delta

set.edge.attribute(simple_edge_network_delta2, "color", 
                   ifelse(simple_edge_network_delta2 %e% "weight2" > 1 & simple_edge_network_delta2 %e% "weight2" < 2, "orange",
                          ifelse(simple_edge_network_delta2 %e% "weight2" > 2, "red",
                                 ifelse(simple_edge_network_delta2 %e% "weight2" < -2, "blue",
                                 ifelse(simple_edge_network_delta2 %e% "weight2" < -1, "cyan","grey75")))))

D614G_delta_networkplot <- ggnet2(simple_edge_network_delta2, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "weight3", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha = 0.3, label.color = "purple", label.delta = 0.5, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(D614G_delta_networkplot)
ggsave(file = "Plots/D614G_delta_networkplot.pdf", D614G_delta_networkplot, height = 2.25, width = 3)
# select pair_42_498, ace2 and resi 42 | rbd and resi 498
```

```{r Making a network of interacting residues for ba1}

pdb_7wk6_atoms <- read.table(file = "Data/PDB_contact_maps/parsed_7wk6_atoms.tsv", header = TRUE, sep = "\t")
pdb_7wk6_ace2 <- pdb_7wk6_atoms %>% filter(chain == "A") %>% mutate(position = as.numeric(position))
pdb_7wk6_rbd <- pdb_7wk6_atoms %>% filter(chain == "E") %>% mutate(position = as.numeric(position))
pdb_7wk6_ace2_positionlist <- as.numeric(unique(pdb_7wk6_ace2$position)) ## For interpretation, need to add 18 later since actually starts at residue 19
pdb_7wk6_rbd_positionlist <- as.numeric(unique(pdb_7wk6_rbd$position)) ## For interpretation, need to add 316 later since actually starts at residue 331

## ba1 SEQUENCE
ba1_residues <- pdb_7wk6_rbd %>% filter(name == "C") %>% mutate(aa3 = aa, rbd_position = position) %>% select(rbd_position,aa3)
ba1_residues$orig <- lapply(ba1_residues$aa3, to_single_notation)
ba1_residues$ba1_residue <- paste0(ba1_residues$orig,ba1_residues$rbd_position)

## D614G and ba1
network_pairs2$d614g_ba1_network <- NA
for(x in 1:nrow(network_pairs2)){
  if(network_pairs2$d614g_ba1_outlier[x] == TRUE){network_pairs2$d614g_ba1_network[x] <- network_pairs2$dist_d614g[x] - network_pairs2$dist_ba1[x]} else{network_pairs2$d614g_ba1_network[x] <- 0}
}

network_pairs2_ba1_temp <- network_pairs2[,c("order","ace2_position","rbd_position")]
network_pairs2_ba1_temp2 <- merge(network_pairs2_ba1_temp,ace2_residues[,c("ace2_position","ace2_residue")],by = "ace2_position", all.x = T)
network_pairs2_ba1_temp3 <- merge(network_pairs2_ba1_temp2,d614g_residues[,c("rbd_position","d614g_residue")],by = "rbd_position", all.x = T)
network_pairs2_ba1_temp4 <- merge(network_pairs2_ba1_temp3,ba1_residues[,c("rbd_position","ba1_residue")],by = "rbd_position", all.x = T)
for(x in 1:nrow(network_pairs2_ba1_temp4)){
  if(network_pairs2_ba1_temp4$d614g_residue[x] == network_pairs2_ba1_temp4$ba1_residue[x]){
    network_pairs2_ba1_temp4$rbd_mutant[x] <- FALSE
    network_pairs2_ba1_temp4$rbd_residue[x] <- network_pairs2_ba1_temp4$d614g_residue[x]} else{
      network_pairs2_ba1_temp4$rbd_mutant[x] <- TRUE
      network_pairs2_ba1_temp4$rbd_residue[x] <- paste0(network_pairs2_ba1_temp4$d614g_residue[x],substr(network_pairs2_ba1_temp4$ba1_residue[x],1,1))}
}
network_pairs2_ba1_temp4 <- network_pairs2_ba1_temp4 %>% arrange(order)

simple_edge_df_ba1 <- data.frame(
  from = network_pairs2_ba1_temp4$ace2_residue,
  to = network_pairs2_ba1_temp4$rbd_residue,
  weight = round(network_pairs2$dist_ba1,2), #n_outliers
  weight2 = round(network_pairs2$d614g_ba1_network,1), #n_outliers
  weight3 = round(network_pairs2$d614g_ba1_network,1), #n_outliers
  stringsAsFactors = FALSE
)
simple_edge_df_ba1[simple_edge_df_ba1$weight2 == 0,"weight3"] <- NA
simple_edge_network_ba1 <- as.network(simple_edge_df_ba1)

x = network.vertex.names(simple_edge_network_ba1)
x = ifelse(x %in% unique(network_pairs2_ba1_temp4$ace2_residue), "ACE2", "RBD")
simple_edge_network_ba1 %v% "type" = x
simple_edge_network_ba1 %v% "rbd_mut" = network_pairs2_ba1_temp4$rbd_mutant

## Distances for ba1 by itself
set.edge.attribute(simple_edge_network_ba1, "color", ifelse(simple_edge_network_ba1 %e% "weight" > 0 & simple_edge_network_ba1 %e% "weight" <= 3, "black",
                                                              ifelse(simple_edge_network_ba1 %e% "weight" > 3 & simple_edge_network_ba1 %e% "weight" <= 4, "grey30",
                                                                     ifelse(simple_edge_network_ba1 %e% "weight" > 4 & simple_edge_network_ba1 %e% "weight" <= 6, "grey50",
                                                                            ifelse(simple_edge_network_ba1 %e% "weight" > 6 & simple_edge_network_ba1 %e% "weight" <= 8, "grey70",
                                                                                   ifelse(simple_edge_network_ba1 %e% "weight" > 8 & simple_edge_network_ba1 %e% "weight" <= 10, "grey90","white"))))))


ba1_networkplot <- ggnet2(simple_edge_network_ba1, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha= 0.3, label.color = "purple", label.ba1 = 0.4, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(ba1_networkplot)
ggsave(file = "Plots/ba1_networkplot.pdf", ba1_networkplot, height = 2.25, width = 3)

## Changes in distance as compared to D614G
simple_edge_network_ba12 <- simple_edge_network_ba1

set.edge.attribute(simple_edge_network_ba12, "color", 
                   ifelse(simple_edge_network_ba12 %e% "weight2" > 1 & simple_edge_network_ba12 %e% "weight2" < 2, "orange",
                          ifelse(simple_edge_network_ba12 %e% "weight2" > 2, "red",
                                 ifelse(simple_edge_network_ba12 %e% "weight2" < -2, "blue",
                                 ifelse(simple_edge_network_ba12 %e% "weight2" < -1, "cyan","grey75")))))

D614G_ba1_networkplot <- ggnet2(simple_edge_network_ba12, mode = temp_network, label = TRUE, size = 3, label.size = 1.6, arrow.size = 0, edge.size = 1, arrow.type = "closed", edge.label = "weight3", edge.label.size = 1.5, edge.label.color = "black", edge.label.fill = NA, node.alpha = 0.5, edge.color = "color", edge.alpha = 0.3, label.color = "purple", label.ba1 = 0.5, arrow.gap = 0.01, layout.par = list(repulse.rad = 10, area = 10), color = "type", palette = c("ACE2" = "pink", "RBD" = "lightblue")) +
  ggtitle("RBD scaffold associations with RBMs") +
  theme(panel.background = element_rect(color = "grey"), legend.title = element_text(face = "bold", size = rel(rs/2)), plot.title = element_text(size = rel(rs)), legend.position = "right", legend.text = element_text(size = rel(rs/2)))
print(D614G_ba1_networkplot)
ggsave(file = "Plots/D614G_ba1_networkplot.pdf", D614G_ba1_networkplot, height = 2.25, width = 3)
# select pair_42_498, ace2 and resi 42 | rbd and resi 498
```

```{r For plotting via pymol}
paste0("select alpha_muts, alpha_rbd and resi 501")
paste0("select alpha_closer_to_ace2, alpha_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_alpha_outlier == TRUE & d614g_alpha_network > 1)$rbd_position))))
paste0("select alpha_further_from_ace2, alpha_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_alpha_outlier == TRUE & d614g_alpha_network < -1)$rbd_position))))

paste0("select beta_muts, beta_rbd and resi 417+484+501")
paste0("select beta_closer_to_ace2, beta_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_beta_outlier == TRUE & d614g_beta_network > 1)$rbd_position))))
paste0("select beta_further_from_ace2, beta_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_beta_outlier == TRUE & d614g_beta_network < -1)$rbd_position))))

paste0("select gamma_muts, gamma_rbd and resi 417+484+501")
paste0("select gamma_closer_to_ace2, gamma_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_gamma_outlier == TRUE & d614g_gamma > 0)$rbd_position))))
paste0("select gamma_further_from_ace2, gamma_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_gamma_outlier == TRUE & d614g_gamma < 0)$rbd_position))))

paste0("select delta_muts, delta_rbd and resi 452+478")
paste0("select delta_closer_to_ace2, delta_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_delta_outlier == TRUE & d614g_delta > 0)$rbd_position))))
paste0("select delta_further_from_ace2, delta_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_delta_outlier == TRUE & d614g_delta < 0)$rbd_position))))

paste0("select ba1_muts, ba1_rbd and resi 452+478")
paste0("select ba1_closer_to_ace2, ba1_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_ba1_outlier == TRUE & d614g_ba1 > 0)$rbd_position))))
paste0("select ba1_further_from_ace2, ba1_rbd and resi ", gsub(", ", "+", toString(unique(subset(network_pairs2, d614g_ba1_outlier == TRUE & d614g_ba1 < 0)$rbd_position))))

```

## N501Y mutation in Alpha, Beta, and Gamma
N501Y (RBD) mutation causes Y501 (RBD) to move much closer to D38 (ACE2).
D38 (ACE2) moves further from Q498 (RBD) and G496 (RBD)
K353 (ACE2) also moves further from Q498 (RBD), G496 (RBD), and G447 (RBD), but moves closer to Y453 (RBD)
Q498 (RBD) moves clsoer to L45 (ACE2) and N49 (ACE2)

## K417 and E484K mutations
N501Y already seems primed to make more distant changes:
K417 (RBD) moving away from D30 (ACE2) and Q493 (RBD) moving away from E35 (ACE2)

But this is further accentuated by two additional mutations in Beta and Gamma.
K417N and K417T mutations both increase the shift of the N417/T417 movement away from D30 (ACE2), but also moves it away from H34 (ACE2)
E484K mutation increases the shift of Q493 (RBD) moving away from E35 (ACE2), but K484 (RBD) also moves drastically away from K31 (ACE2), 

BA1 also has K417N, and also exhibits that shift away from D30 (ACE2) and H34 (ACE2).

## E484A in BA1
E484A mutation found alongisde Q493R. A484 moves away from K31 (ACE2) much like E484K seen for Beta/Gamma. Furthermore, K31 also moves away from L492 and F490.





## Alpha -       

Q498 (RBD) also moves away from K353 (ACE2), while moving closer to L45 (ACE2) and N49 (ACE2). 
As K353 (ACE2) moves further from Q498 (RBD), it also moves further from G496 (RBD), F497 (RBD), and G447 (RBD), while moving closer to Y453 (RBD).

## Beta - N501Y (RBD) mutation causes Y501 (RBD) to move much closer to D38 (ACE2) and slightly closer to E37 (ACE2).
[D38 (ACE2) moves further from Q498 (RBD).]
K353 (ACE2) moves further from G496 (RBD) and Q498 (RBD).


```{r Can we somehow quantitate relationships between mutations and structural impacts?}

df_alpha <- data.frame(
  from = network_pairs2_alpha_temp4$ace2_residue,
  to = network_pairs2_alpha_temp4$rbd_residue,
  distance = round(network_pairs2$dist_alpha,2), #n_outliers
  difference = round(network_pairs2$d614g_alpha,2), #n_outliers
  stringsAsFactors = FALSE
)
df_alpha$mutated <- nchar(gsub("[0-9]","",df_alpha$to)) - 1
df_alpha$variant <- "Alpha"
ggplot() + geom_quasirandom(data = df_alpha, aes(x = mutated, y = difference))


df_beta <- data.frame(
  from = network_pairs2_beta_temp4$ace2_residue,
  to = network_pairs2_beta_temp4$rbd_residue,
  distance = round(network_pairs2$dist_beta,2), #n_outliers
  difference = round(network_pairs2$d614g_beta,2), #n_outliers
  stringsAsFactors = FALSE
)
df_beta$mutated <- nchar(gsub("[0-9]","",df_beta$to)) - 1
df_beta$variant <- "Beta"
ggplot() + geom_quasirandom(data = df_beta, aes(x = mutated, y = difference))



df_gamma <- data.frame(
  from = network_pairs2_gamma_temp4$ace2_residue,
  to = network_pairs2_gamma_temp4$rbd_residue,
  distance = round(network_pairs2$dist_gamma,2), #n_outliers
  difference = round(network_pairs2$d614g_gamma,2), #n_outliers
  stringsAsFactors = FALSE
)
df_gamma$mutated <- nchar(gsub("[0-9]","",df_gamma$to)) - 1
df_gamma$variant <- "Gamma"
ggplot() + geom_quasirandom(data = df_gamma, aes(x = mutated, y = difference))



df_delta <- data.frame(
  from = network_pairs2_delta_temp4$ace2_residue,
  to = network_pairs2_delta_temp4$rbd_residue,
  distance = round(network_pairs2$dist_delta,2), #n_outliers
  difference = round(network_pairs2$d614g_delta,2), #n_outliers
  stringsAsFactors = FALSE
)
df_delta$mutated <- nchar(gsub("[0-9]","",df_delta$to)) - 1
df_delta$variant <- "Delta"
ggplot() + geom_quasirandom(data = df_delta, aes(x = mutated, y = difference))


df_ba1 <- data.frame(
  from = network_pairs2_ba1_temp4$ace2_residue,
  to = network_pairs2_ba1_temp4$rbd_residue,
  distance = round(network_pairs2$dist_ba1,2), #n_outliers
  difference = round(network_pairs2$d614g_ba1,2), #n_outliers
  stringsAsFactors = FALSE
)
df_ba1$mutated <- nchar(gsub("[0-9]","",df_ba1$to)) - 1
df_ba1$variant <- "BA1"
ggplot() + geom_quasirandom(data = df_ba1, aes(x = mutated, y = difference))

combined_var_mut_diff <- rbind(df_alpha, df_beta, df_gamma, df_delta, df_ba1)
#combined_var_mut_diff <- rbind(df_alpha[,c("variant","mutated","difference")], df_beta[,c("variant","mutated","difference")], df_gamma[,c("variant","mutated","difference")], df_delta[,c("variant","mutated","difference")], df_ba1[,c("variant","mutated","difference")])

combined_var_mut_diff$variant <- factor(combined_var_mut_diff$variant, levels = c("D614G","Alpha","Beta","Gamma","Delta","BA1"))
combined_var_mut_diff$mutated <- factor(combined_var_mut_diff$mutated)
combined_var_mut_diff_colorvector <- c("0" = "grey80", "1" = "darkred")

Distance_diffs_by_type <- ggplot() + theme(panel.grid.major = element_blank(), legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
  scale_color_manual(values = combined_var_mut_diff_colorvector) + 
  scale_y_continuous(limits = c(-0.9,0.45)) +
  labs(x = NULL, y = "Relative change in distance") + 
  geom_quasirandom(data = combined_var_mut_diff, aes(x = variant, y = difference, color = mutated), alpha = 0.6)
Distance_diffs_by_type
ggsave(file = "Plots/Distance_diffs_by_type.pdf", Distance_diffs_by_type, height = 1.8, width = 2.3)
```













```{r Subsetting for variants we've tested at contact sites}

low_kozak3 <- low_kozak2[,seq(1:6)] - low_kozak2[,1]
low_kozak3$mutant <- low_kozak2$mutant
low_kozak3_melt <- melt(low_kozak3)
low_kozak3_melt2 <- low_kozak3_melt %>% arrange(desc(value)) %>% filter(variable != "D614G")

ggplot() + geom_histogram(data = low_kozak3_melt2, aes(x = value, fill = variable)) + facet_grid(rows = vars(variable))

d614g_alpha <- network_pairs[,c("ace2_rbd","ace2_position","rbd_position","d614g_alpha")]
d614g_alpha$variable <- "Alpha"
colnames(d614g_alpha)[4] <- "distance"

d614g_beta <- network_pairs[,c("ace2_rbd","ace2_position","rbd_position","d614g_beta")]
d614g_beta$variable <- "Beta"
colnames(d614g_beta)[4] <- "distance"

d614g_gamma <- network_pairs[,c("ace2_rbd","ace2_position","rbd_position","d614g_gamma")]
d614g_gamma$variable <- "Gamma"
colnames(d614g_gamma)[4] <- "distance"

d614g_delta <- network_pairs[,c("ace2_rbd","ace2_position","rbd_position","d614g_delta")]
d614g_delta$variable <- "Delta"
colnames(d614g_delta)[4] <- "distance"

d614g_ba1 <- network_pairs[,c("ace2_rbd","ace2_position","rbd_position","d614g_ba1")]
d614g_ba1$variable <- "BA1"
colnames(d614g_ba1)[4] <- "distance"

d614g_distance_diffs <- rbind(d614g_alpha, d614g_beta, d614g_gamma, d614g_delta, d614g_ba1)

human_muts <- read.csv(file = "Keys/Human_mutants.csv", header = T, stringsAsFactors = F)
human_muts2 <- merge(human_muts, low_kozak3_melt2, by = "mutant")

human_muts3 <- merge(human_muts2, d614g_distance_diffs, by = c("ace2_position","variable"))

human_muts3_grouped <- human_muts3 %>% group_by(variable, mutant) %>% summarize(value = mean(value), distance = mean(distance), variable = unique(variable))

Distance_vs_effect <- ggplot() + theme(legend.position = "right") +
  geom_hline(yintercept = 0, linetype = 2, alpha = 0.5) + geom_vline(xintercept = 0, linetype = 2, alpha = 0.5) +
  geom_point(data = human_muts3_grouped, aes(x = distance, y = value), alpha = 0.4) +
  geom_text_repel(data = human_muts3_grouped, aes(x = distance, y = value, label = mutant, color = variable), alpha = 0.8, size = 2) +
  labs(x = "Angstroms closer", y = "Improved infection")
Distance_vs_effect
ggsave(file = "Plots/Distance_vs_effect.pdf", Distance_vs_effect, height = 2, width = 3)

human_muts3_grouped2 <- human_muts3 %>% group_by(mutant) %>% summarize(value = mean(value), distance = mean(distance))

Distance_vs_effect_mean <- ggplot() + theme(legend.position = "right") +
  geom_hline(yintercept = 0, linetype = 2, alpha = 0.5) + geom_vline(xintercept = 0, linetype = 2, alpha = 0.5) +
  geom_point(data = human_muts3_grouped, aes(x = distance, y = value, color = mutant), alpha = 0.2) +
  geom_point(data = human_muts3_grouped2, aes(x = distance, y = value, color = mutant), alpha = 0.4, shape = 4, size = 3) +
  geom_text_repel(data = human_muts3_grouped2, aes(x = distance, y = value, label = mutant, color = mutant), alpha = 0.8, size = 2) +
  labs(x = "Angstroms closer", y = "Improved infection")
Distance_vs_effect_mean
ggsave(file = "Plots/Distance_vs_effect_mean.pdf", Distance_vs_effect_mean, height = 2, width = 3)

```

```{r D38H specifically}
d38 <- human_muts3 %>% filter(ace2_position == 38)
ggplot() + geom_point(data = d38, aes(x = distance, y = value, color = variable)) +
  geom_text_repel(data = d38, aes(x = distance, y = value, label = rbd_position))

d38_ba1 <- d38 %>% filter(variable == "BA1") %>% group_by(rbd_position) %>% summarize(value = mean(value), distance = mean(distance))
d38_not_ba1 <- d38 %>% filter(variable != "BA1") %>% group_by(rbd_position) %>% summarize(value = mean(value), distance = mean(distance))
d38_comb <- merge(d38_ba1, d38_not_ba1, by = "rbd_position")

ggplot() + geom_vline(xintercept = 0) +
  geom_segment(data = d38_comb, aes(x = distance.x, y = value.x, xend = distance.y, yend = value.y), linetype = 2, alpha = 0.4) +
  geom_point(data = d38_not_ba1, aes(x = distance, y = value)) +
  geom_text_repel(data = d38_not_ba1, aes(x = distance, y = value, label = rbd_position)) +
  geom_point(data = d38_ba1, aes(x = distance, y = value), color = "purple") +
  geom_text_repel(data = d38_ba1, aes(x = distance, y = value, label = rbd_position), color = "purple")


```


```{r Other residues specifically}
e35 <- human_muts3 %>% filter(ace2_position == 35)
ggplot() + geom_point(data = e35, aes(x = distance, y = value, color = variable)) +
  geom_text_repel(data = e35, aes(x = distance, y = value, label = rbd_position))


k31 <- human_muts3 %>% filter(ace2_position == 31)
ggplot() + geom_point(data = k31, aes(x = distance, y = value, color = variable)) +
  geom_text_repel(data = k31, aes(x = distance, y = value, label = rbd_position))


k353 <- human_muts3 %>% filter(ace2_position == 353)
ggplot() + geom_point(data = k353, aes(x = distance, y = value, color = variable)) +
  geom_text_repel(data = k353, aes(x = distance, y = value, label = rbd_position))
```

```{r Orthologs}
# select ace2_intxn, d614g_ace2 and resi 30+31+38+353+45+35+34+37+42+21+24+393+49+83+39+27+28+357+41+82"

ortholog_seqs <- read.csv(file = "Keys/Ortholog_seqs.csv", header = T, stringsAsFactors = F)

ortholog_seqs$I21 <- substr(ortholog_seqs$sequence,21,21)
ortholog_seqs$Q24 <- substr(ortholog_seqs$sequence,24,24)
ortholog_seqs$F28 <- substr(ortholog_seqs$sequence,28,28)
ortholog_seqs$D30 <- substr(ortholog_seqs$sequence,30,30)
ortholog_seqs$K31 <- substr(ortholog_seqs$sequence,31,31)
ortholog_seqs$H34 <- substr(ortholog_seqs$sequence,34,34)
ortholog_seqs$E35 <- substr(ortholog_seqs$sequence,35,35)
ortholog_seqs$E38 <- substr(ortholog_seqs$sequence,37,37)
ortholog_seqs$D38 <- substr(ortholog_seqs$sequence,38,38)
ortholog_seqs$Y41 <- substr(ortholog_seqs$sequence,41,41)
ortholog_seqs$Q42 <- substr(ortholog_seqs$sequence,42,42)
ortholog_seqs$L45 <- substr(ortholog_seqs$sequence,45,45)
ortholog_seqs$N49 <- substr(ortholog_seqs$sequence,49,49)
ortholog_seqs$M82 <- substr(ortholog_seqs$sequence,82,82)
ortholog_seqs$Y83 <- substr(ortholog_seqs$sequence,83,83)
ortholog_seqs$K353 <- substr(ortholog_seqs$sequence,353,353)
ortholog_seqs$R357 <- substr(ortholog_seqs$sequence,357,357)
ortholog_seqs$R393 <- substr(ortholog_seqs$sequence,393,393)

## Some sites exhibited no variability in this dataset. I am removed these from the analysis.
ortholog_seqs_variable_sites <- ortholog_seqs[,!(colnames(ortholog_seqs) %in% c("F28", "E38", "L45", "R357", "R393"))]
ortholog_seqs_variable_sites_residues <- ortholog_seqs_variable_sites[,c(1,7:ncol(ortholog_seqs_variable_sites))]

ortholog_variability_df <- data.frame("residue" = colnames(ortholog_seqs_variable_sites_residues)[2:ncol(ortholog_seqs_variable_sites_residues)])

for(x in 2:ncol(ortholog_seqs_variable_sites_residues)){
  unique(ortholog_seqs_variable_sites_residues[,2])
}

## Write those variable interface sites as strings so they can be considered as part of the distance matrices
ortholog_seqs_variable_sites_residues$concat <- ""

for(x in 1:nrow(ortholog_seqs_variable_sites_residues)){ortholog_seqs_variable_sites_residues$concat[x] <- gsub("[^A-Z]","",toString(ortholog_seqs_variable_sites_residues[x,2:14]))}


write.csv(file = "Output_tables/Orthologs.csv", ortholog_seqs_variable_sites_residues[,c("ortholog","concat")], col.names = F, row.names = F)
```

```{r Ortholog Grantham PCA}
ortholog_identity <- read.csv(file = "Data/Grantham/Ortholog_identity_matrix.csv", header= T, stringsAsFactors = F)
rownames(ortholog_identity) <- ortholog_identity$X
ortholog_identity2 <- ortholog_identity[,2:ncol(ortholog_identity)]
ortholog_identity_pca = prcomp(ortholog_identity2, scale. = TRUE)

ortholog_identity_scores = as.data.frame(ortholog_identity_pca$x)
ortholog_identity_labels <- read.csv(file = "Data/Grantham/Updated_ortholog_identity_file.csv", header= T, stringsAsFactors = F)
ortholog_identity_scores$label <- ortholog_identity_labels$Names

ortholog_identity_PCA_plot <- ggplot() + 
  xlab(paste("Principal component 1\n(",round(summary(ortholog_identity_pca)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Principal component 2\n(",round(summary(ortholog_identity_pca)$importance[5] * 100,0),"% of variance)",sep="")) + 
  geom_point(data = ortholog_identity_scores, aes(x = PC1, y = PC2), alpha = 0.5) +
  geom_text_repel(data = ortholog_identity_scores, aes(x = PC1, y = PC2, label = label), color = "red", size = 2, segment.color = "orange")
ortholog_identity_PCA_plot
ggsave(file = "Plots/Ortholog_identity_PCA_plot.pdf", ortholog_identity_PCA_plot, height = 2.25, width = 2.5)
```

```{r Showing ortholog interface residues as an identity matrix heatmap}
ortholog_identity3 <- ortholog_identity2
colnames(ortholog_identity3) <- ortholog_identity_labels$Names

ortholog_identity3_matrix <- ortholog_identity3
colnames(ortholog_identity3_matrix) <- ortholog_identity_labels$Names
rownames(ortholog_identity3_matrix) <- ortholog_identity_labels$Names

ortholog_identity3_matrix_hclust <- hclust(as.dist(ortholog_identity3_matrix, diag = TRUE))

ortholog_identity3_matrix_hclust_order <- ortholog_identity3_matrix_hclust$labels[ortholog_identity3_matrix_hclust$order]

ortholog_identity3_matrix2 <- ortholog_identity3_matrix
ortholog_identity3_matrix2$species <- rownames(ortholog_identity3_matrix)

ortholog_identity3_matrix2_melted <- melt(ortholog_identity3_matrix2, id = "species")
ortholog_identity3_matrix2_melted$variable <- as.character(ortholog_identity3_matrix2_melted$variable)

ortholog_identity3_matrix2_melted$species <- factor(ortholog_identity3_matrix2_melted$species, levels = ortholog_identity3_matrix_hclust_order)
ortholog_identity3_matrix2_melted$variable <- factor(ortholog_identity3_matrix2_melted$variable, levels = ortholog_identity3_matrix_hclust_order)

#ortholog_identity3$label <- ortholog_identity_labels$Names

#ortholog_identity3_melt <- melt(ortholog_identity3, id = "label")
#ortholog_identity3_melt <- ortholog_identity3_melt[c(seq(14-12,14),seq(28-11,28),seq(42-10,42),seq(56-9,56),seq(70-8,70),seq(84-7,84),seq(98-6,98),seq(112-5,112),seq(112-4,112),seq(126-3,126),seq(138-2,140),seq(153,154)),]

#ortholog_identity3_melt$label <- factor(ortholog_identity3_melt$label, levels = ortholog_identity_labels$Names)
#ortholog_identity3_melt$variable <- factor(ortholog_identity3_melt$variable, levels = ortholog_identity_labels$Names)

Ortholog_identity_heatmap_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = NULL, y = NULL) +
  scale_fill_gradient(low = "white", high = "red") +
  scale_x_discrete(position = "top") +
  geom_tile(data = ortholog_identity3_matrix2_melted, aes(x = species, y = variable, fill = value))  +
  geom_text(data = ortholog_identity3_matrix2_melted, aes(x = species, y = variable, label = value), angle = 45, size = 2)
Ortholog_identity_heatmap_plot
ggsave(file = "Plots/Ortholog_identity_heatmap_plot.pdf", Ortholog_identity_heatmap_plot, height = 3, width = 3)

```

### Showing residues of sequence difference between orthologs
```{r Difference between orthologs}

ortholog_seqs <- read.csv(file = "Data/Grantham/Updated_ortholog_identity_file.csv", header = T, stringsAsFactors = F)

human_ace2_seq <- ortholog_seqs$Sequences[1]

ortholog_residue_diff_frame <- data.frame("position" = seq(1:nchar(human_ace2_seq)), "residues" = "", "diffnum" = 0)

for(x in 1:nrow(ortholog_residue_diff_frame)){
  temp_vector = c()
  for(y in 1:nrow(ortholog_seqs)){
    temp_vector <- c(temp_vector,substr(ortholog_seqs$Sequences[y],x,x))
  }
  ortholog_residue_diff_frame$residues[x] <- gsub(", ","",toString(unique(temp_vector)))
  ortholog_residue_diff_frame$diffnum[x] <- nchar(ortholog_residue_diff_frame$residues[x])
}

```


```{r}
combined_ortholog_variant_unmelt <- rbind(
(ortholog1 %>% mutate(sample = ortholog))[,!colnames(ortholog1) %in% "ortholog"],
(low_kozak1 %>% mutate(sample = mutant))[,!colnames(low_kozak1) %in% "mutant"])

for(x in 1:nrow(combined_ortholog_variant_unmelt)){
  for(y in 1:(ncol(combined_ortholog_variant_unmelt)-1)){
    combined_ortholog_variant_unmelt[x,y] <- log10(combined_ortholog_variant_unmelt[x,y])
  }
}
```

```{r Looking at Geomean infectivities across the dataset}

combined_ortholog_variant_melt <- melt(combined_ortholog_variant_unmelt, id = "sample")

combined_ortholog_variant_melt_summary <- combined_ortholog_variant_melt %>% group_by(sample) %>% summarize(mean = mean(value)) %>% arrange(mean)

combined_ortholog_variant_melt_summary$sample <- factor(combined_ortholog_variant_melt_summary$sample, levels = combined_ortholog_variant_melt_summary$sample)

combined_ortholog_variant_melt$sample <- factor(combined_ortholog_variant_melt$sample, levels = combined_ortholog_variant_melt_summary$sample)

Infection_step_plot <- ggplot() + theme(panel.grid.major.x = element_blank(), axis.text.x = element_text(angle = -90, hjust = 0, vjust = 0.5)) + 
  labs(x = NULL, y = "Log10 enrichment") +
  geom_point(data = combined_ortholog_variant_melt, aes(x = sample, y = value, color = variable), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_melt_summary, aes(x = sample, y = mean), size = 6, shape = 95) +
  NULL
Infection_step_plot
ggsave(file = "Plots/Infection_step_plot.pdf", Infection_step_plot, height = 1.75, width = 5)

```






## Below is an analysis method for comparing the differences in infection between the various SARS-CoV-2 RBD variants

```{r}
combined_ortholog_variant_unmelt <- rbind(
(ortholog1 %>% mutate(sample = ortholog))[,!colnames(ortholog1) %in% "ortholog"],
(low_kozak1 %>% mutate(sample = mutant))[,!colnames(low_kozak1) %in% "mutant"])

for(x in 1:nrow(combined_ortholog_variant_unmelt)){
  for(y in 1:(ncol(combined_ortholog_variant_unmelt)-1)){
    combined_ortholog_variant_unmelt[x,y] <- log10(combined_ortholog_variant_unmelt[x,y])
  }
}
```


```{r}

d614g_alpha_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"D614G"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"D614G"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Alpha"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Alpha"])))
lm_d614g_alpha <- lm(d614g_alpha_df[,"virus2"] ~ d614g_alpha_df[,"virus1"])
D614G_Alpha_scatterplot <- ggplot() +
  geom_abline(slope = lm_d614g_alpha$coefficients[2], intercept = lm_d614g_alpha$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Alpha), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = D614G, y = Alpha), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("M.musculus","R.shameli","R.sinicus_200","R.landeri","R.ferrumequinum","R.sinicus_215")), aes(x = D614G, y = Alpha, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/D614G_Alpha_scatterplot.pdf", D614G_Alpha_scatterplot, height = 1.5, width = 2.25)

d614g_beta_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"D614G"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"D614G"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Beta"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Beta"])))
lm_d614g_beta <- lm(d614g_beta_df[,"virus2"] ~ d614g_beta_df[,"virus1"])
D614G_Beta_scatterplot <- ggplot() +
  geom_abline(slope = lm_d614g_beta$coefficients[2], intercept = lm_d614g_beta$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Beta), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = D614G, y = Beta), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Beta, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/D614G_Beta_scatterplot.pdf", D614G_Beta_scatterplot, height = 2, width = 2.25)

d614g_delta_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"D614G"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"D614G"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Delta"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Delta"])))
lm_d614g_delta <- lm(d614g_delta_df[,"virus2"] ~ d614g_delta_df[,"virus1"])
D614G_Delta_scatterplot <- ggplot() +
  geom_abline(slope = lm_d614g_delta$coefficients[2], intercept = lm_d614g_delta$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Delta), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = D614G, y = Delta), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Delta, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/D614G_Delta_scatterplot.pdf", D614G_Delta_scatterplot, height = 2, width = 2.25)

d614g_ba1_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"D614G"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"D614G"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"BA1"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"BA1"])))
lm_d614g_ba1 <- lm(d614g_ba1_df[,"virus2"] ~ d614g_ba1_df[,"virus1"])
D614G_BA1_scatterplot <- ggplot() +
  geom_abline(slope = lm_d614g_ba1$coefficients[2], intercept = lm_d614g_ba1$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = BA1), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = D614G, y = BA1), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = BA1, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/D614G_BA1_scatterplot.pdf", D614G_BA1_scatterplot, height = 2, width = 2.25)


alpha_beta_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Alpha"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Alpha"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Beta"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Beta"])))
lm_alpha_beta <- lm(alpha_beta_df[,"virus2"] ~ alpha_beta_df[,"virus1"])
Alpha_Beta_scatterplot <- ggplot() +
  geom_abline(slope = lm_alpha_beta$coefficients[2], intercept = lm_alpha_beta$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Alpha, y = Beta), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = Alpha, y = Beta), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = Alpha, y = Beta, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/Alpha_Beta_scatterplot.pdf", Alpha_Beta_scatterplot, height = 2, width = 2.25)


alpha_gamma_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Alpha"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Alpha"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Gamma"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Gamma"])))
lm_alpha_gamma <- lm(alpha_gamma_df[,"virus2"] ~ alpha_gamma_df[,"virus1"])
Alpha_Gamma_scatterplot <- ggplot() +
  geom_abline(slope = lm_alpha_gamma$coefficients[2], intercept = lm_alpha_gamma$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Alpha, y = Gamma), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = Alpha, y = Gamma), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = Alpha, y = Gamma, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/Alpha_Gamma_scatterplot.pdf", Alpha_Gamma_scatterplot, height = 2, width = 2.25)







gamma_beta_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Gamma"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Gamma"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Beta"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Beta"])))
lm_gamma_beta <- lm(gamma_beta_df[,"virus2"] ~ gamma_beta_df[,"virus1"])

Gamma_Beta_scatterplot <- ggplot() +
  geom_abline(slope = lm_gamma_beta$coefficients[2], intercept = lm_gamma_beta$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Gamma, y = Beta), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = Gamma, y = Beta), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = Gamma, y = Beta, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/Gamma_Beta_scatterplot.pdf", Gamma_Beta_scatterplot, height = 2, width = 2.25)

gamma_delta_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Gamma"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Gamma"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Delta"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Delta"])))
lm_gamma_delta <- lm(gamma_delta_df[,"virus2"] ~ gamma_delta_df[,"virus1"])
Gamma_Delta_scatterplot <- ggplot() +
  geom_abline(slope = lm_gamma_delta$coefficients[2], intercept = lm_gamma_delta$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Gamma, y = Delta), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = Gamma, y = Delta), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = Gamma, y = Delta, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/Gamma_Delta_scatterplot.pdf", Gamma_Delta_scatterplot, height = 2, width = 2.25)

gamma_ba1_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Gamma"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Gamma"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"BA1"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"BA1"])))
lm_gamma_ba1 <- lm(gamma_ba1_df[,"virus2"] ~ gamma_ba1_df[,"virus1"])
Gamma_BA1_scatterplot <- ggplot() +
  geom_abline(slope = lm_gamma_ba1$coefficients[2], intercept = lm_gamma_ba1$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Gamma, y = BA1), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = Gamma, y = BA1), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = Gamma, y = BA1, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/Gamma_BA1_scatterplot.pdf", Gamma_BA1_scatterplot, height = 2, width = 2.25)



beta_ba1_df <- data.frame(virus1 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"Beta"]),
                                 mean(((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"Beta"]))),
                      virus2 = c(mean((combined_ortholog_variant_unmelt %>% filter(sample == "H.sapiens"))[,"BA1"]),
                                 mean((combined_ortholog_variant_unmelt %>% filter(sample == "dEcto"))[,"BA1"])))
lm_beta_ba1 <- lm(beta_ba1_df[,"virus2"] ~ beta_ba1_df[,"virus1"])
Beta_BA1_scatterplot <- ggplot() +
  geom_abline(slope = lm_beta_ba1$coefficients[2], intercept = lm_beta_ba1$coefficients[1], alpha = 0.1, size = 4) +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Beta, y = BA1), alpha = 0.4) +
  geom_point(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("H.sapiens","dEcto","Control")), aes(x = Beta, y = BA1), color = "blue", size = 2, alpha = 0.4) +
  geom_text_repel(data = combined_ortholog_variant_unmelt %>% filter(sample %in% c("R.sinicus_215","R.pearsonii","M.javanica","K31D","M.musculus")), aes(x = Beta, y = BA1, label = sample), color = "red", size = 1.5, segment.color = "orange", segment.alpha = 0.4, force = 0.5) 
ggsave(file = "Plots/Beta_BA1_scatterplot.pdf", Beta_BA1_scatterplot, height = 1.5, width = 2.25)





ggplot() + 
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Alpha)) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Alpha, label = sample), color = "red") 

ggplot() + 
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Beta)) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Beta, label = sample), color = "red") 

ggplot() + 
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Gamma)) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Gamma, label = sample), color = "red") 

ggplot() + 
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Delta)) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = Delta, label = sample), color = "red") 

ggplot() + 
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = BA1)) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = D614G, y = BA1, label = sample), color = "red") 

ggplot() +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Delta, y = BA1)) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = Delta, y = BA1, label = sample), color = "red") 

ggplot() +
  geom_point(data = combined_ortholog_variant_unmelt, aes(x = Beta, y = Gamma)) +
  geom_text_repel(data = combined_ortholog_variant_unmelt, aes(x = Beta, y = Gamma, label = sample), color = "red") 



## Make a dataframe of pairwise Paerons's correlations
cor(combined_ortholog_variant_unmelt$Beta, combined_ortholog_variant_unmelt$Gamma, method = "pearson")^2

colnames(combined_ortholog_variant_unmelt)
cor_matrix_names <- c("Alpha","Beta","Delta","Gamma","BA1","D614G")

cor_matrix_df <- data.frame(index = seq(1,length(cor_matrix_names) * length(cor_matrix_names)))
cor_matrix_df$virus1 <- rep(cor_matrix_names, length(cor_matrix_names))
cor_matrix_df$virus2 <- rep(cor_matrix_names, each = length(cor_matrix_names))
cor_matrix_df$pearson_rsquared <- 0

for(x in 1:nrow(cor_matrix_df)){
    virus1 <- cor_matrix_df$virus1[x]
    virus2 <- cor_matrix_df$virus2[x]
    cor_matrix_df$pearson_rsquared[x] <- cor(combined_ortholog_variant_unmelt[,virus1], combined_ortholog_variant_unmelt[,virus2], method = "pearson")^2
}

cor_matrix_df$virus1 <- factor(cor_matrix_df$virus1, levels = c("D614G", "Alpha","Beta","Gamma","Delta","BA1"))
cor_matrix_df$virus2 <- factor(cor_matrix_df$virus2, levels = c("D614G", "Alpha","Beta","Gamma","Delta","BA1"))

Virus_pairwise_correlations <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "right", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = NULL, y = NULL) +
  scale_fill_gradient(low  = "white", high = "purple", limits = c(0,1)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = cor_matrix_df, aes(x = virus1, y = virus2, fill = pearson_rsquared)) +
  geom_text(data = cor_matrix_df, aes(x = virus1, y = virus2, label = round(pearson_rsquared, 2)), angle= 45, size = 2)
Virus_pairwise_correlations
ggsave(file = "Plots/Virus_pairwise_correlations.pdf", Virus_pairwise_correlations, height = 1.9, width = 2.8)

```

```{r}

## Try PCA on variant patterns 

for_variant_infection_pca <- data.frame(t(combined_ortholog_variant_unmelt[,1:6]))
colnames(for_variant_infection_pca) <- combined_ortholog_variant_unmelt$sample
for_variant_infection_pca <- for_variant_infection_pca[,17:29]
pca_for_variant_infection = prcomp(for_variant_infection_pca, scale. = TRUE)
pca_variant_infection_scores = as.data.frame(pca_for_variant_infection$x)
summary(pca_for_variant_infection)$rotation
pca_variant_infection_scores$labels <- rownames(pca_variant_infection_scores)

variant_infection_pca <- ggplot() + 
  xlab(paste("Principal component 1\n(",round(summary(pca_for_variant_infection)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Principal Component 2\n(",round(summary(pca_for_variant_infection)$importance[5] * 100,0),"% of variance)",sep="")) + 
  geom_text_repel(data = pca_variant_infection_scores, aes(x = PC1, y = PC2, label = labels), segment.color = 'grey90', point.padding = 0.05, size = 2, segment.alpha = 0.5, color = "red") +
  geom_point(data = pca_variant_infection_scores, aes(x = PC1, y = PC2), size = 1) +
  NULL
variant_infection_pca
ggsave(file = "Plots/variant_infection_pca.pdf", variant_infection_pca, height = 1.8, width = 1.65) #height = 1.6, width = 1.75)


## Try PCA on ortholog usage

ortholog_infection_for_pca <- combined_ortholog_variant_unmelt[,1:6]
pca_ortholog_infection = prcomp(ortholog_infection_for_pca, scale. = TRUE)
pca_ortholog_infection_scores = as.data.frame(pca_ortholog_infection$x)

summary(pca_ortholog_infection)$rotation
pca_ortholog_infection_scores$labels <- combined_ortholog_variant_unmelt$sample

ortholog_infection_pca <- ggplot() + 
  scale_x_continuous(limits = c(-2.5,4)) +
  xlab(paste("Component 1\n(",round(summary(pca_ortholog_infection)$importance[2] * 100,0),"% of variance)",sep="")) + 
  ylab(paste("Component 2\n(",round(summary(pca_ortholog_infection)$importance[5] * 100,0),"% of variance)",sep="")) + 
  geom_text_repel(data = pca_ortholog_infection_scores %>% filter(PC1 > 0 | PC2 > 0.9 | PC2 < -0.9), aes(x = PC1, y = PC2, label = labels), segment.color = 'grey90', point.padding = 0.05, size = 1.5, segment.alpha = 0.5, color = "red") +
  geom_point(data = pca_ortholog_infection_scores, aes(x = PC1, y = PC2), size = 1, alpha = 0.5) +
  NULL
ortholog_infection_pca
ggsave(file = "Plots/ortholog_infection_pca.pdf", ortholog_infection_pca, height = 1.6, width = 2.75)

```





```{r Make a more generalizable function for comparing variants}
## For troubleshooting the function
mini_frame <- combined_ortholog_variant_unmelt[,c("Gamma","BA1","sample")]

virus_infection_comparison <- function(mini_frame){
  
temp_df <- data.frame(virus1 = c(mean((mini_frame %>% filter(sample == "H.sapiens"))[,1]), mean((mini_frame %>% filter(sample == "dEcto"))[,1])), virus2 = c(mean((mini_frame %>% filter(sample == "H.sapiens"))[,2]), mean((mini_frame %>% filter(sample == "dEcto"))[,2])))

lm1 <- lm(temp_df[,"virus2"] ~ temp_df[,"virus1"])

ggplot() + 
  geom_point(data = mini_frame, aes(x = get(colnames(mini_frame)[1]), y = get(colnames(mini_frame)[2]))) +
  geom_text_repel(data = mini_frame, aes(x = get(colnames(mini_frame)[1]), y = get(colnames(mini_frame)[2]), label = sample), color = "red") +
  geom_abline(slope = lm1$coefficients[2], intercept = lm1$coefficients[1])

lm1_res <- data.frame(sample = mini_frame$sample ,"residuals" = lm(mini_frame[,colnames(mini_frame)[2]] ~ mini_frame[,colnames(mini_frame)[1]])$residuals)

lm1_res_sd <- c(mean(lm1_res$residuals) - sd(lm1_res$residuals), mean(lm1_res$residuals) + sd(lm1_res$residuals))

ggplot() + geom_histogram(data = lm1_res, aes(x = residuals), binwidth = 0.2) + geom_vline(xintercept = lm1_res_sd)

all_samples <- lm1_res %>% mutate(name1 = colnames(mini_frame)[1], name2 = colnames(mini_frame)[2], category = "all")

outliers <- lm1_res %>% filter(residuals < lm1_res_sd[1] | residuals > lm1_res_sd[2]) %>% mutate(name1 = colnames(mini_frame)[1], name2 = colnames(mini_frame)[2], category = "outlier")

return_samples <- rbind(all_samples, outliers)
return_samples2 <- rbind(return_samples, c("correlation", cor(mini_frame[,1], mini_frame[,2]), colnames(mini_frame)[1], colnames(mini_frame)[2], "correlation"))

return(return_samples2)
}

virus_list <- c("D614G", "Alpha","Beta","Gamma","Delta","BA1")
final_df <- data.frame("sample" = NA, "residuals" = NA, "name1" = NA, "name2" = NA, "category" = NA)
for(x in 1:length(virus_list)){
  for(y in 1:length(virus_list)){
    if(x != y){
    tempy <- virus_infection_comparison(combined_ortholog_variant_unmelt[,c(virus_list[x],virus_list[y],"sample")])
    final_df <- rbind(final_df, tempy)
    }
  }
}

final_df$name1 <- factor(final_df$name1, levels = c("D614G", "Alpha","Beta","Gamma","Delta","BA1"))
final_df$name2 <- factor(final_df$name2, levels = c("D614G", "Alpha","Beta","Gamma","Delta","BA1"))

all_residuals <- final_df %>% filter(category == "all") %>% mutate(residuals = as.numeric(residuals))
correlations <- final_df %>% filter(category == "correlation") %>% mutate(residuals = as.numeric(residuals))
outlier_residuals <- final_df %>% filter(category == "outlier") %>% mutate(residuals = as.numeric(residuals))
```

```{r Make a more generalizable function for comparing variants}
## Figuring out which residual values were the biggest

all_residuals2 <- all_residuals %>% arrange(desc(residuals))
all_residuals2$sample <- factor(all_residuals2$sample, levels = c("R.ferrumequinum","R.shameli","R.sinicus_215","R.landeri","M.musculus","R.sinicus_200","R.pearsonii","S.scrofa","M.javanica"))

Residuals_histogram <- ggplot() + theme(strip.text.y = element_text(angle = 0)) + 
  scale_x_continuous(breaks = c(-1,0,1)) + scale_y_continuous(breaks = c(0,5,10,100,200,300,400)) + 
  geom_histogram(data = all_residuals2, aes(x = residuals), binwidth = .1, color = "black", fill = "grey80") + facet_grid(rows = vars(sample), scales = "free_y")
ggsave(file = "Plots/Residuals_histogram.pdf", Residuals_histogram, height = 6, width = 2.5)

residuals_sd <- all_residuals %>% group_by(sample) %>% summarize(sd_residual = sd(residuals)) %>% arrange(desc(sd_residual))

ggplot() + geom_histogram(data = residuals_sd, aes(x = sd_residual), binwidth = .05, fill = "grey80", color = "black")

sd_residuals_high <- residuals_sd %>% filter(sd_residual >= 0.3)

```

```{r Make a more generalizable function for comparing variants}

## Ones where N501Y seems to be driving the effect based on pattern

## Where Beta, Gamma, and BA1 do much better

mouse_residuals <- all_residuals %>% filter(sample == "M.musculus")
mouse_residuals <- mouse_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
mouse_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "M.musculus", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = mouse_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = mouse_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
mouse_residuals_plot
ggsave(file = "Plots/mouse_residuals_plot.pdf", mouse_residuals_plot, height = 1.9, width = 2)

sinicus_215_residuals <- all_residuals %>% filter(sample == "R.sinicus_215")
sinicus_215_residuals <- sinicus_215_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
sinicus_215_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "R.sinicus_215", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = sinicus_215_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = sinicus_215_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
sinicus_215_residuals_plot
ggsave(file = "Plots/sinicus_215_residuals_plot.pdf", sinicus_215_residuals_plot, height = 1.9, width = 2)

sinicus_200_residuals <- all_residuals %>% filter(sample == "R.sinicus_200")
sinicus_200_residuals <- sinicus_200_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
sinicus_200_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "R.sinicus_200", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = sinicus_200_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = sinicus_200_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
sinicus_200_residuals_plot
ggsave(file = "Plots/sinicus_200_residuals_plot.pdf", sinicus_200_residuals_plot, height = 1.9, width = 2)












## One's where D614G does better than many of the other variants

shameli_residuals <- all_residuals %>% filter(sample == "R.shameli")
shameli_residuals <- shameli_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
shameli_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "R.shameli", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = shameli_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = shameli_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
shameli_residuals_plot
ggsave(file = "Plots/shameli_residuals_plot.pdf", shameli_residuals_plot, height = 1.9, width = 2)


## One's where the variants do better than D614G

landeri_residuals <- all_residuals %>% filter(sample == "R.landeri")
landeri_residuals <- landeri_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
landeri_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "R.landeri", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = landeri_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = landeri_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
landeri_residuals_plot
ggsave(file = "Plots/landeri_residuals_plot.pdf", landeri_residuals_plot, height = 1.9, width = 2)




## Where BA1 does much better

pearsonii_residuals <- all_residuals %>% filter(sample == "R.pearsonii")
pearsonii_residuals <- pearsonii_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
pearsonii_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "R.pearsonii", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = pearsonii_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = pearsonii_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
pearsonii_residuals_plot
ggsave(file = "Plots/pearsonii_residuals_plot.pdf", pearsonii_residuals_plot, height = 1.9, width = 2)


# Where BA1 does much worse

pangolin_residuals <- all_residuals %>% filter(sample == "M.javanica")
pangolin_residuals <- pangolin_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
pangolin_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "M.javanica", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = pangolin_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = pangolin_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
pangolin_residuals_plot
ggsave(file = "Plots/pangolin_residuals_plot.pdf", pangolin_residuals_plot, height = 1.9, width = 2)





alcyone_residuals <- all_residuals %>% filter(sample == "R.alcyone")
alcyone_residuals <- alcyone_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
alcyone_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "R.alcyone", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = alcyone_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = alcyone_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
alcyone_residuals_plot
ggsave(file = "Plots/alcyone_residuals_plot.pdf", alcyone_residuals_plot, height = 1.9, width = 2)







pig_residuals <- all_residuals %>% filter(sample == "S.scrofa")
pig_residuals <- pig_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
pig_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "S.scrofa", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = pig_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = pig_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
pig_residuals_plot
ggsave(file = "Plots/pig_residuals_plot.pdf", pig_residuals_plot, height = 1.9, width = 2)


```


```{r Make a more generalizable function for comparing variants}

sinicus_215_residuals <- all_residuals %>% filter(sample == "R.sinicus_215")
Sinicus_215_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 45, hjust = 0.2, vjust = 0)) + 
  labs(x = "R.sinicus_215", y = "Residuals (Blue is worse\nthan the other virus)") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = sinicus_215_residuals, aes(x = name1, y = name2, fill = residuals))
ggsave(file = "Plots/Sinicus_215_residuals_plot.pdf", Sinicus_215_residuals_plot, height = 2, width = 2.3)

landeri_residuals <- all_residuals %>% filter(sample == "R.landeri")
landeri_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 45, hjust = 0.2, vjust = 0)) + 
  labs(x = "R.landeri", y = "Residuals (Blue is worse\nthan the other virus)") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = landeri_residuals, aes(x = name1, y = name2, fill = residuals))
ggsave(file = "Plots/Landeri_residuals_plot.pdf", landeri_residuals_plot, height = 1.7, width = 2)

mouse_residuals <- all_residuals %>% filter(sample == "M.musculus")
Mouse_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 45, hjust = 0.2, vjust = 0)) + 
  labs(x = "M.musculus", y = "Residuals (Blue is worse\nthan the other virus)") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = mouse_residuals, aes(x = name1, y = name2, fill = residuals))
ggsave(file = "Plots/Mouse_residuals_plot.pdf", Mouse_residuals_plot, height = 2, width = 2.3)

sinicus_200_residuals <- all_residuals %>% filter(sample == "R.sinicus_200")
Sinicus_200_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 45, hjust = 0.2, vjust = 0)) + 
  labs(x = "R.sinicus_200", y = "Residuals (Blue is worse\nthan the other virus)") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = sinicus_200_residuals, aes(x = name1, y = name2, fill = residuals))
ggsave(file = "Plots/Sinicus_200_residuals_plot.pdf", Sinicus_200_residuals_plot, height = 1.7, width = 2)

pearsonii_residuals <- all_residuals %>% filter(sample == "R.pearsonii")
Pearsonii_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 45, hjust = 0.2, vjust = 0)) + 
  labs(x = "R.pearsonii", y = "Residuals (Blue is worse\nthan the other virus)") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = pearsonii_residuals, aes(x = name1, y = name2, fill = residuals))
ggsave(file = "Plots/Pearsonii_residuals_plot.pdf", Pearsonii_residuals_plot, height = 2.3, width = 2.8)

pig_residuals <- all_residuals %>% filter(sample == "S.scrofa")
Pig_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 45, hjust = 0.2, vjust = 0)) + 
  labs(x = "S.scrofa", y = "Residuals (Blue is worse\nthan the other virus)") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = pig_residuals, aes(x = name1, y = name2, fill = residuals))
ggsave(file = "Plots/Pig_residuals_plot.pdf", Pig_residuals_plot, height = 2.3, width = 2.8)

pangolin_residuals <- all_residuals %>% filter(sample == "M.javanica")
Pangolin_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 45, hjust = 0.2, vjust = 0)) + 
  labs(x = "M.javanica", y = "Residuals (Blue is worse\nthan the other virus)") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = pangolin_residuals, aes(x = name1, y = name2, fill = residuals))
ggsave(file = "Plots/Pangolin_residuals_plot.pdf", Pangolin_residuals_plot, height = 1.7, width = 2)





ferrum_residuals <- all_residuals %>% filter(sample == "R.ferrumequinum")
ferrum_residuals <- ferrum_residuals[c(seq(1,5),seq(7,10),seq(13,15),seq(19,20),25),]
ferrum_residuals_plot <- ggplot() + 
  theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
  legend.position = "none", axis.text.x.top = element_text(angle = 90, hjust = 0, vjust = 0)) + 
  labs(x = "R.ferrumequinum", y = "Residuals") +
  scale_fill_gradient2(low  = "blue", mid = "white", high = "red", limits = c(-1.4,1.4)) +
  scale_x_discrete(position = "top") +
  geom_tile(data = ferrum_residuals, aes(x = name1, y = name2, fill = residuals))  +
  geom_text(data = ferrum_residuals, aes(x = name1, y = name2, label = round(residuals,1)), angle = 45, size = 2)
ferrum_residuals_plot
ggsave(file = "Plots/Ferrum_residuals_plot.pdf", ferrum_residuals_plot, height = 1.9, width = 2)

```

```{r Most variable minimal dataset}
most_variable_orthologs <- c("R.ferrumequinum","M.musculus","R.shameli","R.sinicus_215","R.sinicus_200","R.landeri","R.pearsonii","M.javanica")
ortholog_most_variable <- ortholog2 %>% filter(ortholog %in% most_variable_orthologs)
ortholog_most_variable_matrix <- as.matrix(ortholog_most_variable[,1:6])
rownames(ortholog_most_variable_matrix) <- ortholog_most_variable$ortholog #paste("r_",rbd_parsed_df$V1,sep="")

ortholog_most_variable_matrix_dist <- dist(ortholog_most_variable_matrix, method = 'euclidean')
ortholog_most_variable_matrix_hclust <- hclust(ortholog_most_variable_matrix_dist)

ortholog_most_variable_matrix_dist2 <- dist(t(ortholog_most_variable_matrix), method = 'euclidean')
ortholog_most_variable_matrix_hclust2 <- hclust(ortholog_most_variable_matrix_dist2)

ortholog3_melt <- ortholog2_melt %>% filter(ortholog %in% most_variable_orthologs)
ortholog3_melt$ortholog <- factor(ortholog3_melt$ortholog, levels = ortholog_most_variable_matrix_hclust$labels)
ortholog3_melt$variable <- factor(ortholog3_melt$variable, levels = c("D614G","Delta","Alpha","Beta","Gamma","BA1"))

Most_variable_orthologs_s2variants <- ggplot() + 
  labs(x = NULL, y = NULL) + 
  scale_fill_gradient(low = "white", high = "red") + 
  theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  geom_tile(data = ortholog3_melt, aes(x = variable, y = ortholog, fill = value)); Most_variable_orthologs_s2variants
ggsave(file = "Plots/Most_variable_orthologs_s2variants.pdf", Most_variable_orthologs_s2variants, height = 1.45, width = 2.8)

```








### I THINK THIS IS THE CURRENT STOP POINT

```{r Orthologs}
#ortholog1_melt
combined2_melted <- melt(combined2[,1:7], id = "plasmid_template")
combined2_melted$value <- log10(combined2_melted$value)
ortholog_seqs2 <- merge(ortholog_seqs_variable_sites, combined2_melted, by = "plasmid_template")
ortholog_seqs3 <- ortholog_seqs2[,7:ncol(ortholog_seqs2)] %>% filter(!(variable %in% c("sd_d614g","sd_alpha","sd_beta","sd_gamma","sd_delta","sd_ba1")))

# This is going to be done with the RandomForest package
# Calculate the size of each of the data sets:
data_set_size <- floor(nrow(ortholog_seqs3)*1)
# Generate a random sample of "data_set_size" indexes
indexes <- sample(1:nrow(ortholog_seqs3), size = data_set_size)
# Assign the data to the correct sets
training1 <- ortholog_seqs3[indexes, !(colnames(ortholog_seqs3) %in% "sequence")]
test1 <- ortholog_seqs3[-indexes, !(colnames(ortholog_seqs3) %in% "sequence")]

library(randomForest)
rf_regression = randomForest(value ~ ., data=training1, ntree=200, mtry=5, importance=TRUE)

print(rf_regression)
plot(rf_regression)

feature_frame <- data.frame("IncMSE" = importance(rf_regression,type = 1))
feature_frame2 <- data.frame("IncNodePurity" = importance(rf_regression,type = 2))
feature_frame3 <- cbind(feature_frame, feature_frame2)
feature_frame3$name <- rownames(feature_frame3)
feature_frame3 <- feature_frame3[order(feature_frame3$X.IncMSE),]
feature_frame3$name <- factor(feature_frame3$name, levels = feature_frame3$name)

rf_regression_importance_plot <- ggplot() + theme_bw() + 
  labs(x = "% Increase MSE\nupon permutation", y = NULL) +
  geom_point(data = feature_frame3, aes(x = X.IncMSE, y = name))
rf_regression_importance_plot

feature_frame4 <- data.frame(varImpPlot(rf_regression))
feature_frame4$label <- rownames(feature_frame4)

rf_importance_scatterplot <- ggplot() + 
  labs(x = "Increase to MSE", y = "Node purity") +
  geom_point(data = feature_frame4, aes(x = X.IncMSE, y = IncNodePurity), alpha = 0.4) +
  geom_text_repel(data = feature_frame4 %>% filter(IncNodePurity > 0), aes(x = X.IncMSE, y = IncNodePurity, label = label), color = "red", size = 2, segment.colour = "orange", segment.alpha = 0.4)
rf_importance_scatterplot
ggsave(file = "Plots/rf_importance_scatterplot.pdf", rf_importance_scatterplot, height = 1.3, width = 2.25)

## We can now use the trained model on the validation dataset
prediction_test <- predict(rf_regression,test1[,colnames(test1) != c("calibrated_score","sequence")])
test1$predicted_scores <- prediction_test

ortholog_seqs2$imputed_score <- predict(rf_regression,ortholog_seqs2[,colnames(ortholog_seqs2) != c("value","plasmid_template","ortholog","mutant","protease","kozak","sequence")])

rf_ortholog_predict_scatterplot <- ggplot() + 
  labs(x = "Measured value", y = "Predicted value") +
  #geom_point(data = test1, aes(x = value, y = predicted_scores), alpha = 0.8, fill = "red", shape = 21, color = "black") +
  geom_point(data = ortholog_seqs2, aes(x = value, y = imputed_score, fill = variable), alpha = 0.3, shape = 21, fill = "black") +
  geom_text(data = ortholog_seqs2 %>% filter(value < 1 & imputed_score > 3), aes(x = value*1.1, y = imputed_score*1.2, label = paste(ortholog,variable)), color = "purple", size = 2)
rf_ortholog_predict_scatterplot
ggsave(file = "Plots/rf_ortholog_predict_scatterplot.pdf", rf_ortholog_predict_scatterplot, height = 1.4, width = 2.25)

cor(ortholog_seqs2$value, ortholog_seqs2$imputed_score, method = "pearson")^2

training2 <- training1
training2$predicted <- rf_regression$predicted

ggplot() + 
  geom_point(data = training2, aes(x = value, y = predicted))

cor(training2$value, training2$predicted, method = "pearson")^2

i21_grouped_data <- ortholog_seqs2 %>% group_by(I21, variable) %>% summarize(mean = mean(value))
n49_grouped_data <- ortholog_seqs2 %>% group_by(N49, variable) %>% summarize(mean = mean(value))

ggplot() + #geom_jitter(data = ortholog_seqs2, aes(x = I21, y = value)) +
  geom_point(data = i21_grouped_data, aes(x = I21, y = mean, color = variable))

ggplot() + #geom_jitter(data = ortholog_seqs2, aes(x = I21, y = value)) +
  geom_point(data = n49_grouped_data, aes(x = N49, y = mean, color = variable))

```









































### ALL OF THE STUFF BELOW IS OLD AND DEPRECATED

```{r Table of key amino acids}
tested_ace2 <- read.csv(file = "Data/Sequences/Tested_ACE2.csv")

tested_ace2$a21 <- substr(tested_ace2$sequence,21,21)
tested_ace2$a23 <- substr(tested_ace2$sequence,23,23)
tested_ace2$a24 <- substr(tested_ace2$sequence,24,24)
tested_ace2$a27 <- substr(tested_ace2$sequence,27,27)
tested_ace2$a30 <- substr(tested_ace2$sequence,30,30)
tested_ace2$a31 <- substr(tested_ace2$sequence,31,31)
tested_ace2$a34 <- substr(tested_ace2$sequence,34,34)
tested_ace2$a35 <- substr(tested_ace2$sequence,35,35)
tested_ace2$a38 <- substr(tested_ace2$sequence,38,38)
tested_ace2$a41 <- substr(tested_ace2$sequence,41,41)
tested_ace2$a42 <- substr(tested_ace2$sequence,42,42)
tested_ace2$a82 <- substr(tested_ace2$sequence,82,82)
tested_ace2$a83 <- substr(tested_ace2$sequence,83,83)
tested_ace2$a353 <- substr(tested_ace2$sequence,353,353)
tested_ace2$a355 <- substr(tested_ace2$sequence,355,355)
tested_ace2$a357 <- substr(tested_ace2$sequence,357,357)
tested_ace2$concat <- ""
for(x in 1:nrow(tested_ace2)){
  tested_ace2$concat[x] <- gsub("[^A-Z]","",toString(tested_ace2[x,5:20]))
}

tested_rbd <- read.csv(file = "Data/Sequences/Tested_RBDs.csv")
tested_rbd$r417_99 <- substr(tested_rbd$sequence,99,99)
tested_rbd$r449_131 <- substr(tested_rbd$sequence,131,131)
tested_rbd$r455_137 <- substr(tested_rbd$sequence,137,137)
tested_rbd$r456_138 <- substr(tested_rbd$sequence,138,138)
tested_rbd$r478_160 <- substr(tested_rbd$sequence,160,160)
tested_rbd$r486_168 <- substr(tested_rbd$sequence,168,168)
tested_rbd$r493_175 <- substr(tested_rbd$sequence,175,175)
tested_rbd$r496_178 <- substr(tested_rbd$sequence,178,178)
tested_rbd$r498_180 <- substr(tested_rbd$sequence,180,180)
tested_rbd$r501_183 <- substr(tested_rbd$sequence,183,183)
tested_rbd$r505_187 <- substr(tested_rbd$sequence,187,187)

tested_rbd$concat <- ""
for(x in 1:nrow(tested_rbd)){
  tested_rbd$concat[x] <- gsub("[^A-Z]","",toString(tested_rbd[x,3:13]))
}

ortholog3 <- ortholog1

ortholog3_lowest_four_mean <- c()
for(y in 1:(ncol(ortholog3)-1)){
  ortholog3_lowest_four_mean <- c(ortholog3_lowest_four_mean, mean(sort(ortholog3[,y])[1:4]))
}

for(x in 1:length(ortholog3_control_vector)){
  ortholog3[,x] <- ortholog3[,x] / as.numeric(ortholog3_lowest_four_mean[x])
}

ortholog3_melt <- melt(ortholog3, "id" = "ortholog")
colnames(ortholog3_melt) <- c("name","variable","value")
tested_rbd$variable <- tested_rbd$name


#tested_rbd_colnames_include <- c("r478_160","r493_175","r498_180","r501_183")
tested_rbd_colnames_exclude <- c()
#tested_rbd_colnames <- c("variable",tested_rbd_colnames_include)
tested_ace2_colnnames <- colnames(tested_rbd)[!(colnames(tested_rbd) %in% c("name","sequence","concat",tested_ace2_colnnames_exclude))]
ortholog3_melt_rbd <- merge(ortholog3_melt, tested_rbd[,tested_rbd_colnames])


#tested_ace2_colnames_include <- c("a24","a31","a34","a82")
tested_ace2_colnames_exclude <- c()
#tested_ace2_colnames <- c("name",tested_ace2_colnames_include)
tested_ace2_colnnames <- colnames(tested_ace2)[!(colnames(tested_ace2) %in% c("name","sequence","concat",tested_ace2_colnnames_exclude))]
ortholog3_melt_rbd_ace2 <- merge(ortholog3_melt_rbd, tested_ace2[,tested_ace2_colnames])


ortholog3_melt_for_rf <- ortholog3_melt_rbd_ace2[,3:ncol(ortholog3_melt_rbd_ace2)]

library(randomForest)
# Calculate the size of each of the data sets:
data_set_size <- floor(nrow(ortholog3_melt_for_rf)*0.8)
# Generate a random sample of "data_set_size" indexes
indexes <- sample(1:nrow(ortholog3_melt_for_rf), size = data_set_size)
# Assign the data to the correct sets
training1 <- ortholog3_melt_for_rf[indexes, !(colnames(ortholog3_melt_for_rf) %in% "sequence")]
test1 <- ortholog3_melt_for_rf[-indexes, !(colnames(ortholog3_melt_for_rf) %in% "sequence")]

rf_regression = randomForest(value ~ ., data=training1, ntree=200, mtry=6, importance=TRUE)

print(rf_regression)
plot(rf_regression)

feature_frame <- data.frame("IncMSE" = importance(rf_regression,type = 1))
feature_frame$name <- rownames(feature_frame)
feature_frame <- feature_frame[order(feature_frame$X.IncMSE),]
feature_frame$name <- factor(feature_frame$name, levels = feature_frame$name)

rf_regression_importance_plot <- ggplot() + theme_bw() + 
  labs(x = "% Increase MSE\nupon permutation", y = NULL) +
  geom_point(data = feature_frame, aes(x = X.IncMSE, y = name))
rf_regression_importance_plot
#ggsave(file = "plots/rf_regression_importance_plot.png", rf_regression_importance_plot, height = 3.2, width = 2)
#ggsave(file = "plots/rf_regression_importance_plot.pdf", rf_regression_importance_plot, height = 3.2, width = 2)

feature_Frame2 <- data.frame(varImpPlot(rf_regression))
## We can now use the trained model on the validation dataset
prediction_test <- predict(rf_regression,test1[,colnames(test1) != c("mfi_calcd_norm","sequence")])

test1$predicted_scores <- prediction_test

ggplot() + scale_x_log10() + scale_y_log10() + geom_point(data = test1, aes(x = value, y = predicted_scores))

ortholog3_melt_rbd_ace2$rf_predicted <- predict(rf_regression,ortholog3_melt_for_rf[,2:ncol(ortholog3_melt_for_rf)])

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = ortholog3_melt_rbd_ace2, aes(x = value, y = rf_predicted), alpha = 0.2) +
  #geom_point(data = test1, aes(x = value, y = predicted_scores), alpha = 0.5, color = "red") +
  NULL


## Simple decision tree
tree <- rpart(value ~ ., data=training1, method = "class", control = rpart.control(cp = 0, minsplit = 1, minbucket = 3))
plot(tree)

temp_rules <- rpart.rules(tree, cover = TRUE)
temp_rules$value <- as.numeric(temp_rules$value)

subset1 <- ortholog3_melt_rbd_ace2 %>% filter(a24 == "L" & a34 == c("H","L") & r478_160 == "Q")





# plot tree 
#par(mar = c(1, 1, 1, 1))
# https://cran.r-project.org/web/packages/rpart.plot/rpart.plot.pdf
rpart.plot(tree, type = 3, fallen.leaves = TRUE, cex = 0.5, split.cex = 1)

```
















```{r Human ACE2 with various conditions}
kozak_protease <- combined2 %>% filter(ortholog == "H.sapiens") %>%  filter(mutant == "WT")
for(x in 1:nrow(kozak_protease)){
  kozak_protease$identifier[x] <- strsplit(kozak_protease$plasmid_template[x],"_")[[1]][1]
}

#for(x in c("g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")){
#  temp_value <-  kozak_protease[ kozak_protease$plasmid_template == "G755A_AttB_[koz-mut]ACE2-IRES-mCherry-H2A-P2A-PuroR",x] ##"G828A_AttB_ACE2-2A-TMPRSS2_IRES_mCherry-H2A-P2A-PuroR" ,x] # <- For some reason, this is kind of weird
#  kozak_protease[,x] <- kozak_protease[,x] / temp_value
#}

## Compare high and low Kozak
lowhigh_reps <- kozak_protease %>% filter(protease == "none")
lowhigh_reps_t <- data.frame(t(lowhigh_reps[,c("identifier","Alpha","Beta","Delta","Gamma","BA1","D614G")]))
colnames(lowhigh_reps_t) <- lowhigh_reps_t[1,]
lowhigh_reps_t <- lowhigh_reps_t[-1,]
lowhigh_reps_t$G755A <- as.numeric(lowhigh_reps_t$G755A)
lowhigh_reps_t$G752A <- as.numeric(lowhigh_reps_t$G752A)

ACE2_low_high_plot <- ggplot() + labs(x = "ACE2 only (low)", y = "ACE2 only (high)") +
  scale_x_continuous(limits = c(0,12.5), breaks = seq(0,12,2)) + 
  scale_y_continuous(limits = c(0,12.5), breaks = seq(0,12,2)) + 
  geom_vline(xintercept = 1, alpha = 0.4) + geom_hline(yintercept = 1, alpha = 0.4) +
  geom_point(data = lowhigh_reps_t, aes(x = G755A, y = G752A)) +
  geom_text_repel(data = lowhigh_reps_t, aes(x = G755A, y = G752A, label = rownames(lowhigh_reps_t)), color = "red")
ACE2_low_high_plot
ggsave(file = "Plots/ACE2_low_high_plot.pdf", ACE2_low_high_plot, height = 1.75, width = 2.25)


## Now let's look at the effect of TMPRSS2 on the high ACE2 expressor cells
high_protease <- kozak_protease %>% filter(identifier %in% c("G828A","G752A"))
high_protease_t <- data.frame(t(high_protease[,c("identifier","Alpha","Beta","Delta","Gamma","BA1","D614G")]))
colnames(high_protease_t) <- high_protease_t[1,]
high_protease_t <- high_protease_t[-1,]
high_protease_t$G828A <- as.numeric(high_protease_t$G828A)
high_protease_t$G752A <- as.numeric(high_protease_t$G752A)

ACE2_TMPRESS2_plot <- ggplot() + labs(x = "ACE2 only (G752A)", y = "ACE2-2A-TMPRSS2 (G828A)") +
  scale_x_continuous(limits = c(0,16), breaks = seq(0,16,4)) + 
  scale_y_continuous(limits = c(0,16), breaks = seq(0,16,4)) + 
  geom_vline(xintercept = 1, alpha = 0.4) + geom_hline(yintercept = 1, alpha = 0.4) +
  geom_point(data = high_protease_t, aes(x = G752A, y = G828A)) +
  geom_text_repel(data = high_protease_t, aes(x = G752A, y = G828A, label = rownames(high_protease_t)), color = "red")
ACE2_TMPRESS2_plot
ggsave(file = "Plots/ACE2_TMPRESS2_plot.pdf", ACE2_TMPRESS2_plot, height = 1.75, width = 2.25)


## Compare G852C and G828A
high_tmprss2_reps <- kozak_protease %>% filter(protease == "TMPRSS2")
high_tmprss2_reps_t <- data.frame(t(high_tmprss2_reps[,c("identifier","Alpha","Beta","Delta","Gamma","BA1","D614G")]))
colnames(high_tmprss2_reps_t) <- high_tmprss2_reps_t[1,]
high_tmprss2_reps_t <- high_tmprss2_reps_t[-1,]
high_tmprss2_reps_t$G828A <- as.numeric(high_tmprss2_reps_t$G828A)
high_tmprss2_reps_t$G852C <- as.numeric(high_tmprss2_reps_t$G852C)

ACE2_w_TMPRESS2_rep_plot <- ggplot() + labs(x = "G828A (mCherry)", y = "G852C (miRFP670)") +
  scale_x_continuous(limits = c(0,12.5), breaks = seq(0,12,2)) + 
  scale_y_continuous(limits = c(0,7.5), breaks = seq(0,12,2)) + 
  geom_vline(xintercept = 1, alpha = 0.4) + geom_hline(yintercept = 1, alpha = 0.4) +
  geom_point(data = high_tmprss2_reps_t, aes(x = G828A, y = G852C)) +
  geom_text_repel(data = high_tmprss2_reps_t, aes(x = G828A, y = G852C, label = rownames(high_tmprss2_reps_t)), color = "red")
ACE2_w_TMPRESS2_rep_plot
ggsave(file = "Plots/ACE2_w_TMPRESS2_rep_plot.pdf", ACE2_w_TMPRESS2_rep_plot, height = 1.75, width = 2.25)

## Now combine the redundant data
no_trmpss2_wt <- dbl_combined2 %>% filter(!(plasmid_template %in% c("G852C_AttB_ACE2-2A-TMPRSS2_IRES_miRFP670-H2A-P2A-PuroR", "G828A_AttB_ACE2-2A-TMPRSS2_IRES_mCherry-H2A-P2A-PuroR")))

tmprss2_wt <- dbl_combined2 %>% filter(plasmid_template %in% c("G852C_AttB_ACE2-2A-TMPRSS2_IRES_miRFP670-H2A-P2A-PuroR", "G828A_AttB_ACE2-2A-TMPRSS2_IRES_mCherry-H2A-P2A-PuroR"))

dbl_combined3 <- rbind(no_trmpss2_wt, 
c("NA","ACE2","H.sapiens","WT","TMPRSS2","high","G828A_and_G852C","QSTIEEQAKTFLDKFNHEAEDLFYQSSLAQMYLGKGDFR",10^colMeans(log10(tmprss2_wt[,c("g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")]))))

dbl_combined3[,c("g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")] <- lapply(dbl_combined3[,c("g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")], as.numeric)
```

```{r Compare enrichments of Beta and Gamma since they are identical RBMs}
Beta_and_gamma_correlation <- ggplot() + labs(x = "Fold enrichment\nSARS-CoV-2 Beta", y = "Fold enrichment\nSARS-CoV-2 Gamma") +
  scale_x_log10() + scale_y_log10() + 
  geom_abline(slope = 1, linetype = 2, alpha = 0.3) +
  geom_point(data = dbl_combined3, aes(x = s2beta, y = s2gamma), alpha = 0.4) +
  #geom_text_repel(data = dbl_combined3 %>% filter(ortholog != "H.sapiens"), aes(x = s2beta, y = s2gamma, label = ortholog), color = "red", alpha = 0.5) +
  NULL
Beta_and_gamma_correlation
ggsave(file = "Plots/Beta_and_gamma_correlation.pdf", Beta_and_gamma_correlation, height = 1.75, width = 2)

```

```{r Orthologs}
tmprss2 <- dbl_combined3 %>% filter(protease == "TMPRSS2" & mutant == "WT" | kozak == "high" & mutant == "dEcto")

## Values left unscaled
ortholog1 <- tmprss2[,c("ortholog", "s2alpha","s2beta","s2delta","s2gamma","s2omicron")]
ortholog1_melt <- melt(ortholog1, id = "ortholog")

#ortholog_melt$ortholog <- factor(ortholog_melt$ortholog, levels = c("H.sapiens", "Control", "M.javanica", "S.scrofa", "M.musculus", "R.affinis","R.shameli","R.pearsonii", "R.alcyone", "R.landeri", "R.ferrumequinum", "R.sinicus", "R.sinicus_472", "R.sinicus_215", "R.sinicus_200"))
ortholog1_melt$ortholog <- factor(ortholog1_melt$ortholog, levels = c("Control", "M.musculus", "R.landeri", "R.alcyone", "R.ferrumequinum", "R.shameli", "R.affinis",  "R.sinicus_215", "R.sinicus", "R.sinicus_200", "R.pearsonii", "R.sinicus_472",  "H.sapiens", "M.javanica", "S.scrofa"))
ortholog1_melt$variable <- factor(ortholog1_melt$variable, levels = c("s2alpha", "s2beta","s2gamma","s2delta","s2omicron"))

orthologs1_s2variants <- ggplot() + labs(x = NULL, y = NULL) + 
  scale_fill_gradient(low = "white", high = "red", limits=c(0,10)) + 
  theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  geom_tile(data = ortholog1_melt, aes(x = ortholog, y = variable, fill = value)); orthologs1_s2variants
ggsave(file = "Plots/orthologs1_s2variants.pdf", orthologs1_s2variants, height = 1.5, width = 4)

## Scaled to dEcto and Human High Kozak with TMPRSS2
tmprss2b <- tmprss2
for(x in c("g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")){
  temp_high <-  tmprss2b[ tmprss2b$plasmid_template == "G828A_and_G852C" ,x]
  temp_low <-  tmprss2b[ tmprss2b$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR" ,x]
  tmprss2b[,x] <- (tmprss2b[,x] - temp_low) / (temp_high - temp_low)
}
ortholog2 <- tmprss2b[,c("ortholog","s2alpha","s2beta","s2delta","s2gamma","s2omicron")]
ortholog2_melt <- melt(ortholog2, id = "ortholog")

#ortholog_melt$ortholog <- factor(ortholog_melt$ortholog, levels = c("H.sapiens", "Control", "M.javanica", "S.scrofa", "M.musculus", "R.affinis","R.shameli","R.pearsonii", "R.alcyone", "R.landeri", "R.ferrumequinum", "R.sinicus", "R.sinicus_472", "R.sinicus_215", "R.sinicus_200"))
ortholog2_melt$ortholog <- factor(ortholog2_melt$ortholog, levels = c("Control", "M.musculus", "R.landeri", "R.alcyone", "R.ferrumequinum", "R.shameli", "R.affinis",  "R.sinicus_215", "R.sinicus", "R.sinicus_200", "R.pearsonii", "R.sinicus_472",  "H.sapiens", "M.javanica", "S.scrofa"))

orthologs2_s2variants <- ggplot() + 
  labs(x = NULL, y = NULL) + 
  scale_fill_gradient(low = "white", high = "red") + 
  theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  geom_tile(data = ortholog2_melt, aes(x = ortholog, y = variable, fill = value)); orthologs2_s2variants
ggsave(file = "Plots/orthologs2_s2variants.pdf", orthologs2_s2variants, height = 1.5, width = 4)
```

```{r Variants of human ACE2 with high Kozak}
high_human <- dbl_combined3 %>% filter(kozak == "high" & ortholog %in% c("H.sapiens","Control"))

## Values left unscaled
high_human1_s2variants <- high_human[,c("mutant","s2alpha","s2beta","s2delta","s2gamma","s2omicron")]
high_human1_s2variants_melt <- melt(high_human1_s2variants, id = "mutant")
high_human1_s2variants_melt$variable <- factor(high_human1_s2variants_melt$variable, levels = c("s2alpha", "s2beta","s2gamma","s2delta","s2omicron"))

high_human1_s2variants_melt$mutant <- factor(high_human1_s2variants_melt$mutant, levels = c("WT", "dEcto", "K31D","E329K","D355N"))

high_human1_s2variants_melt <- ggplot() + labs(x = NULL, y = NULL) + 
  scale_fill_gradient(low = "white", high = "red") + 
  geom_tile(data = high_human1_s2variants_melt, aes(x = mutant, y = variable, fill = value)); high_human1_s2variants_melt
ggsave(file = "Plots/high_human1_s2variants_melt.pdf", high_human1_s2variants_melt, height = 2.5, width = 8)

## Scaled to dEcto and Human High Kozak with TMPRSS2
high_human2 <- high_human
for(x in c("g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")){
  temp_high <-  high_human2[ high_human2$plasmid_template == "G828A_and_G852C" ,x]
  temp_low <-  high_human2[ high_human2$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR" ,x]
  high_human2[,x] <- (high_human2[,x] - temp_low) / (temp_high - temp_low)
}

high_human2_s2variants <- high_human2[,c("mutant","s2alpha","s2beta","s2delta","s2gamma","s2omicron")]
high_human2_s2variants_melt <- melt(high_human2_s2variants, id = "mutant")
high_human2_s2variants_melt$variable <- factor(high_human2_s2variants_melt$variable, levels = c("s2alpha", "s2beta","s2gamma","s2delta","s2omicron"))

high_human2_s2variants_melt$mutant <- factor(high_human2_s2variants_melt$mutant, levels = c("WT", "dEcto", "K31D","E329K","D355N"))

high_human2_s2variants_melt <- ggplot() + labs(x = NULL, y = NULL) + 
  scale_fill_gradient(low = "white", high = "red") + 
  geom_tile(data = high_human2_s2variants_melt, aes(x = mutant, y = variable, fill = value)); high_human2_s2variants_melt
ggsave(file = "Plots/high_human2_s2variants_melt.pdf", high_human2_s2variants_melt, height = 2.5, width = 8)
```

```{r Variants of human ACE2 with low Kozak}
low_kozak <- dbl_combined3 %>% filter(kozak == "low" | mutant == "dEcto")

## Normal unscaled
low_kozak1 <- low_kozak

low_kozak1_s2variants <- low_kozak1[,c("mutant","s2alpha","s2beta","s2delta","s2gamma","s2omicron")]
low_kozak1_s2variants_melt <- melt(low_kozak1_s2variants, id = "mutant")
low_kozak1_s2variants_melt$variable <- factor(low_kozak1_s2variants_melt$variable, levels = c("s2alpha", "s2beta","s2gamma","s2delta","s2omicron"))
low_kozak1_s2variants_melt$mutant <- factor(low_kozak1_s2variants_melt$mutant, levels = c("WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "G352V", "K353D", "D355N"))

low_kozak1_ace2mutants_s2variants <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  scale_fill_gradient(low = "white", high = "red", limits=c(0,10)) + 
  geom_tile(data = low_kozak1_s2variants_melt, aes(x = mutant, y = variable, fill = value)); low_kozak1_ace2mutants_s2variants
low_kozak1_ace2mutants_s2variants
ggsave(file = "Plots/low_kozak1_ace2mutants_s2variants.pdf", low_kozak1_ace2mutants_s2variants, height = 1.25, width = 4)


## Scaled to dEcto and Human Low Kozak
low_kozak2 <- low_kozak
for(x in c("g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")){
  temp_high <-  low_kozak2[ low_kozak2$plasmid_template == "G755A_AttB_[koz-mut]ACE2-IRES-mCherry-H2A-P2A-PuroR" ,x]
  temp_low <-  low_kozak2[ low_kozak2$plasmid_template == "G758A_AttB_ACE2[dEcto]-IRES-mCherry-H2A-P2A-PuroR" ,x]
  low_kozak2[,x] <- (low_kozak2[,x] - temp_low) / (temp_high - temp_low)
}

low_kozak2_s2variants <- low_kozak2[,c("mutant","s2alpha","s2beta","s2delta","s2gamma","s2omicron")]
low_kozak2_s2variants_melt <- melt(low_kozak2_s2variants, id = "mutant")
low_kozak2_s2variants_melt$variable <- factor(low_kozak2_s2variants_melt$variable, levels = c("s2alpha", "s2beta","s2gamma","s2delta","s2omicron"))
low_kozak2_s2variants_melt$mutant <- factor(low_kozak2_s2variants_melt$mutant, levels = c("WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "G352V", "K353D", "D355N"))

low_kozak2_ace2mutants_s2variants <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45)) + 
  scale_fill_gradient(low = "white", high = "red") + 
  geom_tile(data = low_kozak2_s2variants_melt, aes(x = mutant, y = variable, fill = value)); low_kozak2_ace2mutants_s2variants
low_kozak2_ace2mutants_s2variants
ggsave(file = "Plots/low_kozak2_ace2mutants_s2variants.pdf", low_kozak2_ace2mutants_s2variants, height = 1.05, width = 3.4)


low_kozak2_s2variants_melt$mutant <- factor(low_kozak2_s2variants_melt$mutant, levels = rev(c("WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "G352V", "K353D", "D355N")))

low_kozak2_ace2mutants_s2variants_flip <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45), legend.position = "top") + 
  scale_fill_gradient(low = "white", high = "red") + 
  geom_tile(data = low_kozak2_s2variants_melt, aes(x = variable, y = mutant, fill = value)); low_kozak2_ace2mutants_s2variants
low_kozak2_ace2mutants_s2variants_flip
ggsave(file = "Plots/low_kozak2_ace2mutants_s2variants_flip.pdf", low_kozak2_ace2mutants_s2variants_flip, height = 3, width = 1.25)
```

```{r Making a combined heatmap with both orthologs and variants}
combined_ortholog_variant <- rbind((ortholog1_melt %>% mutate(sample = ortholog))[,c("sample","variable","value")],
                                   (low_kozak1_s2variants_melt %>% mutate(sample = mutant))[,c("sample","variable","value")])

combined_ortholog_variant$sample <- factor(combined_ortholog_variant$sample, levels = rev(c("Control", "M.musculus", "R.landeri", "R.alcyone", "R.ferrumequinum", "R.shameli", "R.affinis",  "R.sinicus_215", "R.sinicus", "R.sinicus_200", "R.pearsonii", "R.sinicus_472", "M.javanica", "S.scrofa", "H.sapiens", "WT", "dEcto", "I21N", "E23K", "K26E", "K31D", "E35K", "D38H", "G326E","E329G", "G352V", "K353D", "D355N")))

combined_ortholog_variant_plot <- ggplot() + labs(x = NULL, y = NULL) + theme(axis.text.x = element_text(hjust = 1, vjust = 1, angle = 45), legend.position = "top") + 
  scale_fill_gradient(low = "white", high = "red", limits=c(0,10)) + 
  geom_tile(data = combined_ortholog_variant, aes(x = variable, y = sample, fill = value)); low_kozak1_ace2mutants_s2variants
combined_ortholog_variant_plot
ggsave(file = "Plots/combined_ortholog_variant_plot.pdf", combined_ortholog_variant_plot, height = 4, width = 1.75)
```





```{r Comparing to 2021 PlosP data}
human_ace2_variants <- read.delim(file = "Data/Other_papers/2021_Shukla_PlosPathog/journal.ppat.1009715.s008.tsv", sep = "\t")
human_ace2_variants2 <- human_ace2_variants %>% group_by(cell_label) %>% summarize(mean_infection = mean(scaled_infection)) #10^mean(log_scaled_infection))

human_ace2_variants2$mutant <- NA
for(x in 3:nrow(human_ace2_variants2)){
  human_ace2_variants2$mutant[x] <- strsplit(human_ace2_variants2$cell_label[x],"-")[[1]][2]
}
human_ace2_variants2$mutant[1] <- "Neg cntrl"
human_ace2_variants2$mutant[2] <- "WT"

#dbl_combined3_temp <- dbl_combined3 %>% filter(kozak == "low" | mutant == "dEcto")
low_kozak1_s2variants
low_kozak1_s2variants[low_kozak1_s2variants$mutant == "dEcto","mutant"] <- "Neg cntrl"

human_ace2_variants3 <- merge(human_ace2_variants2,low_kozak1_s2variants[,c("mutant","s2alpha")])
human_ace2_variants3$s2alpha <- human_ace2_variants3$s2alpha / human_ace2_variants3[human_ace2_variants3$mutant == "WT","s2alpha"]

Variant_validation_graph <- ggplot() + labs(x = "Flow cytometry infectivity\nmeasurement from published study", y = "Sequencing-based\ninfectivity measurement\nfrom current study") +
  scale_x_continuous(limits = c(0,1.6), expand = c(0,0), breaks = seq(0,1.5,0.5)) + 
  scale_y_continuous(limits = c(0,1.7), expand = c(0,0), breaks = seq(0,1.5,0.5)) +
  geom_abline(slope = 1, linetype = 2, alpha = 0.3) +
  geom_point(data = human_ace2_variants3, aes(x = mean_infection, y = s2alpha), alpha = 0.5) +
  geom_text_repel(data = human_ace2_variants3, aes(x = mean_infection, y = s2alpha, label = mutant), color = "red", alpha = 0.8, size = 2, segment.color = "orange", segment.alpha = 0.5)
Variant_validation_graph
ggsave(file = "Plots/Variant_validation_graph.pdf", Variant_validation_graph, height = 1.75, width = 2.5)
```

```{r Comparing to 2022 PlosP Biol}
ortholog_2color_data <- read.delim(file = "Data/Other_papers/2022_Roelle_PlosBiol/Fig5.csv", sep = ",")
ortholog_2color_data_flow <- ortholog_2color_data %>% filter(type == "flow_cytometry" & virus_label == "SARS-CoV-2 RBD") %>% mutate(ortholog = cell_label)
ortholog_2color_data_micro <- ortholog_2color_data %>%  filter(type == "microscopy" & cell_label == "SARS-CoV-2 RBD") %>% mutate(ortholog = virus_label)

ortholog_2color_data_flow$norm_geomean <- ortholog_2color_data_flow$geomean / ortholog_2color_data_flow[ortholog_2color_data_flow$ortholog == "H.sapiens","geomean"]
ortholog_2color_data_micro$norm_geomean <- ortholog_2color_data_micro$geomean / ortholog_2color_data_micro[ortholog_2color_data_micro$ortholog == "H.sapiens","geomean"]

norm_ortholog_2color_data <- merge(ortholog_2color_data_flow[,c("ortholog","norm_geomean")], ortholog_2color_data_micro[,c("ortholog","norm_geomean")], by = "ortholog")
norm_ortholog_2color_data$norm_geomean <- rowMeans(norm_ortholog_2color_data[,c("norm_geomean.x","norm_geomean.y")])

norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "R.sinicus200","ortholog"] <- "R.sinicus_200"
norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "R.sinicus215","ortholog"] <- "R.sinicus_215"
norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "R.sinicus472","ortholog"] <- "R.sinicus_472"
norm_ortholog_2color_data[norm_ortholog_2color_data$ortholog == "NULL (fs)","ortholog"] <- "Neg cntrl"

ortholog1$non_omicron <- rowMeans(ortholog1[,c("s2alpha","s2beta","s2gamma","s2delta")])

ortholog1[ortholog1$ortholog == "Control","ortholog"] <- "Neg cntrl"


ortholog1_melt_s2_plosbiol <- merge(ortholog1[,c("ortholog","non_omicron")], norm_ortholog_2color_data[,c("ortholog","norm_geomean")], by = "ortholog", all = T)
ortholog1_melt_s2_plosbiol$non_omicron <- ortholog1_melt_s2_plosbiol$non_omicron / ortholog1_melt_s2_plosbiol[ortholog1_melt_s2_plosbiol$ortholog == "H.sapiens","non_omicron"]

Variant_validation_graph2 <- ggplot() + labs(x = "Flow cytometry infectivity\nmeasurement from published study", y = "Sequencing-based\ninfectivity measurement\nfrom current study") +
  scale_x_log10(limits = c(0.1,1.05), expand = c(0,0), breaks = c(0.1,0.3,1)) + 
  scale_y_log10(limits = c(0.05,1.8), expand = c(0,0), breaks = c(0.1,0.3,1)) +
  geom_abline(slope = 1, linetype = 2, alpha = 0.3) +
  geom_point(data = ortholog1_melt_s2_plosbiol, aes(x = norm_geomean, y = non_omicron), alpha = 0.5) +
  geom_text_repel(data = ortholog1_melt_s2_plosbiol, aes(x = norm_geomean, y = non_omicron, label = ortholog), color = "red", alpha = 0.8, size = 2, segment.color = "orange", segment.alpha = 0.5)
Variant_validation_graph2
ggsave(file = "Plots/Variant_validation_graph2.pdf", Variant_validation_graph2, height = 1.75, width = 2.5)
```



```{r Determining compatible or not}
dbl_combined4 <- dbl_combined3 %>% filter(!is.na(concat))

dbl_combined4_melted <- rbind(
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$g740d,"sample" = "SARS2"),
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$s2alpha, "sample" = "Alpha"),
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$s2beta, "sample" = "Beta"),
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$s2delta, "sample" = "Delta"),
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$s2gamma, "sample" = "Gamma"),
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$s2omicron, "sample" = "Omicron"),
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$s1, "sample" = "SARS1"),
  cbind(dbl_combined4[,c("ortholog","mutant","protease","kozak","plasmid_template","concat")], "enrichment" = dbl_combined4$wiv1, "sample" = "WIV1"))

dbl_combined4_melted$compatible <- dbl_combined4_melted$enrichment > 0.9
dbl_combined4_melted <- dbl_combined4_melted %>% filter(!is.na(enrichment))

### Now let's only do this for the SARS2 variants
dbl_combined4_melted <- dbl_combined4_melted %>% filter(sample %in% c("Alpha","Beta","Delta","Gamma"))

unique_seqs <- data.frame("concat" = unique(dbl_combined4_melted$concat))

unique_seqs$compatible <- FALSE
unique_seqs$name <- "NA"
for(x in 1:nrow(unique_seqs)){
  temp_subset <- subset(dbl_combined4_melted, concat == unique_seqs$concat[x])
  if(sum(temp_subset$compatible) >= 1){
    unique_seqs$compatible[x] <- TRUE
  }
  unique_seqs$name[x] <- paste0(temp_subset$ortholog,"_",temp_subset$mutant)[1]
}

dbl_combined4_seqs <- unique_seqs$concat[1]
ace2_matrix2 <- matrix(ncol = nchar(dbl_combined4_seqs), nrow = nrow(unique_seqs))

for(x in 1:nrow(unique_seqs)){
  temp_seq <- unique_seqs$concat[x]
  for(y in 1:nchar(dbl_combined4_seqs)){
    ace2_matrix2[x,y] <- substr(temp_seq,y,y)
  }
}

## For Pymol: select mutated_ace2_positions,  ace2 and resi 18-44+79-83+351-357
ace2_positions <- c(seq(18,44,1),seq(79,83),seq(351,357))
position_rownames <- c()
for(x in ace2_positions){position_rownames <- c(position_rownames,paste("X",x,sep=""))}

ace2_unpublished2 <- data.frame(ace2_matrix2)
colnames(ace2_unpublished2) <- position_rownames
#rownames(ace2_unpublished2) <- dbl_combined4_melted$plamsid_template

ace2_unpublished2$compatible <- unique_seqs$compatible

ace2_unpublished2[ace2_unpublished2$compatible == TRUE,"compatible"] <- "yes"
ace2_unpublished2[ace2_unpublished2$compatible == FALSE,"compatible"] <- "no"
ace2_unpublished2$compatible <- factor(ace2_unpublished2$compatible, levels = c("yes","no"))

ace2_unpublished2$name <- unique_seqs$name
```


## First looking at published ACE2 utilization data
```{r importing the relevant data}
ace2 <- read_sheet("https://docs.google.com/spreadsheets/d/1o70izIccZisJArP_h1sKBO4FCGCO9PYIE3AjY7pH4qU/edit?usp=sharing", sheet = "ACE2") %>% filter(include == "yes" & wt == "yes")
rbm <- read_sheet("https://docs.google.com/spreadsheets/d/1o70izIccZisJArP_h1sKBO4FCGCO9PYIE3AjY7pH4qU/edit?usp=sharing", sheet = "RBM")
output <- read_sheet("https://docs.google.com/spreadsheets/d/1o70izIccZisJArP_h1sKBO4FCGCO9PYIE3AjY7pH4qU/edit?usp=sharing", sheet = "output_list") %>% filter(!is.na(sciname))

output2 <- merge(output[,c("ace2","sciname","rbd","compatible")], ace2[,c("sciname","region1","region2","region3")], by = "sciname", all.x = T)
output3 <- merge(output2, rbm[,c("rbd","rbm_seq")], by = "rbd", all.x = T)
```

## Honing in only on SARS-CoV-2
```{r}
#sars1_only <- output3 %>% filter(rbd == "SARS1")
sars2_only <- output3 %>% filter(rbd == "SARS2") %>% filter(!is.na(region1))

sars2_only$ace2_intxn <- paste(sars2_only$region1, sars2_only$region2, sars2_only$region3, sep = "")

sars2_only2 <- sars2_only %>% select(rbd, sciname, compatible, ace2_intxn) %>% filter(!(compatible == "no (skip)") & !(compatible == "yes (skip)"))%>%  distinct()

table(sars2_only2$compatible)
```

## Make a matrix of ACE2 sequences for the published data
```{r Make a matrix of ACE2 sequences}
example_intxn_seq <- sars2_only2$ace2_intxn[1]

ace2_matrix <- matrix(ncol = nchar(example_intxn_seq), nrow = nrow(sars2_only2))

for(x in 1:nrow(sars2_only2)){
  temp_seq <- sars2_only2$ace2_intxn[x]
  for(y in 1:nchar(example_intxn_seq)){
    ace2_matrix[x,y] <- substr(temp_seq,y,y)
  }
}

positions <- c(seq(18,44,1),seq(79,83),seq(351,357))
position_rownames <- c()
for(x in positions){position_rownames <- c(position_rownames,paste("X",x,sep=""))}

ace2_published <- data.frame(ace2_matrix)
colnames(ace2_published) <- position_rownames

ace2_published2 <- cbind(ace2_published, data.frame("compatible" = sars2_only2[,"compatible"]))
ace2_published2$compatible <- factor(ace2_published2$compatible, levels = unique(ace2_published2$compatible))

ace2_published3 <- cbind(ace2_published2, data.frame("name" = sars2_only2[,"sciname"]))

table(ace2_published3$compatible)
```

### Combine both sets of data to see if that works any better
```{r Create a simple decision tree separating the data}
# Set random seed to make results reproducible:
set.seed(1234567)
# Calculate the size of each of the data sets:
#data_set_size <- floor(nrow(ace2_dataframe2) * 1) #0.9
# Generate a random sample of "data_set_size" indexes
#indexes <- sample(1:nrow(ace2_dataframe2), size = data_set_size)
# Assign the data to the correct sets
#training2 <- ace2_dataframe2[indexes,]
#test2 <- ace2_dataframe2[-indexes,]

training2 <- rbind(ace2_unpublished2, ace2_published3)
training3 <- training2[,-ncol(training2)]

## Visualizing some trees for this dataset
tree <- rpart(compatible ~ ., data=training3, method = "class", control = rpart.control(cp = 0, minsplit = 1))
plot(tree)

# plot tree 
#par(mar = c(1, 1, 1, 1))
# https://cran.r-project.org/web/packages/rpart.plot/rpart.plot.pdf
rpart.plot(tree, type = 5)
```


```{r Looking at what signatures are incompatible from the upper patch}
upper_patch <- training2 %>% select(X353, X354, compatible)
table(upper_patch)

# New world monkeys
upper_patch_kq <- training2 %>% select(X353, X354, compatible, name) %>% filter(X353 == "K" & X354 == "Q")
# Mink and ferret
upper_patch_kr <- training2 %>% select(X353, X354, compatible, name) %>% filter(X353 == "K" & X354 == "R")
# R.pearsonii and Kuhl's pipistrelle
upper_patch_kd <- training2 %>% select(X353, X354, compatible, name) %>% filter(X353 == "K" & X354 == "D")
# Mouse and rat
upper_patch_hg <- training2 %>% select(X353, X354, compatible, name) %>% filter(X353 == "H" & X354 == "G")
# K353D
upper_patch_dg <- training2 %>% select(X353, X354, compatible, name) %>% filter(X353 == "D" & X354 == "G")
# European hedgehog
upper_patch_ng <- training2 %>% select(X353, X354, compatible, name) %>% filter(X353 == "N" & X354 == "G")
# Lesser hedgehog tenrec
upper_patch_ln <- training2 %>% select(X353, X354, compatible, name) %>% filter(X353 == "L" & X354 == "N")


upper_patch_kg <- training2 %>% filter(X353 == "K" & X354 == "G")
upper_patch_kg_noname <- upper_patch_kg[,-ncol(upper_patch_kg)]

upper_patch_kg_tree <- rpart(compatible ~ ., data=upper_patch_kg_noname, method = "class", control = rpart.control(cp = 0, minsplit = 1))
rpart.plot(upper_patch_kg_tree, type = 5)

middle_patch <- upper_patch_kg %>% select(X31, X34, compatible)
table(middle_patch)

# Koala
middle_patch_tk <- upper_patch_kg %>% select(X31, X34, compatible, name) %>% filter(X31 == "T" & X34 == "K")
# Natal long-fingered bat
middle_patch_gs <- upper_patch_kg %>% select(X31, X34, compatible, name) %>% filter(X31 == "G" & X34 == "S")
# K31D
middle_patch_dh <- upper_patch_kg %>% select(X31, X34, compatible, name) %>% filter(X31 == "D" & X34 == "H")

## These are all compatible!
upper_kg_middle_kh <- training2 %>% filter(X353 == "K" & X354 == "G" & X31 == "K" & X34 == "H")

## Middle neg is only partially ocmpatible
upper_kg_middle_neg <- training2 %>% filter(X353 == "K" & X354 == "G" & X31 %in% c("D","E"))
upper_kg_middle_neg_noname <- upper_kg_middle_neg[,-ncol(upper_kg_middle_neg)]

upper_kg_middle_neg_tree <- rpart(compatible ~ ., data=upper_kg_middle_neg_noname, method = "class", control = rpart.control(cp = 0, minsplit = 1))
rpart.plot(upper_kg_middle_neg_tree, type = 5)

upper_kg_middle_neg_lower <- training2 %>% filter(X353 == "K" & X354 == "G" & X31 %in% c("D","E")) %>% select(X79, X82, compatible)
table(upper_kg_middle_neg_lower)

upper_kg_middle_neg_lower_hphobic <- training2 %>% filter(X353 == "K" & X354 == "G" & X31 %in% c("D","E") & X79 %in% c("I","L")) %>% select(X79, X82, compatible, name)

key_residues <- training2 %>% select(X31, X34, X79, X82, X353, X354, X355, compatible, name)
key_residues_human <- training2 %>% select(X31, X34, X79, X82, X353, X354, X355,compatible, name) %>% filter(name == "Homo sapiens")
```


```{r Perform a random forest}
#install.packages("randomForest")
library(randomForest)

rf_classification = randomForest(compatible ~ ., data=training3, ntree=300, mtry=sqrt(ncol(training3)-1), importance=TRUE, replace = T, confusion = T)

print(rf_classification)

## Pull the first tree out to see what it looks like
rf_tree_1 <- getTree(rf_classification, k=1, labelVar=T)

## This plot tells you the relationship of how the number of trees correlates with error
plot(rf_classification)

error_table <- data.frame(rf_classification$err.rate)
error_table$tree_num <- seq(1,nrow(error_table))

rf_classification_tree_plot <- ggplot() + theme_bw() + scale_y_continuous(limits = c(0,0.5)) +
  labs(x = "Number of trees", y = "Mean Squared Error") +
  geom_line(data = error_table, aes(x = tree_num, y = OOB))
rf_classification_tree_plot
#ggsave(file = "plots/rf_regression_tree_plot.png", rf_regression_tree_plot, height = 3, width = 4)

## Seeing which features were giving the most information
feature_frame <- data.frame("MeanDecreaseAccuracy" = importance(rf_classification,type = 1))
feature_frame$name <- rownames(feature_frame)
feature_frame <- feature_frame[rev(order(feature_frame$MeanDecreaseAccuracy)),]
feature_frame$name <- factor(feature_frame$name, levels = rev(feature_frame$name))

feature_frame2 <- data.frame("MeanDecreaseGini" = importance(rf_classification,type = 2))
feature_frame2$name <- rownames(feature_frame2)
feature_frame2 <- feature_frame2[rev(order(feature_frame2$MeanDecreaseGini)),]
feature_frame2$name <- factor(feature_frame2$name, levels = rev(feature_frame2$name))

rf_classification_importance_plot <- ggplot() + theme_bw() + 
  labs(x = "MeanDecreaseAccuracy upon permutation", y = NULL) +
  geom_point(data = feature_frame[1:30,], aes(x = MeanDecreaseAccuracy, y = name))
rf_classification_importance_plot
ggsave(file = "plots/rf_classification_importance_plot.png", rf_classification_importance_plot, height = 4, width = 3.5)

rf_classification_importance_plot2 <- ggplot() + theme_bw() + 
  labs(x = "MeanDecreaseGini upon permutation", y = NULL) +
  geom_point(data = feature_frame2[1:10,], aes(x = MeanDecreaseGini, y = name))
rf_classification_importance_plot2
ggsave(file = "plots/rf_classification_importance_plot2.png", rf_classification_importance_plot2, height = 4, width = 3.5)

feature_Frame2 <- data.frame(varImpPlot(rf_classification))
## https://stats.stackexchange.com/questions/162465/in-a-random-forest-is-larger-incmse-better-or-worse#:~:text=IncNodePurity%20relates%20to%20the%20loss,small%20intra%20node%20'variance'.

```

```{r rf predictions}

prediction_for_table <- predict(rf_classification,training3[,colnames(training3) != "compatible"])
length(prediction_for_table)

sars2_rf_predicted <- data.frame("name" = training2$name, "compatible" = training3$compatible, "predicted" = prediction_for_table)

table(sars2_rf_predicted[,c("compatible","predicted")])

sars2_rf_predicted$matching <- sars2_rf_predicted$compatible == sars2_rf_predicted$predicted
```







### Remaining is scratch or work in progress









```{r}


ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2beta, y = s1, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2beta, y = s1, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2beta, y = wiv1, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2beta, y = wiv1, label = paste(ortholog,mutant,sep="-")), size = 3)
```


```{r}

```


```{r Heatmap?}
low_kozak <- dbl_combined2 %>% filter(kozak == "low") %>% filter(mutant != "E329K")
low_kozak2 <- low_kozak[,c("mutant","g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")]

low_kozak2_melt <- melt(low_kozak2, id = "mutant")

ggplot() + scale_fill_gradient(low = "white", high = "red") + 
  geom_tile(data = low_kozak2_melt, aes(x = mutant, y = variable, fill = value))
```

```{r SARS2_variant_comparisons}

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2alpha, y = s2beta, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2alpha, y = s2beta, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2alpha, y = s2delta, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2alpha, y = s2delta, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2alpha, y = s2gamma, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2alpha, y = s2gamma, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2alpha, y = s2omicron, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2alpha, y = s2omicron, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2beta, y = s2delta, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2beta, y = s2delta, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2beta, y = s2gamma, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2beta, y = s2gamma, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2beta, y = s2omicron, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2beta, y = s2omicron, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2delta, y = s2gamma, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2delta, y = s2gamma, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2delta, y = s2omicron, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2delta, y = s2omicron, label = paste(ortholog,mutant,sep="-")), size = 3)

ggplot() + scale_x_log10() + scale_y_log10() + 
  geom_point(data = dbl_combined2, aes(x = s2gamma, y = s2omicron, color = protease)) +
  geom_text_repel(data = dbl_combined2, aes(x = s2gamma, y = s2omicron, label = paste(ortholog,mutant,sep="-")), size = 3)
```




```{r Import the flow data as well as relevant samples keys}

combined_infection <- read.csv(file = "Data/Receptor_dependence_data.csv")

recombined_construct_key <- read.csv(file = "Data/Keys/Construct_label_key.csv", header = T, stringsAsFactors = F)
pseudovirus_label_key <- read.csv(file = "Data/Keys/Pseudovirus_label_key.csv", header = T, stringsAsFactors = F) #%>% filter(sequence_confirmed != "no")
clade_labels_key <- read.csv(file = "Data/Keys/Clade_labels.csv", header = T, stringsAsFactors = F)
clade_labels_key$clade = factor(clade_labels_key$clade, levels = c(1,2,3,4))

```

```{r Quality filter the infection data}

# Two-color data
two_color_precombined <- combined_infection %>% filter(expt %in% c("New_constructs") & !(is.na(pseudovirus_env) | pseudovirus_env == "None") & live_singlets != "Too few infected") %>% filter(!is.na(pct_grn_gvn_red))

## Make sure I ignore data where I didn't run enough cells to make a conclusion
two_color_precombined$live_singlets <- as.numeric(two_color_precombined$live_singlets)
two_color_precombined$pct_red  <- as.numeric(two_color_precombined$pct_red)
two_color_precombined$pct_nir <- as.numeric(two_color_precombined$pct_nir)

two_color_precombined$lower_bound_red <- 10^(log10(two_color_precombined$live_singlets * two_color_precombined$pct_red / 100)*-0.99 + 2.64)
two_color_precombined$lower_bound_nir <- 10^(log10(two_color_precombined$live_singlets * two_color_precombined$pct_nir / 100)*-0.99 + 2.64)
two_color_precombined$passes_lower_bound <- "no"
two_color_precombined$passes_upper_bound <- "no"
for(x in 1:nrow(two_color_precombined)){
  if(two_color_precombined$lower_bound_red[x] <= two_color_precombined$pct_grn_gvn_red[x] & two_color_precombined$lower_bound_nir[x] <= two_color_precombined$pct_grn_gvn_nir[x]){two_color_precombined$passes_lower_bound[x] <- "yes"}
  if(two_color_precombined$pct_grn_gvn_red[x] <= 90 & two_color_precombined$pct_grn_gvn_nir[x] <= 90){two_color_precombined$passes_upper_bound[x] <- "yes"}
}
two_color_combined <- merge(two_color_precombined %>% filter(passes_lower_bound == "yes" & passes_upper_bound == "yes"), pseudovirus_label_key, by = "pseudovirus_env", all.x = T) %>% mutate(ratio_nir_mcherry = pct_nir / pct_red)

```

```{r}
new_constructs <- two_color_precombined %>% filter(expt %in% c("New_constructs") & date != "200107")

new_constructs$red_cells <- new_constructs$live_singlets * new_constructs$pct_red / 100
new_constructs$red_then_grn_cells <- round(new_constructs$red_cells * new_constructs$pct_grn_gvn_red / 100,0)
new_constructs$nir_cells <- new_constructs$live_singlets * new_constructs$pct_nir / 100
new_constructs$nir_then_grn_cells <- round(new_constructs$nir_cells * new_constructs$pct_grn_gvn_nir / 100,0)

new_constructs_2 <- merge(new_constructs, pseudovirus_label_key, by = "pseudovirus_env")

new_constructs_2$log10_nir_red_diff <- log10(new_constructs_2$moi_gvn_nir) - log10(new_constructs_2$moi_gvn_red)
new_constructs_2$fold_ace2_dep_infection <- 10^new_constructs_2$log10_nir_red_diff

flow_ace2_muts_summary <- new_constructs_2 %>% filter(!(fold_ace2_dep_infection %in% c(NaN,0,Inf))) %>% group_by(virus_label, recombined_construct) %>% summarize(ratio = 10^mean(log10(fold_ace2_dep_infection)), .groups = "drop")

Human_v_alcyone <- ggplot() + 
  theme(axis.text.x.bottom = element_text(angle = -45, hjust = 0), panel.grid.major.x = element_blank(), panel.grid.minor = element_blank(), legend.position = "top") +
  labs(x = NULL, y = "Fold ACE2\n-dependent infection") + scale_y_log10() + 
  geom_hline(yintercept = 1, alpha = 0.2, size = 1.5) +
  geom_point(data = new_constructs_2, aes(x = virus_label, y = fold_ace2_dep_infection, color = recombined_construct), position = position_dodge(width = 0.8), alpha = 0.4) +
  geom_point(data = flow_ace2_muts_summary, aes(x = virus_label, y = ratio, color = recombined_construct), position = position_dodge(width = 0.8), alpha = 0.8, shape = 95, size = 10) +
#  facet_grid(rows = vars(cell_label)) +
  NULL
Human_v_alcyone
ggsave(file = "Plots/Human_v_alcyone.pdf", Human_v_alcyone, height = 3, width = 5)
```

```{r Initial RBD identity matrix}

rbd_parsed_df <- read.delim(file = "Data/Alignment/221004_All_RBDs/RBDs_aligned_identity_matrix_items.tsv", header = F, stringsAsFactors = F, sep = "\t")
rbd_dist_matrix <- read.delim(file = "Data/Alignment/221004_All_RBDs/RBDs_aligned_identity_matrix.csv", header = F, stringsAsFactors = F, sep = ",")

colnames(rbd_dist_matrix) <- rbd_parsed_df$V1 #paste("c_",rbd_parsed_df$V1,sep="")
rownames(rbd_dist_matrix) <- rbd_parsed_df$V1 #paste("r_",rbd_parsed_df$V1,sep="")

rbd_dist_matrix_hclust <- hclust(as.dist(rbd_dist_matrix, diag = TRUE))

rbd_dist_matrix2 <- rbd_dist_matrix
rbd_dist_matrix2$species <- rownames(rbd_dist_matrix2)

rbd_dist_melted <- melt(rbd_dist_matrix2, id = "species")
rbd_dist_melted$variable <- as.character(rbd_dist_melted$variable)

rbd_dist_melted$species <- factor(rbd_dist_melted$species, levels = rbd_dist_matrix_hclust$labels)
rbd_dist_melted$variable <- factor(rbd_dist_melted$variable, levels = rbd_dist_matrix_hclust$labels)

rbd_identity_matrix_plot <- ggplot() + theme_bw() +
  theme(axis.text = element_text(size = 8), axis.text.x = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  labs(x = NULL, y = NULL, fill = NULL) +
  scale_fill_continuous(low = "white", high = "black") +
  geom_tile(data = rbd_dist_melted, aes(x = species, y = variable, fill = value)) +
  geom_text(data = rbd_dist_melted %>% filter(species != "H.sapiens" & variable != "H.sapiens" & value <= 40), aes(x = species, y = variable, label = value), size = 2) +
  geom_text(data = rbd_dist_melted %>% filter(species != "H.sapiens" & variable != "H.sapiens" & value >= 40), aes(x = species, y = variable, label = value), size = 2, color = "grey70") +
  NULL
ggsave(file = "Plots/RBD_identity_matrix_plot.pdf", rbd_identity_matrix_plot, height = 5.1*3, width = 6*3)
#rbd_identity_matrix_plot
```

```{r Use multi-dimensional scaling to visualize the sequence data for the conclusion slide}
# https://rpubs.com/sinhrks/plot_mds

rbd_dist_matrix3 <- rbd_dist_matrix
for(x in 1:nrow(rbd_dist_matrix3)){rownames(rbd_dist_matrix3)[x] <- strsplit(rownames(rbd_dist_matrix3)[x], "__",1)[[1]][1]}

mds_plotting_frame <- data.frame(cmdscale(rbd_dist_matrix3, eig = TRUE)$points)
mds_plotting_frame$name <- row.names(mds_plotting_frame)

mds_plotting_frame2 <- merge(mds_plotting_frame, clade_labels_key, by = "name", all.x = T)

ACE2_receptor_usage_by_sequence_matrix <- 
ggplot() + #scale_x_continuous(limits = c(-50, 50)) + scale_y_continuous(limits = c(-30, 50)) + 
  theme(panel.grid.minor = element_blank(), panel.grid.major = element_blank()) + 
  labs(x = "Scaled dimension 1", y = "Scaled dimension 2") +
  #geom_hline(yintercept = 0, alpha = 0.2) + geom_vline(xintercept = 0, alpha = 0.2) +
  geom_point(data = mds_plotting_frame2, aes(x = X1, y = X2, color = clade), alpha = 0.4) +
  geom_text_repel(data = mds_plotting_frame2 %>% filter(name %in% c("BtKY72","SARS-CoV-2","SARS-CoV","YN2013","RaTG15","Khosta-2","Khosta-1", "Rc-o319", "Rc-kw8", "YN2016E", "Rs4084", "RShSTT182", "BM48-31", "BB9904", "RhGB01", "RsSHC014")), aes(x = X1, y = X2, label = name), segment.color = "orange", segment.alpha = 0.5, size = 2)
ACE2_receptor_usage_by_sequence_matrix

ggsave(file = "Plots/ACE2_receptor_usage_by_sequence_matrix.pdf", ACE2_receptor_usage_by_sequence_matrix, height = 1.7*2, width = 3.7*2)

#write.csv(file = "Supp_tables/Export_csv_for_S2_Data/Fig_8.csv", mds_plotting_frame2, quote = FALSE, row.names = FALSE)
```

















```{r Clade-wise differences in RBD sequences}

rbd_parsed_df <- read.delim(file = "Data/Alignment/221004_All_RBDs/RBDs_aligned_identity_matrix_items.tsv", header = F, stringsAsFactors = F, sep = "\t")
rbd_dist_matrix <- read.delim(file = "Data/Alignment/221004_All_RBDs/RBDs_aligned_identity_matrix.csv", header = F, stringsAsFactors = F, sep = ",")

colnames(rbd_dist_matrix) <- rbd_parsed_df$V1 #paste("c_",rbd_parsed_df$V1,sep="")
rownames(rbd_dist_matrix) <- rbd_parsed_df$V1 #paste("r_",rbd_parsed_df$V1,sep="")

rbd_dist_matrix_hclust <- hclust(as.dist(rbd_dist_matrix, diag = TRUE))

rbd_dist_matrix2 <- rbd_dist_matrix
rbd_dist_matrix2$species <- rownames(rbd_dist_matrix2)

rbd_dist_melted <- melt(rbd_dist_matrix2, id = "species")
rbd_dist_melted$variable <- as.character(rbd_dist_melted$variable)

rbd_dist_melted$species <- factor(rbd_dist_melted$species, levels = rbd_dist_matrix_hclust$labels)
rbd_dist_melted$variable <- factor(rbd_dist_melted$variable, levels = rbd_dist_matrix_hclust$labels)

rbd_identity_matrix_plot <- ggplot() + theme_bw() +
  theme(axis.text = element_text(size = 8), axis.text.x = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  labs(x = NULL, y = NULL, fill = NULL) +
  scale_fill_continuous(low = "white", high = "black") +
  geom_tile(data = rbd_dist_melted, aes(x = species, y = variable, fill = value)) +
  geom_text(data = rbd_dist_melted %>% filter(species != "H.sapiens" & variable != "H.sapiens" & value <= 40), aes(x = species, y = variable, label = value), size = 2) +
  geom_text(data = rbd_dist_melted %>% filter(species != "H.sapiens" & variable != "H.sapiens" & value >= 40), aes(x = species, y = variable, label = value), size = 2, color = "grey70") +
  NULL
#rbd_identity_matrix_plot
ggsave(file = "Plots/All_RBD_identity_matrix_plot.pdf", rbd_identity_matrix_plot, height = 5.3*2, width = 6*2)


clade_frame <- data.frame("species" = c("Yunnan2011__C.plicata","Rf4092__Rhinolophus_ferrumequium","ZXC21__Rhinolophus_sinicus","279-2005_R.macrotis","Rs4237__Rhinolophus_sinicus","Rp3__Rhinolophus_pearsonii","JL2012__Rhinolophus_ferrumequinum","73-2005__Rhinolophus_ferrumequium","Rf1__Rhinolophus_ferrumequium","HeB2013__Rhinolophus_ferrumequium","HuB2013__Rhinolophus_sinicus","Rs806/2006__Rhinolophus_sinicus","tRI-SC2018__Rhinolophus","Shaanxi2011__Rhinolophus_pusillus","Rs4081__Rhinolophus_sinicus","Rs4237__Rhinolophus_sinicus",
"YN2013__Rhinolophus_sinicus","ZC45__R.sinicus","SARS_CoV-2__Homo_sapiens","RaTG13__Rhinolophus_affinis",
"Rs4084__Rhinolophus_sinicus","Rs4231__Rhinolophus_sinicus","RsSHC014__Rhinolophus_sinicus","SARS_CoV__Homo_sapiens",
"RaTG15__Rhinolophus_affinis","LYRa11__Rhinolophus_affinis","Rs7327__Rhinolophus_sinicus","WIV1__Rhinolophus_sinicus",
"RhGB01__Rhinolophus_hipposideros","Khosta2__Rhinolophus_hipposideros","BtKY72__Rhinolophus","Khosta1__Rhinolophus_ferrumequinum",
"BB9904__Rhinolophus_euryale","BM48-31__Rhinolophus_blasii","PRD-2386__Rhinolophus","PRD-0038__Rhinolophus"), "clade_species" = c(2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,1,1,1,1,1,1,1,3,3,3,3,3,3,3,3))
clade_frame$variable <- clade_frame$species
clade_frame$clade_variable <- clade_frame$clade_species

rbd_dist_melted2 <- merge(rbd_dist_melted, clade_frame[,c("species","clade_species")], by = "species")
rbd_dist_melted3 <- merge(rbd_dist_melted2, clade_frame[,c("variable","clade_variable")], by = "variable")

Clade_difference_plot <- ggplot() + theme(panel.grid.minor.x = element_blank(), panel.grid.major.x = element_blank()) + 
  labs(x = "Clade", y = "Amino acid\ndifferences") +
  scale_y_continuous(breaks = c(0,50,100), limits = c(0,100)) +
  geom_quasirandom(data = rbd_dist_melted3, aes(x = clade_species, y = value), alpha = 0.1, size = 0.25) + facet_grid(cols = vars(clade_variable))
ggsave(file = "Plots/Clade_difference_plot.pdf", Clade_difference_plot, height = 1.2, width = 2.5)
Clade_difference_plot
```





## SCRAP


```{r}
g928a <- makeExperimentFrame2(c(1,1,2,2)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1209a <- makeExperimentFrame2(c(3,3,4,4)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211a <- makeExperimentFrame2(c(5,5,6,6)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213a <- makeExperimentFrame2(c(7,7,8,8)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g928b <- makeExperimentFrame2(c(9,9,10,10)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211b <- makeExperimentFrame2(c(11,11,12,12)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213b <- makeExperimentFrame2(c(9,11,13,13)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1232a <- makeExperimentFrame2(c(14,14,15,15)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g928c <- makeExperimentFrame2(c(16,16,17,17)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1209b <- makeExperimentFrame2(c(18,18,19,19)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211c <- makeExperimentFrame2(c(20,20,21,21)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213c <- makeExperimentFrame2(c(22,22,23,23)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1232b <- makeExperimentFrame2(c(24,24,25,25)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g928d <- makeExperimentFrame2(c(26,26,27,27)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1209c <- makeExperimentFrame2(c(26,26,28,28)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211d <- makeExperimentFrame2(c(26,26,29,29)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213d <- makeExperimentFrame2(c(26,26,30,30)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1232c <- makeExperimentFrame2(c(26,26,31,31)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1194a <- makeExperimentFrame2(c(32,32,33,33)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1121a <- makeExperimentFrame2(c(32,32,34,34)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1195a <- makeExperimentFrame2(c(35,35,36,36)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1230a <- makeExperimentFrame2(c(37,37,38,38)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))

g928a <- makeExperimentFrame2(c(39,39,2,2)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1209a <- makeExperimentFrame2(c(39,39,4,4)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211a <- makeExperimentFrame2(c(39,39,6,6)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213a <- makeExperimentFrame2(c(39,39,8,8)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g928b <- makeExperimentFrame2(c(39,39,10,10)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211b <- makeExperimentFrame2(c(39,39,12,12)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213b <- makeExperimentFrame2(c(39,39,13,13)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1232a <- makeExperimentFrame2(c(39,39,15,15)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g928c <- makeExperimentFrame2(c(39,39,17,17)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1209b <- makeExperimentFrame2(c(39,39,19,19)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211c <- makeExperimentFrame2(c(39,39,21,21)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213c <- makeExperimentFrame2(c(39,39,23,23)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1232b <- makeExperimentFrame2(c(39,39,25,25)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g928d <- makeExperimentFrame2(c(39,39,27,27)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1209c <- makeExperimentFrame2(c(39,39,28,28)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1211d <- makeExperimentFrame2(c(39,39,29,29)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1213d <- makeExperimentFrame2(c(39,39,30,30)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1232c <- makeExperimentFrame2(c(39,39,31,31)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1194a <- makeExperimentFrame2(c(39,39,33,33)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1121a <- makeExperimentFrame2(c(39,39,34,34)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1195a <- makeExperimentFrame2(c(39,39,36,36)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
g1230a <- makeExperimentFrame2(c(39,39,38,38)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment))
```


```{r}
g740d <- makeExperimentFrame2(c(4,5,18,19)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 195, 196, 209, 210
s2alpha <- makeExperimentFrame2(c(6,7,20,21)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 197, 198, 211, 212
s2beta <- makeExperimentFrame2(c(8,9,22,23)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 199, 200, 213, 214
s2delta <- makeExperimentFrame2(c(10,11,24,25)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 201, 202, 215, 216
s2gamma <- makeExperimentFrame2(c(12,13,26,27)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 203, 204, 217, 218
s2omicron <- makeExperimentFrame2(c(14,15,28,29)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 205, 206, 219, 220
s1<- makeExperimentFrame2(c(16,16,30,30)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 207, 221
wiv1 <- makeExperimentFrame2(c(17,17,31,31)) %>% filter(h_enrichment != "Inf") %>% arrange(desc(h_enrichment)) ## 208, 222

dbl_combined <- merge(merge(merge(merge(merge(merge(merge(g740d[,c("sequence","gene","ortholog","mutant","protease","kozak","plasmid_template","concat","h_enrichment")], s2alpha[,c("sequence","h_enrichment")], by = "sequence", all = T),s2beta[,c("sequence","h_enrichment")], by = "sequence", all = T),s2delta[,c("sequence","h_enrichment")], by = "sequence", all = T),s2gamma[,c("sequence","h_enrichment")], by = "sequence", all = T),s2omicron[,c("sequence","h_enrichment")], by = "sequence", all = T),s1[,c("sequence","h_enrichment")], by = "sequence", all = T),wiv1[,c("sequence","h_enrichment")], by = "sequence", all = T)
colnames(dbl_combined) <- c("sequence","gene","ortholog","mutant","protease","kozak","plasmid_template","concat","g740d","s2alpha","s2beta","s2delta","s2gamma","s2omicron","s1","wiv1")

dbl_combined2 <- dbl_combined
```
